Index: system/application/controllers/office/crosswords.php
===================================================================
--- system/application/controllers/office/crosswords.php (revision 0e81e4d38b30760ff5ad69b0ba410340662d763a)
+++ system/application/controllers/office/crosswords.php (revision 4b830c10e5bbb5edb455e471a5ab326e54cee248)
@@ -893,7 +893,7 @@
 				$data['Stats']['winners'] = $crossword_info['winners_so_far'];
 				$data['StatLabels'] = array(
-					//'saves' => 'Total number of saves (approx 30 sec between saves)',
+					//'saves' => 'Total number of saves',
 					'save_users' => 'Number of users who have attempted crossword (based on saves)',
-					'save_mean_per_user' => 'Mean number of saves per user',
+					'save_mean_per_user' => 'Mean number of saves per user (approx 30 secs between saves)',
 					'winners' => 'Number of winners',
 				);
