Index: design/theyorker_tables.sql
===================================================================
--- design/theyorker_tables.sql (revision 187424fc1e007d128a6d565d2427fae67b9292b2)
+++ design/theyorker_tables.sql (revision 6e624aaffdd97ae4bd7337b2227cafd541fb1f73)
@@ -328,7 +328,7 @@
 DROP TABLE IF EXISTS article_photos;
 CREATE TABLE article_photos (
-	article_photo_article_id	INTEGER		NOT NULL,
-	article_photo_photo_id	INTEGER		NOT NULL,
-	article_photo_number		INTEGER		NOT NULL,
+	article_photo_article_id			INTEGER		NOT NULL,
+	article_photo_photo_id				INTEGER		NOT NULL,
+	article_photo_number				INTEGER		NOT NULL,
 
 	PRIMARY KEY(article_photo_article_id, article_photo_photo_id)
@@ -337,18 +337,18 @@
 DROP TABLE IF EXISTS article_tags;
 CREATE TABLE article_tags (
-	article_tag_article_id	INTEGER		NOT NULL,
-	article_tag_tag_id		INTEGER		NOT NULL,
-	
+	article_tag_article_id	                        INTEGER		NOT NULL,
+	article_tag_tag_id		                INTEGER		NOT NULL,
+
 	PRIMARY KEY(article_tag_article_id, article_tag_tag_id)
 );
 
--- News, reviews features, lifstyle etc..  Can be in a hierachy.  
+-- News, reviews features, lifstyle etc..  Can be in a hierachy.
 DROP TABLE IF EXISTS article_types;
 CREATE TABLE article_types (
-	article_type_id		INTEGER		NOT NULL	AUTO_INCREMENT,
-	article_type_parent_article_type_id INTEGER	NULL,
-	article_type_name		VARCHAR(255)	NOT NULL,
-	article_type_archive		BOOL		NOT NULL,
-	article_type_blurb		TEXT		NOT NULL,
+	article_type_id		                        INTEGER		NOT NULL	AUTO_INCREMENT,
+	article_type_parent_article_type_id             INTEGER	        NULL,
+	article_type_name		                VARCHAR(255)	NOT NULL,
+	article_type_archive		                BOOL		NOT NULL,
+	article_type_blurb		                TEXT		NOT NULL,
 	
 	PRIMARY KEY(article_type_id)
@@ -358,6 +358,6 @@
 DROP TABLE IF EXISTS article_writers;
 CREATE TABLE article_writers (
-	article_writer_user_entity_id		INTEGER		NOT NULL,
-	article_writer_article_content_id INTEGER	NOT NULL,
+	article_writer_user_entity_id		        INTEGER		NOT NULL,
+	article_writer_article_content_id               INTEGER	        NOT NULL,
 	
 	PRIMARY KEY(article_writer_user_entity_id, article_writer_article_content_id)
@@ -366,10 +366,10 @@
 DROP TABLE IF EXISTS fact_boxes;
 CREATE TABLE fact_boxes (
-	fact_box_id		INTEGER		NOT NULL	AUTO_INCREMENT,
-	fact_box_article_content_id INTEGER	NOT NULL,
-	fact_box_text		TEXT		NOT NULL,
-	fact_box_deleted	BOOL		NOT NULL	DEFAULT FALSE,
-	fact_box_timestamp	TIMESTAMP	NOT NULL	DEFAULT CURRENT_TIMESTAMP,
-	
+	fact_box_id		                        INTEGER		NOT NULL	AUTO_INCREMENT,
+	fact_box_article_content_id                     INTEGER	        NOT NULL,
+	fact_box_text		                        TEXT		NOT NULL,
+	fact_box_deleted	                        BOOL		NOT NULL	DEFAULT FALSE,
+	fact_box_timestamp	                        TIMESTAMP	NOT NULL	DEFAULT CURRENT_TIMESTAMP,
+
 	PRIMARY KEY(fact_box_id)
 );
@@ -377,6 +377,6 @@
 DROP TABLE IF EXISTS related_articles;
 CREATE TABLE related_articles (
-	related_article_1_article_id	INTEGER		NOT NULL,
-	related_article_2_article_id	INTEGER		NOT NULL,
+	related_article_1_article_id	                INTEGER		NOT NULL,
+	related_article_2_article_id	                INTEGER		NOT NULL,
 
 	PRIMARY KEY(related_article_1_article_id, related_article_2_article_id)
@@ -385,11 +385,11 @@
 DROP TABLE IF EXISTS pull_quotes;
 CREATE TABLE pull_quotes (
-	pull_quote_id INTEGER NOT NULL AUTO_INCREMENT,
-	pull_quote_article_content_id INTEGER NOT NULL,
-	pull_quote_text TEXT NOT NULL,
-	pull_quote_person VARCHAR(255) NOT NULL,
-	pull_quote_position VARCHAR(255) NOT NULL,
-	pull_quote_order INTEGER NOT NULL,
-	pull_quote_deleted BOOL NOT NULL,
+	pull_quote_id                                   INTEGER         NOT NULL        AUTO_INCREMENT,
+	pull_quote_article_content_id                   INTEGER         NOT NULL,
+	pull_quote_text                                 TEXT            NOT NULL,
+	pull_quote_person                               VARCHAR(255)    NOT NULL,
+	pull_quote_position                             VARCHAR(255)    NOT NULL,
+	pull_quote_order                                INTEGER         NOT NULL,
+	pull_quote_deleted                              BOOL            NOT NULL,
 
 	PRIMARY KEY(pull_quote_id)
@@ -398,18 +398,18 @@
 DROP TABLE IF EXISTS requests;
 CREATE TABLE requests (
-	request_id INTEGER NOT NULL AUTO_INCREMENT,
-	request_timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+	request_id                                      INTEGER         NOT NULL        AUTO_INCREMENT,
+	request_timestamp                               TIMESTAMP       NOT NULL        DEFAULT CURRENT_TIMESTAMP,
 		-- Who requested the thing
-	request_entity_id INTEGER NOT NULL,
+	request_entity_id                               INTEGER         NOT NULL,
 		-- An article is created with a request for an article
-	request_article_id INTEGER NOT NULL,
-	request_type_id INTEGER NOT NULL,
-	request_article_type_id INTEGER NOT NULL,
-	request_organisation_entity_id INTEGER NULL,
-	request_text TEXT NOT NULL,
-	request_blurb TEXT NOT NULL,
-	request_deadline TIMESTAMP NOT NULL,
-	request_accepted BOOL NOT NULL,
-	request_deleted BOOL NOT NULL,
+	request_article_id                              INTEGER         NOT NULL,
+	request_type_id                                 INTEGER         NOT NULL,
+	request_article_type_id                         INTEGER         NOT NULL,
+	request_organisation_entity_id                  INTEGER         NULL,
+	request_text                                    TEXT            NOT NULL,
+	request_blurb                                   TEXT            NOT NULL,
+	request_deadline                                TIMESTAMP       NOT NULL,
+	request_accepted                                BOOL            NOT NULL,
+	request_deleted                                 BOOL            NOT NULL,
 	
 	PRIMARY KEY(request_id)
@@ -418,8 +418,8 @@
 DROP TABLE IF EXISTS request_photos;
 CREATE TABLE request_photos (
-	request_photo_request_id INTEGER NOT NULL,
-	request_photo_photo_id INTEGER NULL,
-	request_photo_deleted BOOL NOT NULL,
-	request_photo_timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+	request_photo_request_id                        INTEGER         NOT NULL,
+	request_photo_photo_id                          INTEGER         NULL,
+	request_photo_deleted                           BOOL            NOT NULL,
+	request_photo_timestamp                         TIMESTAMP       NOT NULL         DEFAULT CURRENT_TIMESTAMP,
 
 	PRIMARY KEY(request_photo_request_id, request_photo_photo_id)
@@ -429,6 +429,6 @@
 DROP TABLE IF EXISTS request_types;
 CREATE TABLE request_types (
-	request_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	request_type_name VARCHAR(255) NOT NULL,
+	request_type_id                                 INTEGER         NOT NULL         AUTO_INCREMENT,
+	request_type_name                               VARCHAR(255)    NOT NULL,
 	
 	PRIMARY KEY(request_type_id)
@@ -437,8 +437,8 @@
 DROP TABLE IF EXISTS request_users;
 CREATE TABLE request_users (
-	request_user_request_id INTEGER NOT NULL,
-	request_user_user_entity_id INTEGER NOT NULL,
-	request_user_accepted BOOL NOT NULL,
-	request_user_rejected BOOL NOT NULL,
+	request_user_request_id                         INTEGER         NOT NULL,
+	request_user_user_entity_id                     INTEGER         NOT NULL,
+	request_user_accepted                           BOOL            NOT NULL,
+	request_user_rejected                           BOOL            NOT NULL,
 	
 	PRIMARY KEY(request_user_request_id, request_user_user_entity_id)
@@ -449,13 +449,13 @@
 DROP TABLE IF EXISTS events;
 CREATE TABLE events (
-	event_id INTEGER NOT NULL,
-	event_image_id INTEGER NOT NULL,
-	event_parent_id INTEGER NOT NULL,
-	event_type_id INTEGER NOT NULL,
-	event_name TEXT NULL,
-	event_description TEXT NULL,
-	event_blurb TEXT NULL,
-	event_deleted BOOL NULL,
-	event_timestamp TIMESTAMP NULL,
+	event_id                                        INTEGER          NOT NULL,
+	event_image_id                                  INTEGER          NOT NULL,
+	event_parent_id                                 INTEGER          NOT NULL,
+	event_type_id                                   INTEGER          NOT NULL,
+	event_name                                      TEXT             NULL,
+	event_description                               TEXT             NULL,
+	event_blurb                                     TEXT             NULL,
+	event_deleted                                   BOOL             NULL,
+	event_timestamp                                 TIMESTAMP        NULL,
 	
 	PRIMARY KEY(event_id)
@@ -465,6 +465,6 @@
 DROP TABLE IF EXISTS events_entities;
 CREATE TABLE events_entities (
-	event_entity_entity_id INTEGER NOT NULL,
-	event_entity_event_id INTEGER NOT NULL,
+	event_entity_entity_id                          INTEGER          NOT NULL,
+	event_entity_event_id                           INTEGER          NOT NULL,
 	
 	PRIMARY KEY(event_entity_entity_id, event_entity_event_id)
@@ -474,8 +474,8 @@
 DROP TABLE IF EXISTS event_occurrence_users;
 CREATE TABLE event_occurrence_users (
-	event_occurrence_user_user_entity_id INTEGER NOT NULL,
-	event_occurrence_user_event_occurrence_id INTEGER NOT NULL,
-	event_occurrence_user_hide BOOL NOT NULL,
-	event_occurrence_user_rsvp BOOL NOT NULL,
+	event_occurrence_user_user_entity_id            INTEGER          NOT NULL,
+	event_occurrence_user_event_occurrence_id       INTEGER          NOT NULL,
+	event_occurrence_user_hide                      BOOL             NOT NULL,
+	event_occurrence_user_rsvp                      BOOL             NOT NULL,
 	
 	PRIMARY KEY(event_occurrence_user_user_entity_id, event_occurrence_user_event_occurrence_id)
@@ -484,16 +484,16 @@
 DROP TABLE IF EXISTS event_occurrences;
 CREATE TABLE event_occurrences (
-	event_occurrence_id INTEGER NOT NULL,
-	event_occurrence_timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
-	event_occurrence_next_id INTEGER NOT NULL,
-	event_occurrence_event_id INTEGER NOT NULL,
-	event_occurrence_state_id INTEGER NOT NULL,
-	event_occurrence_description TEXT NOT NULL,
-	event_occurrence_location VARCHAR(15) NOT NULL,
-	event_occurrence_postcode VARCHAR(15) NOT NULL,
-	event_occurrence_start_time TIMESTAMP NOT NULL,
-	event_occurrence_end_time TIMESTAMP NOT NULL,
-	event_occurrence_all_day BOOL NOT NULL,
-	event_occurrence_ends_late BOOL NOT NULL,
+	event_occurrence_id                             INTEGER          NOT NULL,
+	event_occurrence_timestamp                      TIMESTAMP        NOT NULL        DEFAULT CURRENT_TIMESTAMP,
+	event_occurrence_next_id                        INTEGER          NOT NULL,
+	event_occurrence_event_id                       INTEGER          NOT NULL,
+	event_occurrence_state_id                       INTEGER          NOT NULL,
+	event_occurrence_description                    TEXT             NOT NULL,
+	event_occurrence_location                       VARCHAR(15)      NOT NULL,
+	event_occurrence_postcode                       VARCHAR(15)      NOT NULL,
+	event_occurrence_start_time                     TIMESTAMP        NOT NULL,
+	event_occurrence_end_time                       TIMESTAMP        NOT NULL,
+	event_occurrence_all_day                        BOOL             NOT NULL,
+	event_occurrence_ends_late                      BOOL             NOT NULL,
 	
 	PRIMARY KEY(event_occurrence_id)
@@ -502,7 +502,7 @@
 DROP TABLE IF EXISTS event_occurrence_states;
 CREATE TABLE event_occurrence_states (
-	event_occurrence_state_id INTEGER NOT NULL AUTO_INCREMENT,
-	event_occurrence_state_name VARCHAR(255) NOT NULL,
-	event_occurrence_state_published BOOL NOT NULL,
+	event_occurrence_state_id                       INTEGER          NOT NULL        AUTO_INCREMENT,
+	event_occurrence_state_name                     VARCHAR(255)     NOT NULL,
+	event_occurrence_state_published                BOOL             NOT NULL,
 	
 	PRIMARY KEY(event_occurrence_state_id)
@@ -512,6 +512,6 @@
 DROP TABLE IF EXISTS event_types;
 CREATE TABLE event_types (
-	event_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	event_type_name VARCHAR(255) NOT NULL,
+	event_type_id                                   INTEGER          NOT NULL        AUTO_INCREMENT,
+	event_type_name                                 VARCHAR(255)     NOT NULL,
 	
 	PRIMARY KEY(event_type_id)
@@ -520,9 +520,9 @@
 DROP TABLE IF EXISTS reminders;
 CREATE TABLE reminders (
-	reminder_id INTEGER NOT NULL AUTO_INCREMENT,
-	reminder_user_entity_id INTEGER NOT NULL,
-	reminder_name VARCHAR(255) NOT NULL,
-	reminder_description TEXT NOT NULL,
-	reminder_timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+	reminder_id                                     INTEGER          NOT NULL        AUTO_INCREMENT,
+	reminder_user_entity_id                         INTEGER          NOT NULL,
+	reminder_name                                   VARCHAR(255)     NOT NULL,
+	reminder_description                            TEXT             NOT NULL,
+	reminder_timestamp                              TIMESTAMP        NOT NULL        DEFAULT CURRENT_TIMESTAMP,
 	
 	PRIMARY KEY(reminder_id)
@@ -531,8 +531,8 @@
 DROP TABLE IF EXISTS anniversaries;
 CREATE TABLE anniversaries (
-	anniversary_id INTEGER NOT NULL AUTO_INCREMENT,
-	anniversary_entity_id INTEGER NOT NULL,
-	anniversary_name VARCHAR(255) NOT NULL,
-	anniversary_start_date TIMESTAMP NOT NULL,
+	anniversary_id                                  INTEGER          NOT NULL        AUTO_INCREMENT,
+	anniversary_entity_id                           INTEGER          NOT NULL,
+	anniversary_name                                VARCHAR(255)     NOT NULL,
+	anniversary_start_date                          TIMESTAMP        NOT NULL,
 	
 	PRIMARY KEY(anniversary_id)
@@ -541,14 +541,14 @@
 DROP TABLE IF EXISTS todo_list_items;
 CREATE TABLE todo_list_items (
-	todo_list_item_id INTEGER NOT NULL AUTO_INCREMENT,
-	todo_list_item_event_occurrence_id INTEGER NOT NULL,
-	todo_list_item_reminder_id INTEGER NOT NULL,
-	todo_list_item_todo_priority_id INTEGER NOT NULL,
-	todo_list_item_user_entity_id INTEGER NOT NULL,
-	todo_list_item_name VARCHAR(255) NOT NULL,
-	todo_list_item_description TEXT NOT NULL,
-	todo_list_item_done BOOL NOT NULL,
-	todo_list_item_event_occurence_id INTEGER NOT NULL,
-	todo_list_item_deadline TIMESTAMP NOT NULL,
+	todo_list_item_id                               INTEGER          NOT NULL        AUTO_INCREMENT,
+	todo_list_item_event_occurrence_id              INTEGER          NOT NULL,
+	todo_list_item_reminder_id                      INTEGER          NOT NULL,
+	todo_list_item_todo_priority_id                 INTEGER          NOT NULL,
+	todo_list_item_user_entity_id                   INTEGER          NOT NULL,
+	todo_list_item_name                             VARCHAR(255)     NOT NULL,
+	todo_list_item_description                      TEXT             NOT NULL,
+	todo_list_item_done                             BOOL             NOT NULL,
+	todo_list_item_event_occurence_id               INTEGER          NOT NULL,
+	todo_list_item_deadline                         TIMESTAMP        NOT NULL,
 	
 	PRIMARY KEY(todo_list_item_id)
@@ -557,7 +557,7 @@
 DROP TABLE IF EXISTS todo_priorities;
 CREATE TABLE todo_priorities (
-	todo_priority_id INTEGER NOT NULL AUTO_INCREMENT,
-	todo_priority_name VARCHAR(255) NOT NULL,
-	todo_priority_order INTEGER NOT NULL,
+	todo_priority_id                                INTEGER          NOT NULL        AUTO_INCREMENT,
+	todo_priority_name                              VARCHAR(255)     NOT NULL,
+	todo_priority_order                             INTEGER          NOT NULL,
 	
 	PRIMARY KEY(todo_priority_id)
@@ -566,11 +566,11 @@
 DROP TABLE IF EXISTS years;
 CREATE TABLE years (
-	year_id INTEGER NOT NULL,
-	start_autumn TIMESTAMP NOT NULL,
-	end_autumn TIMESTAMP NOT NULL,
-	start_spring TIMESTAMP NOT NULL,
-	end_spring TIMESTAMP NOT NULL,
-	start_summer TIMESTAMP NOT NULL,
-	end_summer TIMESTAMP NOT NULL,
+	year_id                                         INTEGER          NOT NULL,
+	start_autumn                                    TIMESTAMP        NOT NULL,
+	end_autumn                                      TIMESTAMP        NOT NULL,
+	start_spring                                    TIMESTAMP        NOT NULL,
+	end_spring                                      TIMESTAMP        NOT NULL,
+	start_summer                                    TIMESTAMP        NOT NULL,
+	end_summer                                      TIMESTAMP        NOT NULL,
 	
 	PRIMARY KEY(year_id)
@@ -583,8 +583,8 @@
 DROP TABLE IF EXISTS reviews;
 CREATE TABLE reviews (
-	review_id INTEGER NOT NULL AUTO_INCREMENT,
-	rto_review_type_id INTEGER NOT NULL,
-	rto_organisation_entity_id INTEGER NOT NULL,
-	review_article_id INTEGER NOT NULL,
+	review_id                                       INTEGER          NOT NULL        AUTO_INCREMENT,
+	rto_review_type_id                              INTEGER          NOT NULL,
+	rto_organisation_entity_id                      INTEGER          NOT NULL,
+	review_article_id                               INTEGER          NOT NULL,
 	
 	PRIMARY KEY(review_id)
@@ -593,10 +593,10 @@
 DROP TABLE IF EXISTS review_types;
 CREATE TABLE review_types (
-	review_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	review_type_article_type_id INTEGER NOT NULL,
-	review_type_image_id INTEGER NOT NULL,
-	review_type_name VARCHAR(255) NOT NULL,
-	review_type_blurb TEXT NOT NULL,
-	review_type_blurb_draft TEXT NOT NULL,
+	review_type_id                                  INTEGER          NOT NULL        AUTO_INCREMENT,
+	review_type_article_type_id                     INTEGER          NOT NULL,
+	review_type_image_id                            INTEGER          NOT NULL,
+	review_type_name                                VARCHAR(255)     NOT NULL,
+	review_type_blurb                               TEXT             NOT NULL,
+	review_type_blurb_draft                         TEXT             NOT NULL,
 	
 	PRIMARY KEY(review_type_id)
@@ -607,11 +607,11 @@
 DROP TABLE IF EXISTS review_type_organisations;
 CREATE TABLE review_type_organisations (
-	rto_organisation_entity_id INTEGER NOT NULL,
-	rto_review_type_id INTEGER NOT NULL,
-	rto_live_content_id INTEGER NOT NULL,
-	rto_user_rate_count INTEGER NOT NULL,
-	rto_average_user_rating INTEGER NOT NULL,
-	rto_deleted BOOL NOT NULL,
-	rto_timestamp TIMESTAMP NOT NULL,
+	rto_organisation_entity_id                      INTEGER          NOT NULL,
+	rto_review_type_id                              INTEGER          NOT NULL,
+	rto_live_content_id                             INTEGER          NOT NULL,
+	rto_user_rate_count                             INTEGER          NOT NULL,
+	rto_average_user_rating                         INTEGER          NOT NULL,
+	rto_deleted                                     BOOL             NOT NULL,
+	rto_timestamp                                   TIMESTAMP        NOT NULL,
 	
 	PRIMARY KEY(rto_organisation_entity_id, rto_review_type_id)
@@ -621,16 +621,16 @@
 DROP TABLE IF EXISTS rto_content;
 CREATE TABLE rto_content (
-	rto_content_id INTEGER NOT NULL AUTO_INCREMENT,
-	rto_content_organisation_entity_id INTEGER NOT NULL,
-	rto_content_review_type_id INTEGER NOT NULL,
-	rto_content_cost_id INTEGER NOT NULL,
-	rto_content_blurb TEXT NOT NULL,
-	rto_content_price INTEGER NOT NULL,
-	rto_content_recommend TEXT NOT NULL,
-	rto_content_average_price_upper INTEGER NOT NULL,
-	rto_content_average_price_lower INTEGER NOT NULL,
-	rto_content_rating INTEGER NOT NULL,
-	rto_content_directions TEXT NOT NULL,
-	rto_content_book_online BOOL NOT NULL,
+	rto_content_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	rto_content_organisation_entity_id 		INTEGER 	NOT NULL,
+	rto_content_review_type_id			INTEGER 	NOT NULL,
+	rto_content_cost_id 				INTEGER 	NOT NULL,
+	rto_content_blurb 				TEXT 		NOT NULL,
+	rto_content_price 				INTEGER 	NOT NULL,
+	rto_content_recommend 				TEXT 		NOT NULL,
+	rto_content_average_price_upper 		INTEGER 	NOT NULL,
+	rto_content_average_price_lower 		INTEGER 	NOT NULL,
+	rto_content_rating 				INTEGER 	NOT NULL,
+	rto_content_directions 				TEXT 		NOT NULL,
+	rto_content_book_online 			BOOL 		NOT NULL,
 	
 	PRIMARY KEY(rto_content_id)
@@ -639,7 +639,7 @@
 DROP TABLE IF EXISTS rto_tags;
 CREATE TABLE rto_tags (
-	rto_tag_tag_id INTEGER NOT NULL,
-	rto_organisation_entity_id INTEGER NOT NULL,
-	rto_review_type_id INTEGER NOT NULL,
+	rto_tag_tag_id 					INTEGER 	NOT NULL,
+	rto_organisation_entity_id 			INTEGER 	NOT NULL,
+	rto_review_type_id 				INTEGER 	NOT NULL,
 
 	PRIMARY KEY(rto_tag_tag_id, rto_organisation_entity_id, rto_review_type_id)
@@ -648,6 +648,6 @@
 DROP TABLE IF EXISTS rto_costs;
 CREATE TABLE rto_costs (
-	rto_cost_id INTEGER NOT NULL AUTO_INCREMENT,
-	rto_cost_name VARCHAR(255) NOT NULL,
+	rto_cost_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	rto_cost_name 					VARCHAR(255) 	NOT NULL,
 
 	PRIMARY KEY(rto_cost_id)
@@ -656,8 +656,8 @@
 DROP TABLE IF EXISTS reviews_slideshows;
 CREATE TABLE reviews_slideshows (
-	reviews_slideshow_rto_review_type_id INTEGER NOT NULL,
-	reviews_slideshow_rto_organisation_entity_id INTEGER NOT NULL,
-	reviews_slideshow_photo_id INTEGER NOT NULL,
-	reviews_slideshow_order INTEGER NOT NULL,
+	reviews_slideshow_rto_review_type_id 		INTEGER 	NOT NULL,
+	reviews_slideshow_rto_organisation_entity_id 	INTEGER 	NOT NULL,
+	reviews_slideshow_photo_id 			INTEGER 	NOT NULL,
+	reviews_slideshow_order 			INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(reviews_slideshow_rto_review_type_id, reviews_slideshow_rto_organisation_entity_id, reviews_slideshow_order)
@@ -667,9 +667,9 @@
 DROP TABLE IF EXISTS bar_crawl_organisations;
 CREATE TABLE bar_crawl_organisations (
-	bar_crawl_organisation_bar_crawl_id INTEGER NOT NULL,
-	bar_crawl_organisation_organisation_entity_id INTEGER NOT NULL,
-	bar_crawl_organisation_order INTEGER NOT NULL,
-	bar_crawl_organisation_recommend VARCHAR(255) NOT NULL,
-	bar_crawl_organisation_recommend_price INTEGER NOT NULL,
+	bar_crawl_organisation_bar_crawl_id 		INTEGER 	NOT NULL,
+	bar_crawl_organisation_organisation_entity_id 	INTEGER 	NOT NULL,
+	bar_crawl_organisation_order 			INTEGER 	NOT NULL,
+	bar_crawl_organisation_recommend 		VARCHAR(255) 	NOT NULL,
+	bar_crawl_organisation_recommend_price 		INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(bar_crawl_organisation_bar_crawl_id, bar_crawl_organisation_organisation_entity_id)
@@ -679,9 +679,9 @@
 DROP TABLE IF EXISTS leagues;
 CREATE TABLE leagues (
-	league_id INTEGER NOT NULL AUTO_INCREMENT,
-	league_image_id INTEGER NOT NULL,
-	league_review_type_id INTEGER NOT NULL,
-	league_name VARCHAR(255) NOT NULL,
-	league_size INTEGER NOT NULL,
+	league_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	league_image_id 				INTEGER 	NOT NULL,
+	league_review_type_id 				INTEGER 	NOT NULL,
+	league_name 					VARCHAR(255) 	NOT NULL,
+	league_size 					INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(league_id)
@@ -690,7 +690,7 @@
 DROP TABLE IF EXISTS league_entries;
 CREATE TABLE league_entries (
-	league_entry_league_id INTEGER NOT NULL AUTO_INCREMENT,
-	league_entry_rto_organisation_entity_id INTEGER NOT NULL,
-	league_entry_position INTEGER NOT NULL,
+	league_entry_league_id 				INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	league_entry_rto_organisation_entity_id 	INTEGER 	NOT NULL,
+	league_entry_position 				INTEGER 	NOT NULL,
 
 	PRIMARY KEY(league_entry_league_id, league_entry_rto_organisation_entity_id)
@@ -702,12 +702,12 @@
 DROP TABLE IF EXISTS campaigns;
 CREATE TABLE campaigns (
-	campaign_id INTEGER NOT NULL AUTO_INCREMENT,
-	campaign_article_id INTEGER NOT NULL,
-	campaign_name VARCHAR(255) NOT NULL,
-	campaign_votes INTEGER NOT NULL,
-	campaign_petition BOOL NOT NULL,
-	campaign_petition_signatures INTEGER NOT NULL,
-	campaign_deleted BOOL NOT NULL,
-	campaign_timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+	campaign_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	campaign_article_id 				INTEGER 	NOT NULL,
+	campaign_name 					VARCHAR(255) 	NOT NULL,
+	campaign_votes 					INTEGER 	NOT NULL,
+	campaign_petition 				BOOL 		NOT NULL,
+	campaign_petition_signatures 			INTEGER 	NOT NULL,
+	campaign_deleted 				BOOL 		NOT NULL,
+	campaign_timestamp 				TIMESTAMP 	NOT NULL 	DEFAULT CURRENT_TIMESTAMP,
 
 	PRIMARY KEY(campaign_id)
@@ -717,6 +717,6 @@
 DROP TABLE IF EXISTS campaign_users;
 CREATE TABLE campaign_users (
-	campaign_user_campaign_id INTEGER NOT NULL,
-	campaign_user_user_entity_id INTEGER NOT NULL,
+	campaign_user_campaign_id 			INTEGER 	NOT NULL,
+	campaign_user_user_entity_id 			INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(campaign_user_campaign_id, campaign_user_user_entity_id)
@@ -725,12 +725,12 @@
 DROP TABLE IF EXISTS progress_reports;
 CREATE TABLE progress_reports (
-	progress_report_id INTEGER NOT NULL AUTO_INCREMENT,
-	progress_report_charity_id INTEGER NOT NULL,
-	progress_report_campaign_id INTEGER NOT NULL,
-	progress_report_text TEXT NOT NULL,
-	progress_report_order INTEGER NOT NULL,
-	progress_report_deleted BOOL NOT NULL,
-	progress_report_timestamp TIMESTAMP NOT NULL,
-	progress_report_good_bad ENUM('good', 'bad') NOT NULL,
+	progress_report_id 				INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	progress_report_charity_id 			INTEGER 	NOT NULL,
+	progress_report_campaign_id 			INTEGER 	NOT NULL,
+	progress_report_text 				TEXT 		NOT NULL,
+	progress_report_order 				INTEGER 	NOT NULL,
+	progress_report_deleted 			BOOL 		NOT NULL,
+	progress_report_timestamp 			TIMESTAMP	NOT NULL,
+	progress_report_good_bad 			ENUM('good', 'bad') NOT NULL,
 	
 	PRIMARY KEY(progress_report_id)
@@ -739,10 +739,10 @@
 DROP TABLE IF EXISTS charities;
 CREATE TABLE charities (
-	charity_id INTEGER NOT NULL AUTO_INCREMENT,
-	charity_name VARCHAR(255) NOT NULL,
-	charity_article_id INTEGER NOT NULL,
-	charity_goal_text TEXT NOT NULL,
-	charity_goal INTEGER NOT NULL,
-	charity_total FLOAT NOT NULL,
+	charity_id					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	charity_name 					VARCHAR(255) 	NOT NULL,
+	charity_article_id 				INTEGER 	NOT NULL,
+	charity_goal_text 				TEXT 		NOT NULL,
+	charity_goal 					INTEGER 	NOT NULL,
+	charity_total 					FLOAT 		NOT NULL,
 	
 	PRIMARY KEY(charity_id)
@@ -751,10 +751,10 @@
 DROP TABLE IF EXISTS charity_donors;
 CREATE TABLE charity_donors (
-	charity_donor_id INTEGER NOT NULL AUTO_INCREMENT,
-	charity_donor_charity_id INTEGER NOT NULL,
-	charity_donor_name VARCHAR(255) NOT NULL,
-	charity_donor_organisation_entity_id INTEGER NOT NULL,
-	charity_donor_amount INTEGER NOT NULL,
-	charity_donor_timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+	charity_donor_id 				INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	charity_donor_charity_id 			INTEGER 	NOT NULL,
+	charity_donor_name 				VARCHAR(255) 	NOT NULL,
+	charity_donor_organisation_entity_id 		INTEGER 	NOT NULL,
+	charity_donor_amount 				INTEGER 	NOT NULL,
+	charity_donor_timestamp 			TIMESTAMP 	NOT NULL 	DEFAULT CURRENT_TIMESTAMP,
 
 	PRIMARY KEY(charity_donor_id)
@@ -764,9 +764,9 @@
 DROP TABLE IF EXISTS tags;
 CREATE TABLE tags (
-	tag_id INTEGER NOT NULL AUTO_INCREMENT,
-	tag_name VARCHAR(255) NOT NULL,
-	tag_banner_name VARCHAR(255) NOT NULL,
-	tag_archive BOOL NOT NULL DEFAULT 0,
-	tag_deleted BOOL NOT NULL,
+	tag_id 						INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	tag_name 					VARCHAR(255) 	NOT NULL,
+	tag_banner_name 				VARCHAR(255) 	NOT NULL,
+	tag_archive 					BOOL 		NOT NULL 	DEFAULT 0,
+	tag_deleted 					BOOL 		NOT NULL,
 	
 	PRIMARY KEY(tag_id)
@@ -776,7 +776,7 @@
 DROP TABLE IF EXISTS colleges;
 CREATE TABLE colleges (
-	college_id INTEGER NOT NULL AUTO_INCREMENT,
-	college_name VARCHAR(255) NOT NULL,
-	college_ranking INTEGER NOT NULL,
+	college_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	college_name 					VARCHAR(255) 	NOT NULL,
+	college_ranking 				INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(college_id)
@@ -785,14 +785,14 @@
 DROP TABLE IF EXISTS college_rankings;
 CREATE TABLE college_rankings (
-	college_ranking_id INTEGER NOT NULL AUTO_INCREMENT,
-	college_ranking_college_id INTEGER NOT NULL,
+	college_ranking_id 				INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	college_ranking_college_id 			INTEGER 	NOT NULL,
 		-- Writers ID.  
-	college_ranking_user_entity_id INTEGER NOT NULL,
-	college_ranking_publisher_id INTEGER NOT NULL,
-	college_ranking_text TEXT NOT NULL,
-	college_ranking_rank INTEGER SIGNED NOT NULL,
-	college_ranking_published BOOL NOT NULL,
-	college_ranking_deleted BOOL NOT NULL,
-	college_ranking_timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+	college_ranking_user_entity_id			INTEGER 	NOT NULL,
+	college_ranking_publisher_id 			INTEGER 	NOT NULL,
+	college_ranking_text 				TEXT 		NOT NULL,
+	college_ranking_rank 				INTEGER SIGNED 	NOT NULL,
+	college_ranking_published 			BOOL 		NOT NULL,
+	college_ranking_deleted 			BOOL 		NOT NULL,
+	college_ranking_timestamp 			TIMESTAMP 	NOT NULL 	DEFAULT CURRENT_TIMESTAMP,
 
 	PRIMARY KEY(college_ranking_id)
@@ -803,15 +803,15 @@
 DROP TABLE IF EXISTS quiz_questions;
 CREATE TABLE quiz_questions (
-	quiz_question_id INTEGER NOT NULL AUTO_INCREMENT,
-	quiz_question_question TEXT NOT NULL,
-	quiz_question_answer1 TEXT NOT NULL,
-	quiz_question_answer2 TEXT NOT NULL,
-	quiz_question_answer3 TEXT NOT NULL,
-	quiz_question_answer4 TEXT NOT NULL,
-	quiz_question_answer INTEGER NOT NULL,
-	quiz_question_hits INTEGER NOT NULL,
+	quiz_question_id 				INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	quiz_question_question 				TEXT 		NOT NULL,
+	quiz_question_answer1 				TEXT 		NOT NULL,
+	quiz_question_answer2 				TEXT 		NOT NULL,
+	quiz_question_answer3 				TEXT 		NOT NULL,
+	quiz_question_answer4 				TEXT 		NOT NULL,
+	quiz_question_answer 				INTEGER 	NOT NULL,
+	quiz_question_hits 				INTEGER 	NOT NULL,
 		-- TODO: wtf
-	quiz_question_correct INTEGER NOT NULL,
-	quiz_question_active BOOL NOT NULL,
+	quiz_question_correct 				INTEGER 	NOT NULL,
+	quiz_question_active 				BOOL 		NOT NULL,
 
 	PRIMARY KEY(quiz_question_id)
@@ -820,7 +820,7 @@
 DROP TABLE IF EXISTS quiz_results;
 CREATE TABLE quiz_results (
-	quiz_result_user_entity_id INTEGER NOT NULL,
-	quiz_result_date TIMESTAMP NOT NULL,
-	quiz_result_score INTEGER NOT NULL,
+	quiz_result_user_entity_id 			INTEGER 	NOT NULL,
+	quiz_result_date 				TIMESTAMP 	NOT NULL,
+	quiz_result_score 				INTEGER 	NOT NULL,
 
 	PRIMARY KEY(quiz_result_user_entity_id)
@@ -829,7 +829,7 @@
 DROP TABLE IF EXISTS quiz_winners;
 CREATE TABLE quiz_winners (
-	quiz_winner_user_entity_id INTEGER NOT NULL,
-	quiz_winner_score INTEGER NOT NULL,
-	quiz_winner_date TIMESTAMP NOT NULL,
+	quiz_winner_user_entity_id 			INTEGER 	NOT NULL,
+	quiz_winner_score 				INTEGER 	NOT NULL,
+	quiz_winner_date 				TIMESTAMP 	NOT NULL,
 
 	PRIMARY KEY(quiz_winner_user_entity_id)
@@ -842,14 +842,14 @@
 DROP TABLE IF EXISTS comments;
 CREATE TABLE comments (
-	comment_id INTEGER NOT NULL AUTO_INCREMENT,
+	comment_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
 		-- The type of page eg. news, faq
-	comment_page_id INTEGER NOT NULL,
+	comment_page_id 				INTEGER 	NOT NULL,
 		-- The id of the content of the page e.g. article number
-	comment_subject_id INTEGER NOT NULL,
-	comment_text TEXT NOT NULL,
-	comment_rating INTEGER NOT NULL,
-	comment_reported_count INTEGER NOT NULL,
-	comment_deleted BOOL NOT NULL,
-	comment_timestamp TIMESTAMP NOT NULL,
+	comment_subject_id 				INTEGER 	NOT NULL,
+	comment_text 					TEXT 		NOT NULL,
+	comment_rating 					INTEGER 	NOT NULL,
+	comment_reported_count 				INTEGER 	NOT NULL,
+	comment_deleted 				BOOL 		NOT NULL,
+	comment_timestamp 				TIMESTAMP 	NOT NULL,
 
 	PRIMARY KEY(comment_id)
@@ -858,9 +858,9 @@
 DROP TABLE IF EXISTS pages;
 CREATE TABLE pages (
-	page_id INTEGER NOT NULL AUTO_INCREMENT,
-	page_title VARCHAR(255) NOT NULL,
-	page_comments BOOL NOT NULL,
-	page_ratings BOOL NOT NULL,
-	page_permission INTEGER NOT NULL,
+	page_id 					INTEGER 	NOT NULL AUTO_INCREMENT,
+	page_title 					VARCHAR(255) 	NOT NULL,
+	page_comments 					BOOL 		NOT NULL,
+	page_ratings 					BOOL 		NOT NULL,
+	page_permission 				INTEGER 	NOT NULL,
 
 	PRIMARY KEY(page_id)
@@ -869,12 +869,12 @@
 DROP TABLE IF EXISTS page_properties;
 CREATE TABLE page_properties (
-	page_property_id INTEGER NOT NULL AUTO_INCREMENT,
-	page_property_property_type_id INTEGER NOT NULL,
-	page_property_page_id INTEGER NOT NULL,
-	page_property_photo_id INTEGER NOT NULL,
-	page_property_image_id INTEGER NOT NULL,
-	page_property_label VARCHAR(255) NOT NULL,
-	page_property_text TEXT NOT NULL,
-	page_property_permission INTEGER NOT NULL,
+	page_property_id 				INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	page_property_property_type_id 			INTEGER 	NOT NULL,
+	page_property_page_id 				INTEGER 	NOT NULL,
+	page_property_photo_id 				INTEGER 	NOT NULL,
+	page_property_image_id 				INTEGER 	NOT NULL,
+	page_property_label 				VARCHAR(255) 	NOT NULL,
+	page_property_text 				TEXT 		NOT NULL,
+	page_property_permission 			INTEGER 	NOT NULL,
 
 	PRIMARY KEY(page_property_id)
@@ -883,8 +883,8 @@
 DROP TABLE IF EXISTS images;
 CREATE TABLE images (
-	image_id INTEGER NOT NULL AUTO_INCREMENT,
-	image_title VARCHAR(255) NOT NULL,
-	image_image_type_id INTEGER NOT NULL,
-	image_file_extension CHAR(4) NOT NULL,
+	image_id 					INTEGER 	NOT NULL	AUTO_INCREMENT,
+	image_title 					VARCHAR(255) 	NOT NULL,
+	image_image_type_id 				INTEGER 	NOT NULL,
+	image_file_extension 				CHAR(4) 	NOT NULL,
 
 	PRIMARY KEY(image_id)
@@ -893,10 +893,10 @@
 DROP TABLE IF EXISTS image_types;
 CREATE TABLE image_types (
-	image_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	image_type_name VARCHAR(255) NOT NULL,
-	image_type_width INTEGER NOT NULL,
-	image_type_height INTEGER NOT NULL,
+	image_type_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	image_type_name 				VARCHAR(255) 	NOT NULL,
+	image_type_width 				INTEGER 	NOT NULL,
+	image_type_height 				INTEGER 	NOT NULL,
 		-- True if all photos must have a thumb of this size
-	image_type_photo BOOL NOT NULL,
+	image_type_photo 				BOOL 		NOT NULL,
 
 	PRIMARY KEY(image_type_id)
@@ -905,13 +905,13 @@
 DROP TABLE IF EXISTS photos;
 CREATE TABLE photos (
-	photo_id INTEGER NOT NULL AUTO_INCREMENT,
-	photo_timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
-	photo_user_entity_id INTEGER NOT NULL,
-	photo_title VARCHAR(255) NOT NULL,
-	photo_width INTEGER NOT NULL,
-	photo_height INTEGER NOT NULL,
-	photo_gallery BOOL NOT NULL,
-	photo_homepage TIMESTAMP NOT NULL,
-	photo_deleted BOOL NOT NULL,
+	photo_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	photo_timestamp 				TIMESTAMP 	NOT NULL 	DEFAULT CURRENT_TIMESTAMP,
+	photo_user_entity_id 				INTEGER 	NOT NULL,
+	photo_title 					VARCHAR(255) 	NOT NULL,
+	photo_width 					INTEGER 	NOT NULL,
+	photo_height 					INTEGER 	NOT NULL,
+	photo_gallery 					BOOL 		NOT NULL,
+	photo_homepage 					TIMESTAMP 	NOT NULL,
+	photo_deleted 					BOOL 		NOT NULL,
 
 	PRIMARY KEY(photo_id)
@@ -920,6 +920,6 @@
 DROP TABLE IF EXISTS photo_tags;
 CREATE TABLE photo_tags (
-	photo_tag_photo_id INTEGER NOT NULL,
-	photo_tag_tag_id INTEGER NOT NULL,
+	photo_tag_photo_id 				INTEGER 	NOT NULL,
+	photo_tag_tag_id 				INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(photo_tag_photo_id, photo_tag_tag_id)
@@ -931,12 +931,12 @@
 DROP TABLE IF EXISTS adverts;
 CREATE TABLE adverts (
-	advert_id INTEGER NOT NULL AUTO_INCREMENT,
-	advert_organisation_entity_id INTEGER NOT NULL,
-	advert_name VARCHAR(255) NOT NULL,
-	advert_description TEXT NOT NULL,
-	advert_url VARCHAR(255) NOT NULL,
-	advert_start_date TIMESTAMP NOT NULL,
-	advert_end_date TIMESTAMP NOT NULL,
-	advert_max_total INTEGER NOT NULL,
+	advert_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	advert_organisation_entity_id 			INTEGER 	NOT NULL,
+	advert_name 					VARCHAR(255) 	NOT NULL,
+	advert_description 				TEXT 		NOT NULL,
+	advert_url 					VARCHAR(255) 	NOT NULL,
+	advert_start_date 				TIMESTAMP 	NOT NULL,
+	advert_end_date 				TIMESTAMP 	NOT NULL,
+	advert_max_total 				INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(advert_id)
@@ -945,9 +945,9 @@
 DROP TABLE IF EXISTS advert_bills;
 CREATE TABLE advert_bills (
-	advert_bill_id INTEGER NOT NULL AUTO_INCREMENT,
-	advert_bill_organisation_entity_id INTEGER NOT NULL,
-	advert_bill_total INTEGER NOT NULL,
-	advert_bill_date TIMESTAMP NOT NULL,
-	advert_bill_paid BOOL NOT NULL,
+	advert_bill_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	advert_bill_organisation_entity_id 		INTEGER 	NOT NULL,
+	advert_bill_total 				INTEGER 	NOT NULL,
+	advert_bill_date 				TIMESTAMP 	NOT NULL,
+	advert_bill_paid 				BOOL 		NOT NULL,
 	
 	PRIMARY KEY(advert_bill_id)
@@ -956,11 +956,11 @@
 DROP TABLE IF EXISTS advert_bill_items;
 CREATE TABLE advert_bill_items (
-	advert_bill_item_id INTEGER NOT NULL AUTO_INCREMENT,
-	advert_bill_item_advert_instance_id INTEGER NOT NULL,
-	advert_bill_item_advert_bill_id INTEGER NOT NULL,
-	advert_bill_item_amount INTEGER NOT NULL,
-	advert_bill_item_clicks INTEGER NOT NULL,
-	advert_bill_item_views INTEGER NOT NULL,
-	advert_bill_item_made_date INTEGER NOT NULL,
+	advert_bill_item_id 				INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	advert_bill_item_advert_instance_id 		INTEGER 	NOT NULL,
+	advert_bill_item_advert_bill_id 		INTEGER 	NOT NULL,
+	advert_bill_item_amount 			INTEGER 	NOT NULL,
+	advert_bill_item_clicks 			INTEGER 	NOT NULL,
+	advert_bill_item_views 				INTEGER 	NOT NULL,
+	advert_bill_item_made_date 			INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(advert_bill_item_id)
@@ -969,13 +969,13 @@
 DROP TABLE IF EXISTS advert_instances;
 CREATE TABLE advert_instances (
-	advert_instance_id INTEGER NOT NULL AUTO_INCREMENT,
-	advert_instance_space_type_id INTEGER NOT NULL,
-	advert_instance_advert_id INTEGER NOT NULL,
-	advert_instance_views INTEGER NOT NULL,
-	advert_instance_clicks INTEGER NOT NULL,
-	advert_instance_view_cost INTEGER NOT NULL,
-	advert_instance_click_cost INTEGER NOT NULL,
-	advert_instance_extension CHAR(4) NOT NULL,
-	advert_instance_deleted BOOL NOT NULL,
+	advert_instance_id 				INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	advert_instance_space_type_id 			INTEGER 	NOT NULL,
+	advert_instance_advert_id 			INTEGER 	NOT NULL,
+	advert_instance_views 				INTEGER 	NOT NULL,
+	advert_instance_clicks 				INTEGER 	NOT NULL,
+	advert_instance_view_cost 			INTEGER 	NOT NULL,
+	advert_instance_click_cost 			INTEGER 	NOT NULL,
+	advert_instance_extension 			CHAR(4) 	NOT NULL,
+	advert_instance_deleted 			BOOL 		NOT NULL,
 
 	PRIMARY KEY(advert_instance_id)
@@ -984,6 +984,6 @@
 DROP TABLE IF EXISTS advert_related_articles;
 CREATE TABLE advert_related_articles (
-	advert_related_article_advert_id INTEGER NOT NULL,
-	advert_related_article_article_id INTEGER NOT NULL,
+	advert_related_article_advert_id 		INTEGER 	NOT NULL,
+	advert_related_article_article_id 		INTEGER 	NOT NULL,
 
 	PRIMARY KEY(advert_related_article_advert_id, advert_related_article_article_id)
@@ -992,6 +992,6 @@
 DROP TABLE IF EXISTS advert_related_organisations;
 CREATE TABLE advert_related_organisations (
-	advert_related_organisation_advert_id INTEGER NOT NULL,
-	advert_related_organisation_organisation_entity_id INTEGER NOT NULL,
+	advert_related_organisation_advert_id 		INTEGER 	NOT NULL,
+	advert_related_organisation_organisation_entity_id INTEGER 	NOT NULL,
 
 	PRIMARY KEY(advert_related_organisation_advert_id, advert_related_organisation_organisation_entity_id)
@@ -1000,7 +1000,7 @@
 DROP TABLE IF EXISTS page_space_types;
 CREATE TABLE page_space_types (
-	page_space_type_page_id INTEGER NOT NULL,
-	page_space_type_space_type_id INTEGER NOT NULL,
-	page_space_type_number INTEGER NOT NULL,
+	page_space_type_page_id 			INTEGER 	NOT NULL,
+	page_space_type_space_type_id 			INTEGER 	NOT NULL,
+	page_space_type_number 				INTEGER 	NOT NULL,
 
 	PRIMARY KEY(page_space_type_page_id, page_space_type_space_type_id, page_space_type_number)
@@ -1009,9 +1009,9 @@
 DROP TABLE IF EXISTS space_types;
 CREATE TABLE space_types (
-	space_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	space_type_view_cost FLOAT NOT NULL,
-	space_type_click_cost FLOAT NOT NULL,
-	space_type_width INTEGER NOT NULL,
-	space_type_height INTEGER NOT NULL,
+	space_type_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	space_type_view_cost 				FLOAT 		NOT NULL,
+	space_type_click_cost 				FLOAT 		NOT NULL,
+	space_type_width 				INTEGER 	NOT NULL,
+	space_type_height 				INTEGER 	NOT NULL,
 
 	PRIMARY KEY(space_type_id)
@@ -1025,9 +1025,9 @@
 DROP TABLE IF EXISTS buildings;
 CREATE TABLE buildings (
-	building_id INTEGER NOT NULL AUTO_INCREMENT,
-	buidling_name VARCHAR(255) NOT NULL,
-	building_code CHAR(15) NOT NULL,
-	building_x INTEGER NOT NULL,
-	buidling_y INTEGER NOT NULL,
+	building_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	buidling_name 					VARCHAR(255) 	NOT NULL,
+	building_code 					CHAR(15) 	NOT NULL,
+	building_x 					INTEGER 	NOT NULL,
+	buidling_y 					INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(building_id)
@@ -1036,7 +1036,7 @@
 DROP TABLE IF EXISTS rooms;
 CREATE TABLE rooms (
-	room_id CHAR(15) NOT NULL,
-	room_room_building_id INTEGER NOT NULL,
-	room_type_id INTEGER NOT NULL,
+	room_id 					CHAR(15) 	NOT NULL,
+	room_room_building_id 				INTEGER 	NOT NULL,
+	room_type_id 					INTEGER 	NOT NULL,
 	
 	PRIMARY KEY(room_id)
@@ -1045,6 +1045,6 @@
 DROP TABLE IF EXISTS room_types;
 CREATE TABLE room_types (
-	room_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	room_type_name VARCHAR(255) NOT NULL,
+	room_type_id 					INTEGER 	NOT NULL 	AUTO_INCREMENT,
+	room_type_name 					VARCHAR(255) 	NOT NULL,
 	
 	PRIMARY KEY(room_type_id)
