Index: design/crosswords/crossword_uri_structure.dia
===================================================================
--- design/crosswords/crossword_uri_structure.dia (revision 8eb2ef402eaa466ad98a3c76cb1f078ee66adfb1)
+++ design/crosswords/crossword_uri_structure.dia (revision 85f056503972a2467ec4a9e86935252c419f1f05)
@@ -151,5 +151,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="-40.5987,-5.2175"/>
+            <dia:point val="-40.5988,-5.2175"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -346,5 +346,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="19.8037,23.8825"/>
+            <dia:point val="19.8038,23.8825"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -510,5 +510,5 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="-32.1225,5.94;-21.235,9.34"/>
+        <dia:rectangle val="-32.1225,5.94;-20.645,9.34"/>
       </dia:attribute>
       <dia:attribute name="elem_corner">
@@ -516,5 +516,5 @@
       </dia:attribute>
       <dia:attribute name="elem_width">
-        <dia:real val="10.887499999999999"/>
+        <dia:real val="11.477500000000001"/>
       </dia:attribute>
       <dia:attribute name="elem_height">
@@ -531,5 +531,5 @@
           <dia:attribute name="string">
             <dia:string>#/office/crosswords/cats
-[CROSSWORD_CATEGORY_INDEX]
+[CROSSWORD_CATEGORIES_INDEX]
 shows categories information#</dia:string>
           </dia:attribute>
@@ -541,5 +541,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="-26.6787,6.9825"/>
+            <dia:point val="-26.3838,6.9825"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -619,5 +619,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="-11.3338,7.0975"/>
+            <dia:point val="-11.3337,7.0975"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -632,12 +632,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O14">
       <dia:attribute name="obj_pos">
-        <dia:point val="-21.235,7.64"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="-21.2864,7.255;-16.5725,8.255"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="-21.235,7.64"/>
-        <dia:point val="-19.4375,7.59"/>
+        <dia:point val="-20.645,7.64"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="-20.6964,7.255;-16.5725,8.255"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="-20.645,7.64"/>
+        <dia:point val="-18.8475,7.59"/>
         <dia:point val="-18.0169,7.755"/>
         <dia:point val="-16.6225,7.755"/>
@@ -741,14 +741,14 @@
     <dia:object type="UML - Activity" version="0" id="O17">
       <dia:attribute name="obj_pos">
-        <dia:point val="5.9275,-2.945"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="5.9275,-2.945;15.0575,3.655"/>
+        <dia:point val="4.65471,-3.29855"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="4.65471,-3.29855;16.9322,3.30145"/>
       </dia:attribute>
       <dia:attribute name="elem_corner">
-        <dia:point val="5.9275,-2.945"/>
+        <dia:point val="4.65471,-3.29855"/>
       </dia:attribute>
       <dia:attribute name="elem_width">
-        <dia:real val="9.129999999999999"/>
+        <dia:real val="12.2775"/>
       </dia:attribute>
       <dia:attribute name="elem_height">
@@ -764,5 +764,5 @@
         <dia:composite type="text">
           <dia:attribute name="string">
-            <dia:string>#/office/crosswords/$xwordid
+            <dia:string>#/office/crosswords/crossword/$xwordid
 [CROSSWORD_VIEW]
 view crossword information
@@ -779,5 +779,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="10.4925,-1.9025"/>
+            <dia:point val="10.7935,-2.25605"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -844,5 +844,5 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="27.0451,-7.345;37.4926,-2.345"/>
+        <dia:rectangle val="27.0451,-7.345;40.6401,-2.345"/>
       </dia:attribute>
       <dia:attribute name="elem_corner">
@@ -850,5 +850,5 @@
       </dia:attribute>
       <dia:attribute name="elem_width">
-        <dia:real val="10.4475"/>
+        <dia:real val="13.595000000000001"/>
       </dia:attribute>
       <dia:attribute name="elem_height">
@@ -864,5 +864,5 @@
         <dia:composite type="text">
           <dia:attribute name="string">
-            <dia:string>#/office/crosswords/$xwordid/edit
+            <dia:string>#/office/crosswords/crossword/$xwordid/edit
 [CROSSWORD_MODIFY]
 [CROSSWORD_AUTHOR_ADD]
@@ -877,5 +877,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="32.2689,-6.3025"/>
+            <dia:point val="33.8426,-6.3025"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -890,12 +890,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O20">
       <dia:attribute name="obj_pos">
-        <dia:point val="15.0575,0.355"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="14.999,0.29648;38.3799,13.0311"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="15.0575,0.355"/>
-        <dia:point val="16.1493,6.05"/>
+        <dia:point val="16.9322,0.00144661"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="16.8737,-0.0570733;38.38,13.0311"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="16.9322,0.00144661"/>
+        <dia:point val="18.024,5.69645"/>
         <dia:point val="38.0592,6.34"/>
         <dia:point val="37.8788,12.9675"/>
@@ -924,11 +924,11 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="-6.50054,3.5219;10.9938,8.0531"/>
+        <dia:rectangle val="-6.50054,3.16835;11.2948,8.0531"/>
       </dia:attribute>
       <dia:attribute name="bez_points">
         <dia:point val="-6.045,7.755"/>
         <dia:point val="-2.87572,2.4125"/>
-        <dia:point val="9.94928,6.85"/>
-        <dia:point val="10.4925,3.655"/>
+        <dia:point val="10.2502,6.49645"/>
+        <dia:point val="10.7935,3.30145"/>
       </dia:attribute>
       <dia:attribute name="corner_types">
@@ -961,12 +961,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O22">
       <dia:attribute name="obj_pos">
-        <dia:point val="15.0575,0.355"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="14.6437,-5.34547;27.1001,0.716361"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="15.0575,0.355"/>
-        <dia:point val="17.3493,-2.35"/>
+        <dia:point val="16.9322,0.00144661"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="16.5184,-5.34547;27.1001,0.362809"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="16.9322,0.00144661"/>
+        <dia:point val="19.224,-2.70355"/>
         <dia:point val="22.5117,-4.8"/>
         <dia:point val="27.0451,-4.845"/>
@@ -1004,5 +1004,5 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="27.1267,-0.995;37.8817,3.205"/>
+        <dia:rectangle val="27.1267,-0.995;41.0292,3.205"/>
       </dia:attribute>
       <dia:attribute name="elem_corner">
@@ -1010,5 +1010,5 @@
       </dia:attribute>
       <dia:attribute name="elem_width">
-        <dia:real val="10.754999999999999"/>
+        <dia:real val="13.9025"/>
       </dia:attribute>
       <dia:attribute name="elem_height">
@@ -1024,5 +1024,5 @@
         <dia:composite type="text">
           <dia:attribute name="string">
-            <dia:string>#/office/crosswords/$xwordid/stats
+            <dia:string>#/office/crosswords/crossword/$xwordid/stats
 [CROSSWORD_STATS_BASIC]
 [CROSSWORD_STATS_DETAILED]
@@ -1036,5 +1036,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="32.5042,0.0475"/>
+            <dia:point val="34.078,0.0475"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -1049,12 +1049,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O24">
       <dia:attribute name="obj_pos">
-        <dia:point val="15.0575,0.355"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="14.8408,-0.132289;27.217,1.60741"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="15.0575,0.355"/>
-        <dia:point val="17.3993,1.2"/>
+        <dia:point val="16.9322,0.00144661"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="16.7155,-0.485843;27.217,1.60741"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="16.9322,0.00144661"/>
+        <dia:point val="19.274,0.846447"/>
         <dia:point val="22.2617,1.5"/>
         <dia:point val="27.1267,1.105"/>
@@ -1092,5 +1092,5 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="-30.3733,-12.9075;-21.7258,-9.5075"/>
+        <dia:rectangle val="-30.3733,-12.9075;-21.3558,-9.5075"/>
       </dia:attribute>
       <dia:attribute name="elem_corner">
@@ -1098,5 +1098,5 @@
       </dia:attribute>
       <dia:attribute name="elem_width">
-        <dia:real val="8.6474999999999991"/>
+        <dia:real val="9.0175000000000001"/>
       </dia:attribute>
       <dia:attribute name="elem_height">
@@ -1113,5 +1113,5 @@
           <dia:attribute name="string">
             <dia:string>#/office/crosswords/tips
-[CROSSWORD_TIP_INDEX]
+[CROSSWORD_TIPS_INDEX]
 list tip categories#</dia:string>
           </dia:attribute>
@@ -1123,5 +1123,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="-26.0496,-11.865"/>
+            <dia:point val="-25.8646,-11.865"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -1170,5 +1170,5 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="-16.3233,-15.12;-3.85832,-6.92"/>
+        <dia:rectangle val="-16.3233,-15.12;-3.8133,-6.92"/>
       </dia:attribute>
       <dia:attribute name="elem_corner">
@@ -1176,5 +1176,5 @@
       </dia:attribute>
       <dia:attribute name="elem_width">
-        <dia:real val="12.465"/>
+        <dia:real val="12.51"/>
       </dia:attribute>
       <dia:attribute name="elem_height">
@@ -1194,5 +1194,5 @@
 [CROSSWORD_TIP_CATEGORY_INDEX]
 [CROSSWORD_TIP_CATEGORY_ADD]
-[CROSSWORD_TIP_CATEGORY_EDIT]
+[CROSSWORD_TIP_CATEGORY_MODIFY]
 [CROSSWORD_TIP_CATEGORY_DELETE]
 [CROSSWORD_TIP_DELETE]
@@ -1207,5 +1207,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="-10.0908,-14.0775"/>
+            <dia:point val="-10.0683,-14.0775"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -1270,12 +1270,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O29">
       <dia:attribute name="obj_pos">
-        <dia:point val="-21.7258,-11.2075"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="-21.7772,-11.52;-16.2733,-10.52"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="-21.7258,-11.2075"/>
-        <dia:point val="-19.9283,-11.2575"/>
+        <dia:point val="-21.3558,-11.2075"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="-21.4072,-11.52;-16.2733,-10.52"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="-21.3558,-11.2075"/>
+        <dia:point val="-19.5583,-11.2575"/>
         <dia:point val="-17.7177,-11.02"/>
         <dia:point val="-16.3233,-11.02"/>
@@ -1301,12 +1301,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O30">
       <dia:attribute name="obj_pos">
-        <dia:point val="-3.85832,-11.02"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="-3.9222,-11.5212;3.77668,-10.37"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="-3.85832,-11.02"/>
-        <dia:point val="-2.06082,-11.07"/>
+        <dia:point val="-3.8133,-11.02"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="-3.87718,-11.5212;3.77668,-10.37"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="-3.8133,-11.02"/>
+        <dia:point val="-2.0158,-11.07"/>
         <dia:point val="2.33228,-10.87"/>
         <dia:point val="3.72668,-10.87"/>
@@ -1344,11 +1344,11 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="9.99062,-8.04425;11.3048,-2.86999"/>
+        <dia:rectangle val="10.2916,-8.04424;11.3047,-3.22355"/>
       </dia:attribute>
       <dia:attribute name="bez_points">
         <dia:point val="10.8029,-7.97"/>
         <dia:point val="10.6243,-4.3"/>
-        <dia:point val="10.3743,-5.3"/>
-        <dia:point val="10.4925,-2.945"/>
+        <dia:point val="10.6753,-5.65355"/>
+        <dia:point val="10.7935,-3.29855"/>
       </dia:attribute>
       <dia:attribute name="corner_types">
@@ -1431,5 +1431,5 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="-16.5607,-4.72;-5.06322,-0.52"/>
+        <dia:rectangle val="-16.5607,-4.72;-5.0632,-0.52"/>
       </dia:attribute>
       <dia:attribute name="elem_corner">
@@ -1538,12 +1538,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O36">
       <dia:attribute name="obj_pos">
-        <dia:point val="5.9275,0.355"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="-5.12293,-3.12088;5.98093,0.408429"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="5.9275,0.355"/>
-        <dia:point val="1.64928,0.05"/>
+        <dia:point val="4.65471,0.00144661"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="-5.12293,-3.12088;4.70814,0.0548756"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="4.65471,0.00144661"/>
+        <dia:point val="0.376488,-0.303553"/>
         <dia:point val="-1.46322,-2.55"/>
         <dia:point val="-5.06322,-2.62"/>
@@ -1643,5 +1643,5 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="-21.2332,15.1525;-15.9957,17.625"/>
+        <dia:rectangle val="-21.2332,15.28;-16.0382,17.625"/>
       </dia:attribute>
       <dia:attribute name="text">
@@ -1657,5 +1657,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="-16.0382,16.55"/>
+            <dia:point val="-16.0382,16.635"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -1676,5 +1676,5 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="-15.0382,18.7025;-9.78322,21.175"/>
+        <dia:rectangle val="-15.0382,18.83;-9.8257,21.175"/>
       </dia:attribute>
       <dia:attribute name="text">
@@ -1690,5 +1690,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="-15.0382,20.1"/>
+            <dia:point val="-15.0382,20.185"/>
           </dia:attribute>
           <dia:attribute name="color">
