Index: system/application/views/homepages/roses.php
===================================================================
--- system/application/views/homepages/roses.php (revision ce7a920f029ed645529b4f75f35cc520a1c8072b)
+++ system/application/views/homepages/roses.php (revision ab9d8b7bcfd976ba7b7dae73ee9130c7165896f7)
@@ -25,4 +25,7 @@
 	}
 	?>
+	<div style="text-align:center">
+		<a href="/news/<?php echo($liveblog[0]['id']); ?>">Read more...</a>
+	</div>
 </div>
 
