Index: system/application/controllers/office/liveblog.php
===================================================================
--- system/application/controllers/office/liveblog.php (revision d779b0dcf81057940f9a1145c7d0191eaed57683)
+++ system/application/controllers/office/liveblog.php (revision dcf42327b6cf233d4aa7327fb240ff6a09539a53)
@@ -608,4 +608,8 @@
 			//$revision = $this->article_model->CreateNewRevision($article_id, $user_id, $rev_data['headline'], $rev_data['subheadline'], $rev_data['subtext'], $rev_data['blurb'], $content['all']['wikitext'], $content['all']['cache']);
 			$content['all']['wikitext'] = '!!! DO NOT EDIT THIS ARTICLE, THIS IS A LIVE BLOG, SPEAK TO webmaster@theyorker.co.uk INSTEAD !!!';
+
+			$score = $this->_getScore();
+			$rev_data['subtext'] = "YORK " . $score['york'] . " - LANCASTER " . $score['lancs'];
+
 			$revision = $this->article_model->CreateNewRevision($article_id, $user_id, $rev_data['headline'], $rev_data['subheadline'], $rev_data['subtext'], $rev_data['blurb'], $content['all']['wikitext'], $content['all']['cache']);
 			$publish_date = $this->roses_model->getPublishDate($article_id);
