Index: system/application/views/general/home.php
===================================================================
--- system/application/views/general/home.php (revision 337b54f68ea5ca9905cea626bbc7c5ae51c14245)
+++ system/application/views/general/home.php (revision 578d276d4b27ec9ea81efd093d95b0721e047792)
@@ -46,5 +46,9 @@
 	</div>
 </div>
-<div class="FlexiBox Box13b FlexiBoxLast"><br /><br /><br /><br /></div>
+<div class="FlexiBox Box13 FlexiBoxLast">
+	<a href="http://www.facebook.com/event.php?eid=111317892288">
+		<img src="/images/adverts/BrownsAdSm.jpg" alt="Vote Browns for British Sandwich Week" />
+	</a>
+</div>
 <div class="clear"></div>
 
