Index: system/application/controllers/roses.php
===================================================================
--- system/application/controllers/roses.php (revision 1b9b919ad8c49533c61400905c4aa561e682db5c)
+++ system/application/controllers/roses.php (revision f339d6bb48a63077214a0a0017cf0735f3580e31)
@@ -28,4 +28,22 @@
 		$data['events'] = $this->db->query($sql)->result_array();
 
+		$score_york = 0;
+		$score_lancs = 0;
+
+		foreach ($data['events'] as $events) {
+			if (empty($events['event_score_time'])) continue;
+			if ($events['event_york_score'] > $events['event_lancaster_score']) {
+				$score_york += $events['event_points'];
+			} else if ($events['event_york_score'] < $events['event_lancaster_score']) {
+				$score_lancs += $events['event_points'];
+			} else {
+				$score_york += $events['event_points'] / 2;
+				$score_lancs += $events['event_points'] / 2;
+			}
+		}
+
+		$data['score_york'] = $score_york;
+		$data['score_lancs'] = $score_lancs;
+
 		$this->pages_model->SetPageCode('homepage_roses');
 		$this->main_frame->SetData('menu_tab', 'roses');
Index: system/application/views/homepages/roses.php
===================================================================
--- system/application/views/homepages/roses.php (revision 882b087a2c4fa7f8b0e5a143077fa946f8d04f04)
+++ system/application/views/homepages/roses.php (revision f339d6bb48a63077214a0a0017cf0735f3580e31)
@@ -1,6 +1,11 @@
 <div class="FlexiBox Box23">
 	<div id="DisplayBox">
-		<div id="DisplayBoxBg"><?php echo(xml_escape($liveblog[0]['headline'])); ?></div>
-		<div id="DisplayBoxText"><a href="/news/<?php echo($liveblog[0]['id']); ?>"><?php echo(xml_escape($liveblog[0]['headline'])); ?></a></div>
+		<div style="background-color:#999;color:#999;font-weight:bold;font-size:30px;position:absolute;bottom:0;left:0;opacity:0.6;">York <?php echo(xml_escape($score_york)); ?></div>
+		<div style="color:#fff;font-weight:bold;font-size:30px;position:absolute;bottom:0;left:0;">York <?php echo(xml_escape($score_york)); ?></div>
+		<div style="background-color:#999;color:#999;font-weight:bold;font-size:30px;position:absolute;bottom:0;right:0;opacity:0.6;">Lancaster <?php echo(xml_escape($score_lancs)); ?></div>
+		<div style="color:#ba0000;font-weight:bold;font-size:30px;position:absolute;bottom:0;right:0;">Lancaster <?php echo(xml_escape($score_lancs)); ?></div>
+
+		<div id="DisplayBoxBg" style="top:0;bottom:auto;"><?php echo(xml_escape($liveblog[0]['headline'])); ?></div>
+		<div id="DisplayBoxText" style="top:0;bottom:auto;"><a href="/news/<?php echo($liveblog[0]['id']); ?>"><?php echo(xml_escape($liveblog[0]['headline'])); ?></a></div>
 		<a href="/news/<?php echo($liveblog[0]['id']); ?>"><img src="/photos/home/<?php echo($liveblog[0]['photo_id']); ?>" alt="<?php echo(xml_escape($liveblog[0]['photo_title'])); ?>" /></a>
 	</div>
