Index: export/template/public_html/.htaccess
===================================================================
--- export/template/public_html/.htaccess (revision 06a334b45812dc7ea2abf7318afb620bfb251a5b)
+++ export/template/public_html/.htaccess (revision 5d7f29e50e7776666e04f80762aed6c5e40ceaf7)
@@ -3,4 +3,14 @@
 RewriteEngine on
 RewriteBase /
+
+# Redirect http://theyorker.co.uk/* to http://www.theyorker.co.uk/*
+RewriteCond %{SERVER_NAME} ^theyorker\.co\.uk$
+RewriteCond %{SERVER_PORT} !^443$
+RewriteRule ^(.*)$ http://www.theyorker.co.uk/$1 [L,R]
+
+# Redirect https://theyorker.co.uk/* to https://www.theyorker.co.uk/*
+RewriteCond %{SERVER_NAME} ^theyorker\.co\.uk$
+RewriteCond %{SERVER_PORT} ^443$
+RewriteRule ^(.*)$ https://www.theyorker.co.uk/$1 [L,R]
 
 # Redirect secure pages to https
