Index: system/application/models/notifications_model.php
===================================================================
--- system/application/models/notifications_model.php (revision c4214e394cf80fe31abb2444d072698355014e66)
+++ system/application/models/notifications_model.php (revision ae7ff387c6801a2e373cbd78bdf15b72bb77a1dc)
@@ -188,5 +188,7 @@
 				notification_user_entity_id = ?';
 		foreach ($users as $user) {
-			$this->db->query($sql, array($new_id, $user));
+			if (!empty($user)) {
+				$this->db->query($sql, array($new_id, $user));
+			}
 		}
 	}
