Index: system/application/controllers/food.php
===================================================================
--- system/application/controllers/food.php (revision e1fd2f2a01473d44ceba09feb55781fcdf9ce5cc)
+++ system/application/controllers/food.php (revision e9121dea03fda64dc58f9db18a0779b508fb9ffb)
@@ -1,27 +1,10 @@
 <?php
-
-/**
- *	Food section home page
- *	@author	Owen Jones
- */
 
 class Food extends Controller
 {
-
 	function __construct()
 	{
 		parent::Controller();
-		$this->load->model('News_model');
-		$this->load->model('Home_Model');
-		$this->load->library('Homepage_boxes');
-	}
-	
-	private function getNumberOfType ($articles, $type_codename)
-	{
-		$count = 0;
-		foreach ($articles as $article) {
-			if ($article['article_type'] == $type_codename) $count++;
-		}
-		return $count;
+		$this->load->model('home_hack_model');
 	}
 	
@@ -29,84 +12,38 @@
 	{
 		if (!CheckPermissions('public')) return;
+
+		$page_codename = 'food';
+
+		$this->pages_model->SetPageCode('homepage_' . $page_codename);
+
+		// Obtain banner for homepage
+		//$data['banner'] = $this->Home_Model->GetBannerImageForHomepage($homepage_article_type);
+
+		$data = array();
+		$data['spotlight'] = $this->home_hack_model->getArticlesByTags(array('food', 'spotlight'), 1);
+		$data['features'] = array(
+			'title'		=>	'features',
+			'articles'	=>	$this->home_hack_model->getArticlesByTags(array('food', 'feature'), 9)
+		);
+		$data['sections'] = array(
+			array(
+				'title'		=>	'the broke student',
+				'articles'	=>	$this->home_hack_model->getArticlesByTags(array('the-broke-student'))
+			),
+                        array(
+                                'title'         =>      'weird and wonderful',
+                                'articles'      =>      $this->home_hack_model->
+getArticlesByTags(array('weird-and-wonderful'))
+                        ),
+			array(
+				'title'		=>	'the taste test',
+				'articles'	=>	$this->home_hack_model->getArticlesByTags(array('the-taste-test'))
+			)
+		);
 		
-		$homepage_article_type = 'food';
-		//Get page properties information
-		$this->pages_model->SetPageCode('homepage_'.$homepage_article_type);
-		$data['latest_heading'] = $this->pages_model->GetPropertyText('latest_heading');
-		$data['more_heading'] = $this->pages_model->GetPropertyText('more_heading');
-		$data['links_heading'] = $this->pages_model->GetPropertyText('links_heading');
-		$data['featured_puffer_title'] = $this->pages_model->GetPropertyText('featured_puffer_title',TRUE);
-		
-		$main_articles_num = (int)$this->pages_model->GetPropertyText('max_num_main_articles');//Max number of main articles to show
-		$more_articles_num = (int)$this->pages_model->GetPropertyText('max_num_more_articles');//Max number of more articles to show
-		
-		//Obtain banner for homepage
-		$data['banner'] = $this->Home_Model->GetBannerImageForHomepage($homepage_article_type);
-		
-		//////////////Information for main article(s)
-		//Get article ids for the main section
-		$main_article_ids = $this->News_model->GetLatestId($homepage_article_type,$main_articles_num);
-		//First article has summery, rest are simple articles
-		if(empty($main_article_ids)){
-			//TODO better error page for no results!
-			$main_article_summarys = array();
-		}else{
-			$main_article_summarys[0] = $this->News_model->GetSummaryArticle($main_article_ids[0], "Left", '%W, %D %M %Y', "medium");
-			for ($index = 1; $index <= ($main_articles_num-1) && $index < count($main_article_ids); $index++) {
-				array_push($main_article_summarys, $this->News_model->GetSimpleArticle($main_article_ids[$index], "Left"));
-			}
-		}
 
-		
-		//////////////Information for more article list(s)
-		//Get list of article types
-		$more_article_types = $this->News_model->getSubArticleTypes($homepage_article_type);
-		//////For each article type get list of simple articles to the limit of $more_articles_num
-		$article_index = 0;
-		$articles_summarys = array();
-		foreach ($more_article_types as $an_article){
-			//Get article id's for that article type up to limit of $more_articles_num
-			//with an offset of $offset to prevent picking up duplicate articles.
-			$offset = $this->getNumberOfType($main_article_summarys,$an_article['codename']);
-			$articles_ids[$article_index] = $this->News_model->GetLatestId($an_article['codename'],$more_articles_num,$offset);
-				//for the new article type found get a simple article for each of the ids found.
-				for ($index = 0; $index <= ($more_articles_num-1) && $index < count($articles_ids[$article_index]); $index++) {
-					//no need to check that the ids havent been used, due to the offset.
-					$articles_summarys[$article_index][] = $this->News_model->GetSimpleArticle($articles_ids[$article_index][$index], "Left");
-				}
-			$article_index++;
-		}
-		
-		/////////////Get information for side puffers
-		//use article types already found by more articles
-		$data['puffers'] = array();
-		$index = 0;
-		foreach ($more_article_types as $puffer) {
-			$data['puffers'][$index] = $puffer;
-			$data['puffers'][$index]['image'] = '/image/'.$puffer['image_codename'].'/'.$puffer['image'];
-			$index++;
-		}
-		
-		//////////////Information for special/featured puffer
-		//Get article ID
-		$featured_puffer_id = $this->News_model->GetLatestFeaturedId($homepage_article_type);
-		
-		//get and article summery for the article id.
-		if(!empty($featured_puffer_id)){
-			$data['show_featured_puffer'] = true;
-			$data['featured_puffer'] = $this->News_model->GetSummaryArticle($featured_puffer_id);
-		}else{
-			$data['show_featured_puffer'] = false;
-		}
-		
-		//Move article information into send data
-		$data['main_articles'] = $main_article_summarys;//list of main article summaryss
-		$data['lists_of_more_articles'] = $articles_summarys;//array of article lists for a category list box
-		$data['more_article_types'] = $more_article_types;//list of article types for a category list box.
-		
-		// Set up the public frame
+		$this->main_frame->SetData('menu_tab', $page_codename);
 		$this->main_frame->IncludeCss('stylesheets/home.css');
-		$this->main_frame->SetContentSimple('homepages/'.$homepage_article_type, $data);
-		// Load the public frame view (which will load the content view)
+		$this->main_frame->SetContentSimple('homepages/'.$page_codename, $data);
 		$this->main_frame->Load();
 	}
Index: system/application/views/frames/public_frame.php
===================================================================
--- system/application/views/frames/public_frame.php (revision bab8e2b098716262ea99656180e109f393ef7ccd)
+++ system/application/views/frames/public_frame.php (revision e9121dea03fda64dc58f9db18a0779b508fb9ffb)
@@ -122,5 +122,5 @@
 	array('arts', '/arts', array()),
 	array('lifestyle', '/lifestyle', array()),
-	array('roses', '/roses', array('last')),
+	array('food', '/food', array('last')),
 );
 // Has a tab been set as selected?
Index: system/application/views/homepages/food.php
===================================================================
--- system/application/views/homepages/food.php (revision df1d7f91a2a59280246afec98c616a4e723dab5a)
+++ system/application/views/homepages/food.php (revision e9121dea03fda64dc58f9db18a0779b508fb9ffb)
@@ -1,25 +1,58 @@
-<div id="RightColumn">
-	<h2 class="first">
-		<?php echo(xml_escape($links_heading)); ?>
-	</h2>
-	<div class="Entry">
-		<div class="Puffer">
-			<a href="/reviews/food">
-				<img src="/images/prototype/news/food_reviews.jpg" alt="Food Reviews (Photo: Jennifer Dickert)" title="Food Reviews (Photo: Jennifer Dickert)" />
+<?php if (!empty($spotlight)) { ?>
+<div class="FlexiBox Box23">
+	<div id="DisplayBox">
+		<div id="DisplayBoxBg"><?php echo(xml_escape($spotlight[0]['headline'])); ?></div>
+		<div id="DisplayBoxText"><a href="/news/<?php echo(xml_escape($spotlight[0]['id'])); ?>"><?php echo(xml_escape($spotlight[0]['headline'])); ?></a></div>
+		<a href="/news/<?php echo(xml_escape($spotlight[0]['id'])); ?>"><img src="/photos/home/<?php echo(xml_escape($spotlight[0]['photo_id'])); ?>" alt="" /></a>
+	</div>
+</div>
+<?php } ?>
+
+<?php if (!empty($features['articles'])) { ?>
+	<div class="ArticleListBox FlexiBox Box13 FlexiBoxLast" style="float:right">
+		<div class="ArticleListTitle">
+			<a href="/news/<?php echo(xml_escape($features['articles'][0]['id'])); ?>">
+				<?php echo(xml_escape($features['title'])); ?>
 			</a>
 		</div>
-		<?php $this->homepage_boxes->print_puffer_column($puffers); ?>
+		<?php foreach ($features['articles'] as $article) { ?>
+		<div>
+                        <a href="/news/<?php echo(xml_escape($article['id'])); ?>">
+                                <img src="/photos/small/<?php echo(xml_escape($article['photo_id'])); ?>" alt="<?php echo(xml_escape($article['photo_title'])); ?>" title="<?php echo(xml_escape($article['photo_title'])); ?>" />
+                                <?php echo(xml_escape($article['headline'])); ?>
+                        </a>
+                        <div class="Date"><?php echo(xml_escape(date('l, jS F Y', $article['date']))); ?></div>
+			<div class="clear"></div>
+		</div>
+		<?php } ?>
 	</div>
-</div>
-<div id="MainColumn">
-	<div id="HomeBanner">
-		<?php
-		$this->homepage_boxes->print_homepage_banner($banner);
-		?>
+<?php } ?>
+
+<?php function ArticleList ($section, $articles, $last = false) {
+	if (count($articles) == 0) return; ?>
+	<div class="ArticleListBox FlexiBox Box13<?php if ($last) echo(' FlexiBoxLast'); ?>">
+		<div class="ArticleListTitle">
+			<a href="/news/<?php echo($articles[0]['id']); ?>"><?php echo($section); ?></a>
+		</div>
+		<?php foreach ($articles as $article) { ?>
+		<div>
+			<a href="/news/<?php echo(xml_escape($article['id'])); ?>">
+				<img src="/photos/small/<?php echo(xml_escape($article['photo_id'])); ?>" alt="<?php echo(xml_escape($article['photo_title'])); ?>" title="<?php echo(xml_escape($article['photo_title'])); ?>" />
+				<?php echo(xml_escape($article['headline'])); ?>
+			</a>
+			<div class="Date"><?php echo(xml_escape(date('l, jS F Y', $article['date']))); ?></div>
+			<div class="clear"></div>
+		</div>
+		<?php } ?>
 	</div>
-	<?php
-	$this->homepage_boxes->print_box_with_picture_list($main_articles,$latest_heading,'news');
-	if($show_featured_puffer) $this->homepage_boxes->print_specials_box($featured_puffer_title,$featured_puffer);
-	if(!empty($lists_of_more_articles)) $this->homepage_boxes->print_box_of_category_lists($more_heading,$more_article_types,$lists_of_more_articles);
-	?>
-</div>
+	<?php if ($last) { ?><div class="clear"></div><?php } ?>
+<?php } ?>
+
+<?php
+$column = 0;
+foreach ($sections as $section) {
+	$column++;
+	ArticleList($section['title'], $section['articles'], ($column % 3) == 0 ? true : false);
+}
+?>
+<div class="clear"></div>
