Index: system/application/views/office/business.php
===================================================================
--- system/application/views/office/business.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
+++ system/application/views/office/business.php (revision f3808ecc21a309f80a91da92cef35ca8ae77ab64)
@@ -6,8 +6,10 @@
 		<li><a href="/office/prlist">Directory</a></li>
 		<li><a href="/office/pr/summary">PR System</a></li>
+		<li><a href="/office/whatson">What's On (Event Management)</a></li>
 		<li><a href="/office/campaign">Campaigns</a></li>
 		<li><a href="/office/charity">Charities</a></li>
 		<li><a href="/office/howdoi">How Do I</a></li>
 		<li><a href="/office/links">Links</a></li>
+		<li><a href="/office/crosswords">Crosswords System</a></li>
 	</ul>
 </div>
