Index: system/application/controllers/admin/index.php
===================================================================
--- system/application/controllers/admin/index.php (revision 6f50883ab1476680883a34764db1f4ecc8968fdc)
+++ system/application/controllers/admin/index.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
@@ -11,14 +11,14 @@
 	function index()
 	{
-		if (!CheckPermissions('admin')) return;
-		
+		if (!CheckPermissions('office')) return;
 		$this->pages_model->SetPageCode('admin_index');
+		$data = array();
 
 		$this->load->model('feedback_model');
 		$data['feedback_count'] = $this->feedback_model->GetFeedbackCount();
 
+		// Set up the content
+		$this->main_frame->SetData('menu_tab', 'admin');
 		$this->main_frame->SetContentSimple('admin/admin',$data);
-
-		// Load the public frame view (which will load the content view)
 		$this->main_frame->Load();
 	}
Index: system/application/controllers/office/article.php
===================================================================
--- system/application/controllers/office/article.php (revision 75c96f6680cf6806f6ecbc739d89bd71b9168b83)
+++ system/application/controllers/office/article.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
@@ -98,4 +98,5 @@
 		);
 		$this->main_frame->SetContentSimple('office/article/main', $data);
+		$this->main_frame->SetData('menu_tab', 'articles');
 		$this->main_frame->IncludeJs('javascript/wikitoolbar.js');
 		$this->main_frame->IncludeJs('/javascript/office_interface.js');
Index: system/application/controllers/office/articles.php
===================================================================
--- system/application/controllers/office/articles.php (revision 2d147e05f7fa06245084144878154cd358a26ec8)
+++ system/application/controllers/office/articles.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
@@ -28,4 +28,5 @@
 
 		/// Set up the main frame
+		$this->main_frame->SetData('menu_tab', 'articles');
 		$this->main_frame->SetContentSimple('office/news/scheduled_and_live', $data);
 		$this->main_frame->Load();
Index: system/application/controllers/office/business.php
===================================================================
--- system/application/controllers/office/business.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
+++ system/application/controllers/office/business.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ *	@brief	Office homepage 2.0
+ *	@author	Chris Travis (cdt502 - ctravis@gmail.com)
+ */
+
+class Business extends Controller
+{
+
+	function __construct()
+	{
+		parent::Controller();
+	}
+
+	function index()
+	{
+		if (!CheckPermissions('office')) return;
+		$this->pages_model->SetPageCode('office_index');
+		$data = array();
+
+		// Set up the content
+		$this->main_frame->SetData('menu_tab', 'business');
+		$this->main_frame->SetContentSimple('office/business', $data);
+		$this->main_frame->Load();
+	}
+
+}
+
+?>
Index: system/application/controllers/office/gallery.php
===================================================================
--- system/application/controllers/office/gallery.php (revision 91403c7751374d1fbb300bf532a0a3cb93fa330c)
+++ system/application/controllers/office/gallery.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
@@ -99,4 +99,5 @@
 
 		// Set up the master frame.
+		$this->main_frame->SetData('menu_tab', 'gallery');
 		$this->main_frame->IncludeCss('stylesheets/gallery.css');
 		$this->main_frame->SetTitle('Photo Gallery');
@@ -165,4 +166,5 @@
 		$data['photo_tags'] = $this->photos_model->GetPhotosTags($id);
 
+		$this->main_frame->SetData('menu_tab', 'gallery');
 		$this->main_frame->SetTitle('Photo Gallery');
 		$this->main_frame->IncludeJs('javascript/prototype.js');
@@ -224,4 +226,5 @@
 		$this->main_frame->SetContentSimple('office/gallery/massupload', $data);
 		$this->main_frame->IncludeCss('stylesheets/gallery.css');
+		$this->main_frame->SetData('menu_tab', 'gallery');
 		$this->main_frame->AddExtraHead('<style type="text/css">div#massupload_selection div { height: ' . $preview_height . 'px; width: ' . $preview_width . 'px; }</style>');
 
@@ -385,4 +388,5 @@
 		$this->main_frame->SetTitle('Photo Recropper');
 		$this->main_frame->SetExtraHead($this->xajax->getJavascript(null, '/javascript/xajax.js'));
+		$this->main_frame->SetData('menu_tab', 'gallery');
 		$this->main_frame->IncludeCss('stylesheets/cropper.css');
 		$this->main_frame->IncludeJs('javascript/prototype.js');
Index: system/application/views/admin/admin.php
===================================================================
--- system/application/views/admin/admin.php (revision f967daa9d0800331a4d8e87729915d033fc6dcf3)
+++ system/application/views/admin/admin.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
@@ -1,14 +1,38 @@
-<div id="MainColumn">
-	<div class="BlueBox">
-		<h2>Admin Areas</h2>
-		<p>The admin section isnt used for much anymore and there probably isnt any need to see this page, someone might move everything into the office one day.</p>
-		<ul>
-			<li><a href="/admin/useradmin/edit">Edit User Information</a></li>
-			<li><a href="/office/contactus">Edit contact us address'</a></li>
-			<li><a href="/admin/feedback">Feedback (<?php echo $feedback_count; ?> entries)</a></li>
-			<li><a href="/admin/pages">Page properties, custom pages, etc.</a></li>
-			<li><a href="/admin/dev">Webserver information</a></li>
-			<li><a href="/admin/imagecp">Image Management</a></li>
-		</li>
-	</div>
+<div class="BlueBox">
+	<h2>admin tools</h2>
+	<ul>
+		<li><a href="/office/announcements">Announcements</a></li>
+		<li><a href="/office/manage/members">Manage Yorker Members</a></li>
+		<li><a href="/admin/permissions">Permissions</a></li>
+		<li><a href="/office/moderator">Comment Moderation</a></li>
+		<li><a href="/admin/pages">Page Properties</a></li>
+		<li><a href="/admin/feedback">Feedback</a></li>
+		<li><a href="/office/stats">Statistics</a></li>
+		<li><a href="/office/articletypes">Article Types</a></li>
+		<li><a href="/office/specials">Special Articles</a></li>
+		<li><a href="/office/ticker">Facebook Articles</a></li>
+		<li><a href="/office/polls">Polls</a></li>
+		<li><a href="/office/podcasts">Podcasts</a></li>
+		<li><a href="/office/reviewlist/foodreviews">Food Reviews</a></li>
+		<li><a href="/office/reviewlist/drinkreviews">Drink Reviews</a></li>
+		<li><a href="/office/reviewtags">Review Tags</a></li>
+		<li><a href="/office/leagues">Review Leagues</a></li>
+		<li><a href="/office/howdoi">How Do I</a></li>
+		<li><a href="/office/games">Game Zone</a></li>
+		<li><a href="/office/photos">Photo Requests</a></li>
+		<li><a href="/office/banners">Homepage Banners</a></li>
+		<li><a href="/office/quotes">Quotes</a></li>
+		<li><a href="/office/links">Links</a></li>
+		<li><a href="/office/guide">Style Guide</a></li>
+	</ul>
+
+	<h2>developer tools</h2>
+	<ul>
+		<li><a href="/admin/useradmin/edit">Edit User Information</a></li>
+		<li><a href="/office/contactus">Edit contact us address'</a></li>
+		<li><a href="/admin/feedback">Feedback (<?php echo $feedback_count; ?> entries)</a></li>
+		<li><a href="/admin/pages">Page properties, custom pages, etc.</a></li>
+		<li><a href="/admin/dev">Webserver information</a></li>
+		<li><a href="/admin/imagecp">Image Management</a></li>
+	</ul>
 </div>
Index: system/application/views/frames/office_frame.php
===================================================================
--- system/application/views/frames/office_frame.php (revision ed47f54b421284966d39e2bef38df3a5ca6156a2)
+++ system/application/views/frames/office_frame.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
@@ -105,6 +105,43 @@
 	</div>
 
-	<div id="Navigation">
-		<ul id="Tabs" class="next">
+<?php
+$menu = array(
+	array('home', '/office', array('first')),
+	array('articles', '/office/articles', array()),
+	array('gallery', '/office/gallery', array()),
+	array('business', '/office/business', array()),
+	array('admin', '/admin', array('last'))
+);
+// Has a tab been set as selected?
+$menu_style = '';
+if (!empty($menu_tab)) {
+        $menu_key = null;
+        for ($x = 0; $x < count($menu); $x++) {
+                if ($menu[$x][0] == $menu_tab) {
+                        $menu_key = $x;
+                        break;
+                }
+        }
+        if ($menu_key !== null) {
+                $menu[$menu_key][2][] = 'current';
+                if (($menu_key - 1) > -1) {
+                        $menu[$menu_key - 1][2][] = 'next';
+                } else {
+                        $menu_style = 'next';
+                }
+        }
+}
+?>
+        <div id="Navigation">
+                <ul id="Tabs"<?php if (!empty($menu_style)) echo(' class="' . $menu_style . '"'); ?>>
+<?php foreach ($menu as $tab) { ?>
+                        <li<?php if (!empty($tab[2])) echo(' class="' . implode(' ', $tab[2]) . '"'); ?>><a href="<?php echo($tab[1]); ?>"><?php echo($tab[0]); ?></a></li>
+<?php } ?>
+<?php foreach ($links as $link) { ?>
+                        <li class="link"><?php echo($link); ?></li>
+<?php } ?>
+                </ul>
+        </div>
+
 <!--
 			<li class="first next"><a href="/">Public Site</a></li>
@@ -112,11 +149,4 @@
 			<li class="last"><a href="http://mail.theyorker.co.uk">Webmail</a></li>
 -->
-			<li class="first current"><a href="/office">home</a></li>
-			<li><a href="/office/articles">articles</a></li>
-			<li><a href="/office/gallery">gallery</a></li>
-			<li><a href="/office/business">business</a></li>
-			<li class="last"><a href="/office/admin">admin</a></li>
-		</ul>
-	</div>
 
 	<div id="Page">
Index: system/application/views/office/business.php
===================================================================
--- system/application/views/office/business.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
+++ system/application/views/office/business.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
@@ -0,0 +1,13 @@
+<div class="BlueBox">
+	<h2>business tools</h2>
+	<ul>
+		<li><a href="/office/vipmanager">Manage VIPs</a></li>
+		<li><a href="/office/advertising">Advertising</a></li>
+		<li><a href="/office/prlist">Directory</a></li>
+		<li><a href="/office/pr/summary">PR System</a></li>
+		<li><a href="/office/campaign">Campaigns</a></li>
+		<li><a href="/office/charity">Charities</a></li>
+		<li><a href="/office/howdoi">How Do I</a></li>
+		<li><a href="/office/links">Links</a></li>
+	</ul>
+</div>
Index: system/application/views/office/index.php
===================================================================
--- system/application/views/office/index.php (revision d38c637f4ffd74ce5a5318aa1cbd96e3fdacb4a0)
+++ system/application/views/office/index.php (revision 093571abf748ac14e0ecc8f5617c52ac6fa82284)
@@ -36,4 +36,5 @@
 	<h2>links</h2>
 	<p><a href="/office/bylines">Bylines</a></p>
+	<p><a href="/office/irc">IRC Chat</a></p>
 	<h2>notifications</h2>
 	<?php
