Index: export/template/system/application/config/config.php
===================================================================
--- export/template/system/application/config/config.php (revision b45b6a05de69a2d6238e921b6dea66655d9ff66d)
+++ export/template/system/application/config/config.php (revision 495cc0954360b05b81d7c6a9d2e4c66e9d5397a1)
@@ -122,5 +122,5 @@
 |
 */
-$config['permitted_uri_chars'] = 'a-z 0-9~%.:_-';
+$config['permitted_uri_chars'] = '@a-z 0-9~%.:_-';
 
 
