Index: system/application/helpers/crosswords_helper.php
===================================================================
--- system/application/helpers/crosswords_helper.php (revision a8879de74f98612afdbf5fc4889f28f3382bc7b9)
+++ system/application/helpers/crosswords_helper.php (revision 4d2d2d2d04ed914d43b1fe7734a3af99e75c7796)
@@ -936,5 +936,4 @@
 					$check_actions = array(
 						// check buttons
-						'stop' => 'crosswordStopCheck('.js_literalise($name).');',
 						'check selected answer' => 'crosswordCheck('.js_literalise($name).','.js_literalise('cur_light').', false);',
 						'check all answers' => 'crosswordCheck('.js_literalise($name).','.js_literalise('all_lights').', false);',
@@ -943,8 +942,12 @@
 						'reveal selected answer' => 'crosswordCheck('.js_literalise($name).','.js_literalise('cur_light').', true);',
 						'reveal all answers' => 'crosswordCheck('.js_literalise($name).','.js_literalise('all_lights').', true);',
+						null,
+						'clear grid' => 'crosswordClear('.js_literalise($name).');',
+						'stop checking' => 'crosswordStopCheck('.js_literalise($name).');',
 					);
 					foreach ($check_actions as $action_name => $javascript) {
 						if ($javascript == null) {
-							?><br /><?php
+							?></fieldset><?php
+							?><fieldset><?php
 							continue;
 						}
@@ -959,14 +962,4 @@
 				?></fieldset><?php
 			?></div><?php
-			// Clear grid button
-			?><fieldset><?php
-				?><input	type="button"<?php
-						?>	class="button"<?php
-						?>	value="clear grid"<?php
-						?>	onclick="<?php echo(xml_escape(
-								'crosswordClear('.js_literalise($name).');'
-							)); ?>"<?php
-						?>	/><?php
-			?></fieldset><?php
 		}
 		?></div><?php
