Index: design/crosswords/create_tables.sql
===================================================================
--- design/crosswords/create_tables.sql (revision cd98c170fb151153e6f3271267995b7d5a0e8f21)
+++ design/crosswords/create_tables.sql (revision 2106f3d69a4421f9539af9e40baa244000833512)
@@ -61,8 +61,8 @@
 CREATE TABLE `crossword_authors` (
 	`crossword_author_crossword_id` INT NOT NULL ,
-	`crossword_author_user_entity_id` INT NOT NULL ,
+	`crossword_author_business_card_id` INT NOT NULL ,
 	PRIMARY KEY (
 		`crossword_author_crossword_id` ,
-		`crossword_author_user_entity_id`
+		`crossword_author_business_card_id`
 	),
 	FOREIGN KEY (
