Index: export/template/system/application/config/yorker_override.php
===================================================================
--- export/template/system/application/config/yorker_override.php (revision e30df09986b70e48d8ed9b641387e6d0aa0ad53f)
+++ export/template/system/application/config/yorker_override.php (revision 03fbaaf08e646568c346e75f59266847674a2c29)
@@ -13,5 +13,6 @@
 $config['enable_analytics'] = true;
 
-$config['temp_local_path'] = '/srv/www/public_html/tmp/uploads/';
+$config['temp_local_path'] = '/srv/www/public_html/tmp/';
+$config['temp_web_address'] = '/tmp/';
 $config['static_local_path'] = '/srv/static';
 
