Index: system/application/views/homepages/roses.php
===================================================================
--- system/application/views/homepages/roses.php (revision 3b47ba93be6804427c6bc742f6cc3f247b8c4bab)
+++ system/application/views/homepages/roses.php (revision 1420ed1eae4cd88cc5a4fefb5d6f72838b68fa4f)
@@ -14,5 +14,5 @@
 <div style="float:right;" class="ArticleListBox FlexiBox Box13 FlexiBoxLast">
 	<div class="ArticleListTitle">
-		<a href="/news/<?php echo($liveblog[0]['headline']); ?>">latest updates</a>
+		<a href="/news/<?php echo($liveblog[0]['id']); ?>">latest updates</a>
 	</div>
 	<?php
