Index: public_html/stylesheets/crosswords.css
===================================================================
--- public_html/stylesheets/crosswords.css (revision 1093f66b8178069ced62cd7402f1ce408b8c3ce9)
+++ public_html/stylesheets/crosswords.css (revision 00144a18691f6be24e91586672dbb837b9288a24)
@@ -15,4 +15,7 @@
 div.crosswordWinners ol li.new_winner {
 	border: 1px solid #999999;
+}
+div.crosswordWinners div.expiry_time {
+	font-size: 8pt;
 }
 /* gold */
Index: public_html/stylesheets/crosswords_index.css
===================================================================
--- public_html/stylesheets/crosswords_index.css (revision 6e2c86c9a08b0e018aa9918ae55993be5e7d4037)
+++ public_html/stylesheets/crosswords_index.css (revision 00144a18691f6be24e91586672dbb837b9288a24)
@@ -9,4 +9,8 @@
 	float: left;
 	width: 50%; 
+}
+div.BlueBox.crossword_box {
+	border: none;
+	margin: 0px;
 }
 div.crossword_preview {
@@ -30,2 +34,8 @@
 	float: right;
 }
+div.crossword_new {
+	float: left;
+	font-weight: bold;
+	font-size: large;
+	color: #FF6A00;
+}
Index: system/application/libraries/Academic_calendar.php
===================================================================
--- system/application/libraries/Academic_calendar.php (revision 22033b16e5e93ea06c34477348f0ca1fafb7fb33)
+++ system/application/libraries/Academic_calendar.php (revision 00144a18691f6be24e91586672dbb837b9288a24)
@@ -582,4 +582,22 @@
 				$result[$unit] = 0;
 			}
+		}
+		return $result;
+	}
+
+	/// Turn a difference into strings.
+	static function DescribeDifference($Units, $Diff, $MaxUnits = -1)
+	{
+		$result = array();
+		foreach ($Diff as $unit => $amount) {
+			if ($amount != 0) {
+				$result[] = $amount.' '.$Units[$unit][($amount==1)?1:0];
+				if (count($result) == $MaxUnits) {
+					break;
+				}
+			}
+		}
+		if (empty($result)) {
+			$result[] = '0 '.$Units[$unit][0];
 		}
 		return $result;
Index: system/application/views/crosswords/category.php
===================================================================
--- system/application/views/crosswords/category.php (revision 52a81e29044b174cdf7b4d2e16a67e8e7065a98b)
+++ system/application/views/crosswords/category.php (revision 00144a18691f6be24e91586672dbb837b9288a24)
@@ -32,7 +32,10 @@
 	}
 
+	?><div class="HalfColumns"><?php
 	foreach ($Category['latest'] as $crossword) {
 		$pub = new Academic_time($crossword['publication']);
-		?><div style="clear: both"></div><?php
+		?><div class="Column"><?php
+		?><div class="crossword_box BlueBox"><?php
+
 		?><div class="crossword_preview"><?php
 			?><a href="<?php echo(site_url('crosswords/'.$crossword['id'])); ?>"><?php
@@ -40,4 +43,12 @@
 			?></a><?php
 		?></div><?php
+		// Find if crossword is "new"
+		$now = new Academic_time(time());
+		$since_publication = Academic_time::Difference($pub, $now, array('days'));
+		if (!$crossword['expired'] || $since_publication['days'] < 7)
+		{
+			?><div class="crossword_new">new!</div><?php
+		}
+		// Title and details
 		?><div class="crossword_title"><?php
 			?><a href="<?php echo(site_url('crosswords/'.$crossword['id'])); ?>"><?php
@@ -68,5 +79,9 @@
 			}
 		}
+		?></div><?php
+		?></div><?php
 	}
+	?></div><?php
+
 ?></div><?php
 ?>
Index: system/application/views/crosswords/crossword.php
===================================================================
--- system/application/views/crosswords/crossword.php (revision fbddb156e0fa4716316d368a8bbe0b731010af00)
+++ system/application/views/crosswords/crossword.php (revision 00144a18691f6be24e91586672dbb837b9288a24)
@@ -97,4 +97,25 @@
 				?></ol><?php
 			}
+			if (!$Crossword['expired']) {
+				if (null !== $Crossword['expiry']) {
+					$now = new Academic_time(time());
+					$expiry = new Academic_time($Crossword['expiry']);
+					$units = array(
+						'week' => array('weeks', 'week'),
+						'day' => array('days', 'day'),
+						'hour' => array('hours', 'hour'),
+						'min' => array('minutes', 'minute'),
+						'sec' => array('seconds', 'second'),
+					);
+					$diff = Academic_time::DescribeDifference(
+						$units, Academic_time::Difference($now, $expiry, array_keys($units)), 2
+					);
+					?><div class="expiry_time"><?php
+					echo('solutions available in:');
+					?><br /><?php
+					echo(join(', ', $diff));
+					?></div><?php
+				}
+			}
 		?></div><?php
 	}
Index: system/application/views/crosswords/index.php
===================================================================
--- system/application/views/crosswords/index.php (revision e4f1bf6a2739f084bc8a3f3ad54c46bdb98ccb8a)
+++ system/application/views/crosswords/index.php (revision 00144a18691f6be24e91586672dbb837b9288a24)
@@ -43,5 +43,5 @@
 			foreach ($category['latest'] as $crossword) {
 				$pub = new Academic_time($crossword['publication']);
-				?><div style="clear: both"></div><?php
+				?><div class="crossword_box BlueBox"><?php
 				?><div class="crossword_preview"><?php
 					?><a href="<?php echo(site_url('crosswords/'.$crossword['id'])); ?>"><?php
@@ -49,4 +49,12 @@
 					?></a><?php
 				?></div><?php
+				// Find if crossword is "new"
+				$now = new Academic_time(time());
+				$since_publication = Academic_time::Difference($pub, $now, array('days'));
+				if (!$crossword['expired'] || $since_publication['days'] < 7)
+				{
+					?><div class="crossword_new">new!</div><?php
+				}
+				// Title and details
 				?><div class="crossword_title"><?php
 					?><a href="<?php echo(site_url('crosswords/'.$crossword['id'])); ?>"><?php
@@ -77,4 +85,5 @@
 					}
 				}
+				?></div><?php
 			}
 		?></div><?php
