Index: system/application/models/comments_model.php
===================================================================
--- system/application/models/comments_model.php (revision bcdf62404b94f6f234274cdd45f52a2f6a4bc543)
+++ system/application/models/comments_model.php (revision 833b4fc0ab01afa1a410d938325d8f0a5933a487)
@@ -383,5 +383,5 @@
 						AND			article_comments.comment_post_time <= comments.comment_post_time
 					)	AS article_comment_count,
-						"article" AS source
+						1 AS source
 				FROM	articles,
 						comments,
@@ -412,5 +412,5 @@
 						AND			article_comments.comment_post_time <= comments.comment_post_time
 					)	AS article_comment_count,
-						"crossword" AS source
+						2 AS source
 				FROM	comments
 				INNER JOIN	users
@@ -429,5 +429,5 @@
 		$results = $query->result_array();
 		foreach ($results as &$result) {
-			if ($result['source'] == 'crossword') {
+			if ($result['source'] == 2) {
 				$pub = new Academic_time($result['publication']);
 				$pub_date = $pub->Format('D').' week '.$pub->AcademicWeek();
