Index: system/application/libraries/Wikiparser.php
===================================================================
--- system/application/libraries/Wikiparser.php (revision 5d6483dff7f59a1b683e92c31bf474588e5ebfbd)
+++ system/application/libraries/Wikiparser.php (revision 5154d1b9f0fbfb26b12f0684f775ebd047a07e2b)
@@ -397,5 +397,5 @@
 			$params = array('src', 'http://www.youtube.com/v/' . $href . '&hd=1',
 					'width', '620',
-					'height', '350');
+					'height', '375');
 			$output = $this->get_inline_flash_code($params);
 
