Index: system/application/controllers/office/liveblog.php
===================================================================
--- system/application/controllers/office/liveblog.php (revision 7e42227166b7c13c638a37815515d6ab588080e2)
+++ system/application/controllers/office/liveblog.php (revision 2d569d9ae8f2a5e963894672119f40b9080e7d6c)
@@ -709,6 +709,6 @@
 
 		// Range
-		$o .= '&chxr=1,0,180,25';
-		$o .= '&chds=0,180,0,180';
+		$o .= '&chxr=1,0,200,50';
+		$o .= '&chds=0,200,0,200';
 
 		$sql = 'SELECT * FROM roses_scores WHERE event_score_time IS NOT NULL ORDER BY event_score_time ASC';
