Index: system/application/views/general/home.php
===================================================================
--- system/application/views/general/home.php (revision 634bdd9d4e1012a235b62d88ab803992a89fa194)
+++ system/application/views/general/home.php (revision 337b54f68ea5ca9905cea626bbc7c5ae51c14245)
@@ -73,6 +73,15 @@
 
 <div class="FlexiBox Box12">
-	<div class="FlexiBox Box12b">
-		<br /><br /><br /><br /><br />
+	<div class="FlexiBox Box12">
+		<script type="text/javascript">
+		<!--
+		google_ad_client = "pub-8676956632365960";
+		/* 468x60, created 07/05/09 */
+		google_ad_slot = "5537704417";
+		google_ad_width = 468;
+		google_ad_height = 60;
+		//-->
+		</script>
+		<script type="text/javascript" src="http://pagead2.googlesyndication.com/pagead/show_ads.js"></script>
 	</div>
 	<div class="ArticleListBox FlexiBox Box12">
