Index: system/application/libraries/Account_personal.php
===================================================================
--- system/application/libraries/Account_personal.php (revision 80d9468002cc8f6a2436fa5ac97fa988344ab489)
+++ system/application/libraries/Account_personal.php (revision 1f5c6794212fd84fca356f47adbddad9d2c5e910)
@@ -41,5 +41,5 @@
 		$rules['nick'] = 'trim|required|alpha_numeric';
 		$rules['gender'] = 'trim';
-		$rules['college'] = 'trim|required|is)numeric';
+		$rules['college'] = 'trim|required|is_numeric';
 		$rules['year'] = 'trim|required|is_numeric';
 		$rules['time'] = 'trim|required|numeric';
