Index: export/template/system/application/config/yorker_override.php
===================================================================
--- export/template/system/application/config/yorker_override.php (revision 9862e4ffbf12ebc2b33cc9a217282a2cf3436a68)
+++ export/template/system/application/config/yorker_override.php (revision e71456289bd48cec1d420c382d2b5a0bfff9fb24)
@@ -11,4 +11,5 @@
 // Enable google adsense
 $config['enable_adsense'] = true;
+$config['enable_analytics'] = true;
 
 ?>
