Index: system/application/controllers/office/whatson.php
===================================================================
--- system/application/controllers/office/whatson.php (revision d0844599d1dea2fa1c103a69cda5a417cb3ca048)
+++ system/application/controllers/office/whatson.php (revision b9e8880163ac368118796e12a0070b1a92477c39)
@@ -18,7 +18,56 @@
 
 		$this->load->model('calendar/whatson_model');
-		$info = $this->whatson_model->GetAllCalendarsInfo();
+		$calendars = $this->whatson_model->GetAllCalendarsInfo();
 
+		$data = array(
+			'Calendars' => $calendars,
+		);
+
+		$this->main_frame->SetContentSimple('whatson/office/index', $data);
 		$this->main_frame->Load();
+	}
+
+	function calendar($shortname = null, $op = null)
+	{
+		if (!CheckPermissions('editor')) return;
+
+		$this->load->model('calendar/whatson_model');
+		if (null === $shortname) {
+			show_404();
+		}
+		else {
+			$calendar = $this->whatson_model->GetCalendarInfoByShortName($shortname);
+			if (null === $calendar) {
+				$this->messages->AddMessage('error', xml_escape("No what's on calendar found with the short name '$shortname'"));
+			}
+			else {
+				$name = $calendar['name'];
+				switch ($op) {
+					case 'moveup':
+						if ($this->whatson_model->UpdateCalendarOrder($calendar, $calendar['order']-1)) {
+							$this->messages->AddMessage('success', xml_escape("Calendar $name successfully moved up"));
+						}
+						else {
+							$this->messages->AddMessage('error', xml_escape("Calendar $name not moved up"));
+						}
+						break;
+					case 'movedown':
+						if ($this->whatson_model->UpdateCalendarOrder($calendar, $calendar['order']+1)) {
+							$this->messages->AddMessage('success', xml_escape("Calendar $name successfully moved down"));
+						}
+						else {
+							$this->messages->AddMessage('error', xml_escape("Calendar $name not moved down"));
+						}
+						break;
+					default:
+						show_404();
+				}
+			}
+		}
+
+		if (isset($_GET['ret'])) {
+			redirect($_GET['ret']);
+		}
+		redirect('office/whatson');
 	}
 
Index: system/application/models/calendar/whatson_model.php
===================================================================
--- system/application/models/calendar/whatson_model.php (revision d0844599d1dea2fa1c103a69cda5a417cb3ca048)
+++ system/application/models/calendar/whatson_model.php (revision b9e8880163ac368118796e12a0070b1a92477c39)
@@ -44,8 +44,9 @@
 		$sql =
 			'SELECT '.
-			'	`whatson_calendar_entity_id`		AS `calendar_id`, '.
-			'	`whatson_calendar_order`			AS `order`, '.
-			'	`organisation_name`					AS `name`, '.
-			'	`organisation_directory_entry_name`	AS `shortname` '.
+			'	`whatson_calendar_organisation_entity_id`	AS `organisation_id`, '.
+			'	`whatson_calendar_entity_id`				AS `calendar_id`, '.
+			'	`whatson_calendar_order`					AS `order`, '.
+			'	`organisation_name`							AS `name`, '.
+			'	`organisation_directory_entry_name`			AS `shortname` '.
 			'FROM `whatson_calendars` '.
 			'INNER JOIN `organisations` '.
@@ -159,4 +160,97 @@
 		return $success;
 	}
+
+	function UpdateCalendarOrder($calendar_info, $new_order)
+	{
+		$old_order = $calendar_info['order'];
+		if ($new_order == $old_order) {
+			return false;
+		}
+		elseif ($new_order < $old_order && $new_order >= 0) {
+			// move ones in between up
+			// Â· a b c y Â·
+			// Â· y a b c Â·
+
+			// make space
+			$sql =
+				'UPDATE	`whatson_calendars` '.
+				'SET	`whatson_calendar_order` = -1 '.
+				'WHERE	`whatson_calendar_order` = ? '
+				;
+			$this->db->query($sql, array($old_order));
+			if ($this->db->affected_rows()) {
+				// fill space
+				$sql =
+					'UPDATE	`whatson_calendars` '.
+					'SET	`whatson_calendar_order` = `whatson_calendar_order` + 1 '.
+					'WHERE	`whatson_calendar_order` >= ? '.
+					'	AND	`whatson_calendar_order` < ? '.
+					'ORDER BY `whatson_calendar_order` DESC '
+					;
+				$this->db->query($sql, array($new_order,$old_order));
+				// final move
+				$sql =
+					'UPDATE	`whatson_calendars` '.
+					'SET	`whatson_calendar_order` = ? '.
+					'WHERE	`whatson_calendar_order` = -1 '
+					;
+				$this->db->query($sql, array($new_order));
+				return (bool)$this->db->affected_rows();
+			}
+			else {
+				return false;
+			}
+		}
+		elseif ($old_order < $new_order) {
+			// find the max new order
+			$sql =	'SELECT MAX(`whatson_calendar_order`) AS max_order '.
+					'FROM	`whatson_calendars` '.
+					'WHERE	`whatson_calendar_organisation_entity_id` = ? '
+					;
+			$results = $this->db->query($sql, array($calendar_info['organisation_id']))->result_array();
+			if (empty($results)) {
+				$max = -1;
+			}
+			else {
+				$max = $results[0]['max_order'];
+			}
+			if ($new_order <= $max) {
+				// move ones in between down
+				// Â· y a b c Â·
+				// Â· a b c y Â·
+
+				// make space
+				$sql =
+					'UPDATE	`whatson_calendars` '.
+					'SET	`whatson_calendar_order` = -1 '.
+					'WHERE	`whatson_calendar_order` = ? '
+					;
+				$this->db->query($sql, array($old_order));
+				if ($this->db->affected_rows()) {
+					// fill space
+					$sql =
+						'UPDATE	`whatson_calendars` '.
+						'SET	`whatson_calendar_order` = `whatson_calendar_order` - 1 '.
+						'WHERE	`whatson_calendar_order` > ? '.
+						'	AND	`whatson_calendar_order` <= ? '.
+						'ORDER BY `whatson_calendar_order` ASC '
+						;
+					$this->db->query($sql, array($old_order,$new_order));
+					// final move
+					$sql =
+						'UPDATE	`whatson_calendars` '.
+						'SET	`whatson_calendar_order` = ? '.
+						'WHERE	`whatson_calendar_order` = -1 '
+						;
+					$this->db->query($sql, array($new_order));
+					return (bool)$this->db->affected_rows();
+				}
+				else {
+					return false;
+				}
+			}
+		}
+		return $success;
+	}
 }
 
Index: system/application/views/whatson/office/index.php
===================================================================
--- system/application/views/whatson/office/index.php (revision b9e8880163ac368118796e12a0070b1a92477c39)
+++ system/application/views/whatson/office/index.php (revision b9e8880163ac368118796e12a0070b1a92477c39)
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @file views/whatson/office/index.php
+ * @author James Hogan <james_hogan@theyorker.co.uk>
+ *
+ * @param $Calendars
+ */
+
+?><div class="BlueBox"><?php
+	?><h2>what's on calendars</h2><?php
+
+	?><a href="<?php echo(xml_escape(site_url('office/whatson/add'))); ?>">create new what's on calendar</a><?php
+
+	?><ul style="width:40%"><?php
+	$id = 0;
+	foreach ($Calendars as $calendar) {
+		++$id;
+		?><li><?php
+			?><a href="<?php echo(xml_escape(site_url('office/whatson/calendar/'.$calendar['shortname']))); ?>"><?php
+				echo(xml_escape($calendar['name']));
+			?></a><?php
+			$links = array();
+			if ($id > 1) {
+				$links[site_url('office/whatson/calendar/'.$calendar['shortname'].'/moveup?ret='.urlencode($this->uri->uri_string()))] = 'up';
+			}
+			if ($id < count($Calendars)) {
+				$links[site_url('office/whatson/calendar/'.$calendar['shortname'].'/movedown?ret='.urlencode($this->uri->uri_string()))] = 'down';
+			}
+			$linkHtml = array();
+			foreach ($links as $uri => $label) {
+				$linkHtml[] = '<a href="'.xml_escape($uri).'">'.xml_escape($label).'</a>';
+			}
+			?><span style="float:right"><?php
+			echo(' ('.implode(', ', $linkHtml).')');
+			?></span><?php
+		?></li><?php
+	}
+	?></ul><?php
+?></div><?php
+
+?>
