Index: design/theyorker_tables.sql
===================================================================
--- design/theyorker_tables.sql (revision 043f033889abf589d4fd53efd9be6a8c2e46ccf6)
+++ design/theyorker_tables.sql (revision 07ce6ac214293f47d84b7b9de094e3a5bcff3ede)
@@ -50,9 +50,9 @@
 
 CREATE TABLE organisation_slideshows (
-	organisation_slideshows_organisation_entity_id INTEGER		NOT NULL,
-	organisation_slideshows_photo_id	INTEGER		NOT NULL,
-	organisation_slideshows_order		INTEGER		NULL,
-
-	PRIMARY KEY(organisation_entity_id, photo_id),
+	organisation_slideshow_organisation_entity_id INTEGER		NOT NULL,
+	organisation_slideshow_photo_id	INTEGER		NOT NULL,
+	organisation_slideshow_order		INTEGER		NOT NULL,
+
+	PRIMARY KEY(organisation_slideshow_organisation_entity_id, organisation_slideshow_photo_id),
 );
 
@@ -86,8 +86,8 @@
 -- The types of articles a user can write - reviews, features, news etc.
 CREATE TABLE users_article_types (
-	users_article_type_user_id		INTEGER		NOT NULL,
+	users_article_type_user_entity_id		INTEGER		NOT NULL,
 	users_article_type_article_type_id	INTEGER		NOT NULL,
 
-	PRIMARY KEY(users_article_type_user_id, users_article_type_article_type_id),
+	PRIMARY KEY(users_article_type_user_entity_id, users_article_type_article_type_id),
 );
 
@@ -103,5 +103,5 @@
 -- TODO: possibly work out a nice way of doing this.  
 CREATE TABLE user_has_properties (
-	user_has_properties_user_id		INTEGER		NOT NULL	AUTO_INCREMENT,
+	user_has_properties_user_entity_id		INTEGER		NOT NULL	AUTO_INCREMENT,
 	user_has_properties_property_id	INTEGER		NOT NULL,
 	user_has_properties_text		TEXT		NULL,
@@ -111,5 +111,5 @@
 	user_has_properties_number		FLOAT		NULL,
 
-	PRIMARY KEY(user_has_properties_user_id, user_has_properties_property_id),
+	PRIMARY KEY(user_has_properties_user_entity_id, user_has_properties_property_id),
 );
 
@@ -151,5 +151,5 @@
 CREATE TABLE user_links (
 	user_link_id		INTEGER		NOT NULL	AUTO_INCREMENT,
-	user_link_user_id		INTEGER		NOT NULL,
+	user_link_user_entity_id		INTEGER		NOT NULL,
 	user_link_link_id		INTEGER		NULL,
 	user_link_image_id	INTEGER		NULL,
@@ -167,7 +167,7 @@
 	user_subscription_property_organisation_entity_id	INTEGER		NOT NULL,
 	user_subscription_property_property_id	INTEGER		NOT NULL,
-	user_subscription_property_user_id		INTEGER		NOT NULL,
-
-	PRIMARY KEY(user_subscription_property_organisation_entity_id, user_subscription_property_property_id, user_subscription_property_user_id),
+	user_subscription_property_user_entity_id		INTEGER		NOT NULL,
+
+	PRIMARY KEY(user_subscription_property_organisation_entity_id, user_subscription_property_property_id, user_subscription_property_user_entity_id),
 );
 
@@ -193,5 +193,5 @@
 	business_card_id		INTEGER		NOT NULL	AUTO_INCREMENT,
 	business_card_organisation_entity_id	INTEGER		NOT NULL,
-	business_card_user_id		INTEGER		NOT NULL,
+	business_card_user_entity_id		INTEGER		NOT NULL,
 	business_card_name		VARCHAR(255)	NOT NULL,
 	business_card_title		VARCHAR(255)	NOT NULL,
@@ -242,4 +242,5 @@
 	subscription_deleted BOOL NOT NULL,
 	subscription_timestamp TIMESTAMP NOT NULL CURRENT_TIMESTAMP,
+	
 	PRIMARY KEY(subscription_organisation_entity_id, subscription_user_entity_id),
 );
@@ -251,8 +252,8 @@
 	article_id		INTEGER		NOT NULL	AUTO_INCREMENT,
 	article_article_type_id	INTEGER		NOT NULL,
-	article_last_editor_user_id INTEGER	NULL,
+	article_last_editor_user_entity_id INTEGER	NULL,
 	article_created		TIMESTAMP	NOT NULL,
 	article_publish_date	TIMESTAMP	NULL,
-	article_initial_editor_user_id INTEGER	NULL,
+	article_initial_editor_user_entity_id INTEGER	NULL,
 	article_location	VARCHAR(15)	NULL,
 	article_breaking	BOOL		NOT NULL	FALSE,
@@ -266,4 +267,5 @@
 );
 
+-- TODO: what if someone edits and checks at the same time
 -- An article can have a number of contents (i.e. revisions) written.  Only 1
 --  is live at a time.  
@@ -321,4 +323,5 @@
 	article_type_name		VARCHAR(255)	NOT NULL,
 	article_type_archive		BOOL		NOT NULL,
+	article_type_blurb		TEXT		NOT NULL,
 	
 	PRIMARY KEY(article_type_id),
@@ -327,16 +330,16 @@
 -- Links the authors of a revision of an article.  
 CREATE TABLE article_writers (
-	article_writer_user_id		INTEGER		NOT NULL,
+	article_writer_user_entity_id		INTEGER		NOT NULL,
 	article_writer_article_content_id INTEGER	NOT NULL,
 	
-	PRIMARY KEY(article_writer_user_id, article_writer_article_content_id),
+	PRIMARY KEY(article_writer_user_entity_id, article_writer_article_content_id),
 );
 
 CREATE TABLE fact_boxes (
-	id		INTEGER		NOT NULL	AUTO_INCREMENT,
-	article_content_id INTEGER	NOT NULL,
-	text		TEXT		NOT NULL,
-	deleted		BOOL		NOT NULL	FALSE,
-	timestamp	TIMESTAMP	NOT NULL	CURRENT_TIMESTAMP,
+	fact_box_id		INTEGER		NOT NULL	AUTO_INCREMENT,
+	fact_box_article_content_id INTEGER	NOT NULL,
+	fact_box_text		TEXT		NOT NULL,
+	fact_box_deleted	BOOL		NOT NULL	FALSE,
+	fact_box_timestamp	TIMESTAMP	NOT NULL	CURRENT_TIMESTAMP,
 	
 	PRIMARY KEY(fact_box_id),
@@ -344,8 +347,8 @@
 
 CREATE TABLE related_articles (
-	article_id1	INTEGER		NOT NULL,
-	article_id2	INTEGER		NOT NULL,
-
-	PRIMARY KEY(article_id1, article_id2),
+	related_article_1_article_id	INTEGER		NOT NULL,
+	related_article_2_article_id	INTEGER		NOT NULL,
+
+	PRIMARY KEY(related_article_1_article_id, related_article_2_article_id),
 );
 
@@ -353,9 +356,9 @@
 	pull_quote_id INTEGER NOT NULL AUTO_INCREMENT,
 	pull_quote_article_content_id INTEGER NOT NULL,
-	pull_quote_text TEXT NULL,
-	pull_quote_person TEXT NULL,
-	pull_quote_position TEXT NULL,
-	pull_quote_order INTEGER NULL,
-	pull_quote_deleted BOOL NULL,
+	pull_quote_text TEXT NOT NULL,
+	pull_quote_person VARCHAR(255) NOT NULL,
+	pull_quote_position VARCHAR(255) NOT NULL,
+	pull_quote_order INTEGER NOT NULL,
+	pull_quote_deleted BOOL NOT NULL,
 
 	PRIMARY KEY(pull_quote_id),
@@ -372,8 +375,10 @@
 	request_organisation_entity_id INTEGER NULL,
 	request_text TEXT NOT NULL,
+	request_blurb TEXT NOT NULL,
 	request_deadline TIMESTAMP NOT NULL,
 	request_accepted BOOL NOT NULL,
 	request_deleted BOOL NOT NULL,
 	request_timestamp TIMESTAMP NOT NULL CURRENT_TIMESTAMP,
+	
 	PRIMARY KEY(request_id),
 );
@@ -391,5 +396,6 @@
 CREATE TABLE request_types (
 	request_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	request_type_name TEXT VARCHAR(255),
+	request_type_name VARCHAR(255) NOT NULL,
+	
 	PRIMARY KEY(request_type_id)
 );
@@ -397,8 +403,9 @@
 CREATE TABLE request_users (
 	request_user_request_id INTEGER NOT NULL,
-	request_user_user_id INTEGER NOT NULL,
-	request_user_accepted BOOL NULL,
-	request_user_rejected BOOL NULL,
-	PRIMARY KEY(request_user_request_id, request_user_user_id),
+	request_user_user_entity_id INTEGER NOT NULL,
+	request_user_accepted BOOL NOT NULL,
+	request_user_rejected BOOL NOT NULL,
+	
+	PRIMARY KEY(request_user_request_id, request_user_user_entity_id),
 );
 ----------------------------------------------------------------
@@ -415,4 +422,5 @@
 	event_deleted BOOL NULL,
 	event_timestamp TIMESTAMP NULL,
+	
 	PRIMARY KEY(event_id),
 );
@@ -421,15 +429,17 @@
 CREATE TABLE events_entities (
 	event_entity_entity_id INTEGER NOT NULL,
-	event_id INTEGER NOT NULL,
-	PRIMARY KEY(event_entity_entity_id, event_id),
+	event_entity_event_id INTEGER NOT NULL,
+	
+	PRIMARY KEY(event_entity_entity_id, event_entity_event_id),
 );
 
 -- Users can customize ocurrencies on the timetable.  
-CREATE TABLE event_occurence_users (
-	event_occurence_user_user_id INTEGER NOT NULL,
-	event_occurence_user_event_occurrence_id INTEGER NOT NULL,
-	event_occurrence_user_hide BOOL NULL,
-	event_occurrence_user_rsvp BOOL NULL,
-	PRIMARY KEY(event_occurence_user_user_id, event_occurence_user_event_occurrence_id),
+CREATE TABLE event_occurrence_users (
+	event_occurrence_user_user_entity_id INTEGER NOT NULL,
+	event_occurrence_user_event_occurrence_id INTEGER NOT NULL,
+	event_occurrence_user_hide BOOL NOT NULL,
+	event_occurrence_user_rsvp BOOL NOT NULL,
+	
+	PRIMARY KEY(event_occurrence_user_user_entity_id, event_occurrence_user_event_occurrence_id),
 );
 
@@ -439,20 +449,22 @@
 	event_occurrence_event_id INTEGER NOT NULL,
 	event_occurrence_state_id INTEGER NOT NULL,
-	event_occurrence_description TEXT NULL,
-	event_occurrence_location TEXT NULL,
-	event_occurrence_postcode TEXT NULL,
-	event_occurrence_start_time TIMESTAMP NULL,
-	event_occurrence_end_time TIMESTAMP NULL,
-	event_occurrence_all_day BOOL NULL,
-	event_occurrence_ends_late BOOL NULL,
-	event_occurrence_timestamp TIMESTAMP NULL,
+	event_occurrence_description TEXT NOT NULL,
+	event_occurrence_location VARCHAR(15) NOT NULL,
+	event_occurrence_postcode VARCHAR(15) NOT NULL,
+	event_occurrence_start_time TIMESTAMP NOT NULL,
+	event_occurrence_end_time TIMESTAMP NOT NULL,
+	event_occurrence_all_day BOOL NOT NULL,
+	event_occurrence_ends_late BOOL NOT NULL,
+	event_occurrence_timestamp TIMESTAMP NOT NULL CURRENT_TIMESTAMP,
+	
 	PRIMARY KEY(event_occurrence_id),
 );
 
 CREATE TABLE event_occurrence_states (
-	event_occurrences_state_id INTEGER NOT NULL AUTO_INCREMENT,
-	event_occurrence_state_name TEXT NULL,
-	event_occurrence_state_published BOOL NULL,
-	PRIMARY KEY(event_occurrences_state_id)
+	event_occurrence_state_id INTEGER NOT NULL AUTO_INCREMENT,
+	event_occurrence_state_name VARCHAR(255) NOT NULL,
+	event_occurrence_state_published BOOL NOT NULL,
+	
+	PRIMARY KEY(event_occurrence_state_id)
 );
 
@@ -460,5 +472,6 @@
 CREATE TABLE event_types (
 	event_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	event_type_name TEXT NULL,
+	event_type_name VARCHAR(255) NOT NULL,
+	
 	PRIMARY KEY(event_type_id)
 );
@@ -466,8 +479,8 @@
 CREATE TABLE reminders (
 	reminder_id INTEGER NOT NULL AUTO_INCREMENT,
-	reminder_user_id INTEGER NOT NULL,
-	reminder_name INTEGER NULL,
-	reminder_description TEXT NULL,
-	reminder_timestamp TIMESTAMP NULL,
+	reminder_user_entity_id INTEGER NOT NULL,
+	reminder_name VARCHAR(255) NOT NULL,
+	reminder_description TEXT NOT NULL,
+	reminder_timestamp TIMESTAMP NOT NULL CURRENT_TIMESTAMP,
 	
 	PRIMARY KEY(reminder_id),
@@ -477,6 +490,7 @@
 	anniversary_id INTEGER NOT NULL AUTO_INCREMENT,
 	anniversary_entity_id INTEGER NOT NULL,
-	anniversary_name TEXT NULL,
-	anniversary_start_date TIMESTAMP NULL,
+	anniversary_name VARVHAR(255) NOT NULL,
+	anniversary_start_date TIMESTAMP NOT NULL,
+	
 	PRIMARY KEY(anniversary_id),
 );
@@ -487,10 +501,11 @@
 	todo_list_item_reminder_id INTEGER NOT NULL,
 	todo_list_item_todo_priority_id INTEGER NOT NULL,
-	todo_list_item_user_id INTEGER NULL,
-	todo_list_item_name TEXT NULL,
-	todo_list_item_description TEXT NULL,
-	todo_list_item_done BOOL NULL,
-	todo_list_item_event_occurence_id INTEGER NULL,
-	todo_list_item_deadline TIMESTAMP NULL,
+	todo_list_item_user_entity_id INTEGER NOT NULL,
+	todo_list_item_name VARCHAR(255) NOT NULL,
+	todo_list_item_description TEXT NOT NULL,
+	todo_list_item_done BOOL NOT NULL,
+	todo_list_item_event_occurence_id INTEGER NOT NULL,
+	todo_list_item_deadline TIMESTAMP NOT NULL,
+	
 	PRIMARY KEY(todo_list_item_id),
 );
@@ -498,6 +513,7 @@
 CREATE TABLE todo_priorities (
 	todo_priority_id INTEGER NOT NULL AUTO_INCREMENT,
-	todo_priority_name TEXT NULL,
-	todo_priority_order INTEGER NULL,
+	todo_priority_name VARCHAR(255) NOT NULL,
+	todo_priority_order INTEGER NOT NULL,
+	
 	PRIMARY KEY(todo_priority_id)
 );
@@ -511,4 +527,5 @@
 	start_summer TIMESTAMP NOT NULL,
 	end_summer TIMESTAMP NOT NULL,
+	
 	PRIMARY KEY(year_id)
 );
@@ -522,5 +539,6 @@
 	rto_review_type_id INTEGER NOT NULL,
 	rto_organisation_entity_id INTEGER NOT NULL,
-	review_article_id INTEGER NULL,
+	review_article_id INTEGER NOT NULL,
+	
 	PRIMARY KEY(review_id),
 );
@@ -529,8 +547,9 @@
 	review_type_id INTEGER NOT NULL AUTO_INCREMENT,
 	review_type_article_type_id INTEGER NOT NULL,
-	review_type_image_id INTEGER NULL,
-	review_type_name TEXT NULL,
-	review_type_blurb TEXT NULL,
-	review_type_blurb_draft TEXT NULL,
+	review_type_image_id INTEGER NOT NULL,
+	review_type_name VARCHAR(255) NOT NULL,
+	review_type_blurb TEXT NOT NULL,
+	review_type_blurb_draft TEXT NOT NULL,
+	
 	PRIMARY KEY(review_type_id),
 );
@@ -542,8 +561,9 @@
 	rto_review_type_id INTEGER NOT NULL,
 	rto_live_content_id INTEGER NOT NULL,
-	rto_user_rate_count INTEGER NULL,
-	rto_average_user_rating FLOAT NULL,
-	rto_deleted BOOL NULL,
-	rto_timestamp TIMESTAMP NULL,
+	rto_user_rate_count INTEGER NOT NULL,
+	rto_average_user_rating INTEGER NOT NULL,
+	rto_deleted BOOL NOT NULL,
+	rto_timestamp TIMESTAMP NOT NULL,
+	
 	PRIMARY KEY(rto_organisation_entity_id, rto_review_type_id),
 );
@@ -554,13 +574,14 @@
 	rto_content_organisation_entity_id INTEGER NOT NULL,
 	rto_content_review_type_id INTEGER NOT NULL,
-	rto_content_cost_id INTEGER NULL,
-	rto_content_blurb TEXT NULL,
-	rto_content_price FLOAT NULL,
-	rto_content_reccommend TEXT NULL,
-	rto_content_average_price_upper INTEGER NULL,
-	rto_content_average_price_lower INTEGER NULL,
-	rto_content_rating INTEGER NULL,
-	rto_content_directions TEXT NULL,
-	rto_content_book_online BOOL NULL,
+	rto_content_cost_id INTEGER NOT NULL,
+	rto_content_blurb TEXT NOT NULL,
+	rto_content_price INTEGER NOT NULL,
+	rto_content_recommend TEXT NOT NULL,
+	rto_content_average_price_upper INTEGER NOT NULL,
+	rto_content_average_price_lower INTEGER NOT NULL,
+	rto_content_rating INTEGER NOT NULL,
+	rto_content_directions TEXT NOT NULL,
+	rto_content_book_online BOOL NOT NULL,
+	
 	PRIMARY KEY(rto_content_id),
 );
@@ -574,13 +595,16 @@
 
 CREATE TABLE rto_costs (
-	cost_id INTEGER NOT NULL AUTO_INCREMENT,
-	cost_name TEXT NULL,
+	rto_cost_id INTEGER NOT NULL AUTO_INCREMENT,
+	rto_cost_name VARCHAR(255) NOT NULL,
 	PRIMARY KEY(cost_id)
 );
 
-CREATE TABLE reviews_slideshow (
-	slideshow_review_rto_review_type_id INTEGER NOT NULL,
-	slideshow_review_rto_organisation_entity_id INTEGER NOT NULL,
-	slideshow_review_photo_id INTEGER NOT NULL,
+CREATE TABLE reviews_slideshows (
+	reviews_slideshow_rto_review_type_id INTEGER NOT NULL,
+	reviews_slideshow_rto_organisation_entity_id INTEGER NOT NULL,
+	reviews_slideshow_photo_id INTEGER NOT NULL,
+	reviews_slideshow_order INTEGER NOT NULL,
+	
+	PRIMARY_KEY(reviews_slideshow_rto_review_type_id, reviews_slideshow_rto_organisation_entity_id, reviews_slideshow_order),
 );
 
@@ -589,8 +613,9 @@
 	bar_crawl_organisation_bar_crawl_id INTEGER NOT NULL,
 	bar_crawl_organisation_organisation_entity_id INTEGER NOT NULL,
-	bar_crawl_organisation_order INTEGER NULL,
-	bar_crawl_organisation_recommend TEXT NULL,
-	bar_crawl_organisation_recommend_price FLOAT NULL,
-	PRIMARY KEY(bar_crawl_organisation_organisation_entity_id),
+	bar_crawl_organisation_order INTEGER NOT NULL,
+	bar_crawl_organisation_recommend VARCHAR(255) NOT NULL,
+	bar_crawl_organisation_recommend_price INTEGER NOT NULL,
+	
+	PRIMARY KEY(bar_crawl_organisation_bar_crawl_id, bar_crawl_organisation_organisation_entity_id),
 );
 
@@ -598,8 +623,9 @@
 CREATE TABLE leagues (
 	league_id INTEGER NOT NULL AUTO_INCREMENT,
-	league_image_id INTEGER NULL,
+	league_image_id INTEGER NOT NULL,
 	league_review_type_id INTEGER NOT NULL,
-	league_name TEXT NULL,
-	league_size INTEGER NULL,
+	league_name VARCHAR(255) NOT NULL,
+	league_size INTEGER NOT NULL,
+	
 	PRIMARY KEY(league_id),
 );
@@ -607,7 +633,8 @@
 CREATE TABLE league_entries (
 	league_entry_league_id INTEGER NOT NULL AUTO_INCREMENT,
-	rto_organisation_entity_id INTEGER NOT NULL,
-	league_entry_position INTEGER NULL,
-	PRIMARY KEY(league_entry_league_id, rto_organisation_entity_id, rto_review_type_id),
+	league_entry_rto_organisation_entity_id INTEGER NOT NULL,
+	league_entry_position INTEGER NOT NULL,
+
+	PRIMARY KEY(league_entry_league_id, league_entry_rto_organisation_entity_id),
 );
 
@@ -617,11 +644,12 @@
 CREATE TABLE campaigns (
 	campaign_id INTEGER NOT NULL AUTO_INCREMENT,
-	campaign_article_id INTEGER NULL,
-	campaign_name TEXT NULL,
-	campaign_votes INTEGER NULL,
-	campaign_petition BOOL NULL,
-	campaign_petition_signatures INTEGER NULL,
-	campaign_deleted BOOL NULL,
-	campaign_timestamp TIMESTAMP NULL,
+	campaign_article_id INTEGER NOT NULL,
+	campaign_name VARCHAR(255) NOT NULL,
+	campaign_votes INTEGER NOT NULL,
+	campaign_petition BOOL NOT NULL,
+	campaign_petition_signatures INTEGER NOT NULL,
+	campaign_deleted BOOL NOT NULL,
+	campaign_timestamp TIMESTAMP NOT NULL CURRENT_TIMESTAMP,
+
 	PRIMARY KEY(campaign_id),
 );
@@ -630,16 +658,19 @@
 CREATE TABLE campaign_users (
 	campaign_user_campaign_id INTEGER NOT NULL,
-	campaign_user_user_id INTEGER NOT NULL,
-	PRIMARY KEY(campaign_user_campaign_id, campaign_user_user_id),
+	campaign_user_user_entity_id INTEGER NOT NULL,
+	
+	PRIMARY KEY(campaign_user_campaign_id, campaign_user_user_entity_id),
 );
 
 CREATE TABLE progress_reports (
 	progress_report_id INTEGER NOT NULL AUTO_INCREMENT,
-	progress_report_charity_id INTEGER NULL,
-	progress_report_campaign_id INTEGER NULL,
-	progress_report_text TEXT NULL,
-	progress_report_order INTEGER NULL,
-	progress_report_deleted BOOL NULL,
-	progress_report_timestamp TIMESTAMP NULL,
+	progress_report_charity_id INTEGER NOT NULL,
+	progress_report_campaign_id INTEGER NOT NULL,
+	progress_report_text TEXT NOT NULL,
+	progress_report_order INTEGER NOT NULL,
+	progress_report_deleted BOOL NOT NULL,
+	progress_report_timestamp TIMESTAMP NOT NULL,
+	progress_report_good_bad ENUM('good', 'bad') NOT NULL,
+	
 	PRIMARY KEY(progress_report_id),
 );
@@ -647,9 +678,10 @@
 CREATE TABLE charities (
 	charity_id INTEGER NOT NULL AUTO_INCREMENT,
-	charity_name TEXT NULL,
-	charity_article_id INTEGER NULL,
-	charity_goal_text TEXT NULL,
-	charity_goal INTEGER NULL,
-	charity_total FLOAT NULL,
+	charity_name VARCHAR(255) NOT NULL,
+	charity_article_id INTEGER NOT NULL,
+	charity_goal_text TEXT NOT NULL,
+	charity_goal INTEGER NOT NULL,
+	charity_total FLOAT NOT NULL,
+	
 	PRIMARY KEY(charity_id)
 );
@@ -657,9 +689,10 @@
 CREATE TABLE charity_donors (
 	charity_donor_id INTEGER NOT NULL AUTO_INCREMENT,
-	charity_donor_charity_id INTEGER NULL,
-	charity_donor_name TEXT NULL,
-	charity_donor_organisation_entity_id INTEGER NULL,
-	charity_donor_amount INTEGER NULL,
-	charity_donor_timestamp TIMESTAMP NULL,
+	charity_donor_charity_id INTEGER NOT NULL,
+	charity_donor_name VARCHAR(255) NOT NULL,
+	charity_donor_organisation_entity_id INTEGER NOT NULL,
+	charity_donor_amount INTEGER NOT NULL,
+	charity_donor_timestamp TIMESTAMP NOT NULL,
+
 	PRIMARY KEY(charity_donor_id),
 );
@@ -668,8 +701,9 @@
 CREATE TABLE tags (
 	tag_id INTEGER NOT NULL AUTO_INCREMENT,
-	tag_name TEXT NULL,
-	tag_banner_name VARCHAR(255) NULL,
-	tag_archive BOOL NULL DEFAULT 0,
-	tag_deleted BOOL NULL,
+	tag_name VARCHAR(255) NOT NULL,
+	tag_banner_name VARCHAR(255) NOT NULL,
+	tag_archive BOOL NOT NULL DEFAULT 0,
+	tag_deleted BOOL NOT NULL,
+	
 	PRIMARY KEY(tag_id)
 );
@@ -678,6 +712,7 @@
 CREATE TABLE colleges (
 	college_id INTEGER NOT NULL AUTO_INCREMENT,
-	college_name CHAR NOT NULL,
-	college_ranking INTEGER NULL,
+	college_name VARCHAR(255) NOT NULL,
+	college_ranking INTEGER NOT NULL,
+	
 	PRIMARY KEY(college_id),
 );
@@ -685,13 +720,14 @@
 CREATE TABLE college_rankings (
 	college_ranking_id INTEGER NOT NULL AUTO_INCREMENT,
-	college_ranking_college_id INTEGER NULL,
+	college_ranking_college_id INTEGER NOT NULL,
 		-- Writers ID.  
-	college_ranking_user_id INTEGER NULL,
-	college_ranking_publisher_id INTEGER,
-	college_ranking_text TEXT NULL,
-	college_ranking_rank INTEGER SIGNED NULL,
-	college_ranking_published BOOL NULL,
-	college_ranking_deleted BOOL NULL,
-	college_ranking_timestamp TIMESTAMP NULL,
+	college_ranking_user_entity_id INTEGER NOT NULL,
+	college_ranking_publisher_id INTEGER NOT NULL,
+	college_ranking_text TEXT NOT NULL,
+	college_ranking_rank INTEGER SIGNED NOT NULL,
+	college_ranking_published BOOL NOT NULL,
+	college_ranking_deleted BOOL NOT NULL,
+	college_ranking_timestamp TIMESTAMP NOT NULL CURRENT_TIMESTAMP,
+
 	PRIMARY KEY(college_ranking_id),
 );
@@ -701,31 +737,35 @@
 CREATE TABLE quiz_questions (
 	quiz_question_id INTEGER NOT NULL AUTO_INCREMENT,
-	quiz_question_question TEXT NULL,
-	quiz_question_answer1 TEXT NULL,
-	quiz_question_answer2 TEXT NULL,
-	quiz_question_answer3 TEXT NULL,
-	quiz_question_answer4 TEXT NULL,
-	quiz_question_answer INTEGER NULL,
-	quiz_question_hits INTEGER NULL,
-	quiz_question_correct INTEGER NULL,
-	quiz_question_active BOOL NULL,
+	quiz_question_question TEXT NOT NULL,
+	quiz_question_answer1 TEXT NOT NULL,
+	quiz_question_answer2 TEXT NOT NULL,
+	quiz_question_answer3 TEXT NOT NULL,
+	quiz_question_answer4 TEXT NOT NULL,
+	quiz_question_answer INTEGER NOT NULL,
+	quiz_question_hits INTEGER NOT NULL,
+		-- TODO: wtf
+	quiz_question_correct INTEGER NOT NULL,
+	quiz_question_active BOOL NOT NULL,
+
 	PRIMARY KEY(quiz_question_id)
 );
 
 CREATE TABLE quiz_results (
-	quiz_result_user_id INTEGER NOT NULL AUTO_INCREMENT,
-	user_id INTEGER NOT NULL,
-	quiz_result_date TIMESTAMP NULL,
-	quiz_result_score INTEGER NULL,
-	PRIMARY KEY(quiz_result_user_id),
+	quiz_result_user_entity_id INTEGER NOT NULL,
+	quiz_result_date TIMESTAMP NOT NULL,
+	quiz_result_score INTEGER NOT NULL,
+
+	PRIMARY KEY(quiz_result_user_entity_id),
 );
 
 CREATE TABLE quiz_winners (
-	quiz_winner_id INTEGER NOT NULL AUTO_INCREMENT,
-	quiz_winner_user_id INTEGER NULL,
-	quiz_winner_score INTEGER NULL,
-	quiz_winner_date TIMESTAMP NULL,
-	PRIMARY KEY(quiz_winner_id),
-);
+	quiz_winner_user_entity_id INTEGER NOT NULL,
+	quiz_winner_score INTEGER NOT NULL,
+	quiz_winner_date TIMESTAMP NOT NULL,
+
+	PRIMARY KEY(quiz_winner_user_entity_id),
+);
+
+
 ----------------------------------------------------------------
 -- Page related tables                                        --
@@ -736,8 +776,8 @@
 	comment_page_id INTEGER NOT NULL,
 		-- The id of the content of the page e.g. article number
-	comment_subject_id INTEGER NULL,
-	comment_text TEXT NULL,
-	comment_rating INTEGER NULL,
-	comment_reported_count INTEGER NULL,
+	comment_subject_id INTEGER NOT NULL,
+	comment_text TEXT NOT NULL,
+	comment_rating INTEGER NOT NULL,
+	comment_reported_count INTEGER NOT NULL,
 	comment_deleted BOOL NOT NULL,
 	comment_timestamp TIMESTAMP NOT NULL,
@@ -748,5 +788,5 @@
 CREATE TABLE pages (
 	page_id INTEGER NOT NULL AUTO_INCREMENT,
-	page_title TEXT NOT NULL,
+	page_title VARCHAR(255) NOT NULL,
 	page_comments BOOL NOT NULL,
 	page_ratings BOOL NOT NULL,
@@ -760,9 +800,9 @@
 	page_property_property_type_id INTEGER NOT NULL,
 	page_property_page_id INTEGER NOT NULL,
-	page_property_photo_id INTEGER NULL,
-	page_property_image_id INTEGER NULL,
-	page_property_label TEXT NULL,
-	page_property_text TEXT NULL,
-	page_property_permission INTEGER NULL,
+	page_property_photo_id INTEGER NOT NULL,
+	page_property_image_id INTEGER NOT NULL,
+	page_property_label VARCHAR(255) NOT NULL,
+	page_property_text TEXT NOT NULL,
+	page_property_permission INTEGER NOT NULL,
 
 	PRIMARY KEY(page_property_id),
@@ -773,5 +813,5 @@
 	image_title VARCHAR(255) NOT NULL,
 	image_image_type_id INTEGER NOT NULL,
-	image_file_extension CHAR(4) NULL,
+	image_file_extension CHAR(4) NOT NULL,
 
 	PRIMARY KEY(image_id),
@@ -781,8 +821,9 @@
 	image_type_id INTEGER NOT NULL AUTO_INCREMENT,
 	image_type_name VARCHAR(255) NOT NULL,
-	image_type_width INTEGER NULL,
-	image_type_height INTEGER NULL,
+	image_type_width INTEGER NOT NULL,
+	image_type_height INTEGER NOT NULL,
 		-- True if all photos must have a thumb of this size
 	image_type_photo BOOL NOT NULL,
+
 	PRIMARY KEY(image_type_id)
 );
@@ -790,12 +831,13 @@
 CREATE TABLE photos (
 	photo_id INTEGER NOT NULL AUTO_INCREMENT,
-	photo_user_id INTEGER NULL,
+	photo_user_entity_id INTEGER NOT NULL,
 	photo_title VARCHAR(255) NOT NULL,
-	photo_width INTEGER NULL,
-	photo_height INTEGER NULL,
-	photo_gallery BOOL NULL,
-	photo_homepage TIMESTAMP NULL,
+	photo_width INTEGER NOT NULL,
+	photo_height INTEGER NOT NULL,
+	photo_gallery BOOL NOT NULL,
+	photo_homepage TIMESTAMP NOT NULL,
 	photo_deleted BOOL NOT NULL,
-	photo_timestamp TIMESTAMP NULL,
+	photo_timestamp TIMESTAMP NOT NULL CURRENT_TIMESTAMP,
+
 	PRIMARY KEY(photo_id),
 );
@@ -804,4 +846,5 @@
 	photo_tag_photo_id INTEGER NOT NULL,
 	photo_tag_tag_id INTEGER NOT NULL,
+	
 	PRIMARY KEY(photo_tag_photo_id, photo_tag_tag_id),
 );
@@ -813,10 +856,11 @@
 	advert_id INTEGER NOT NULL AUTO_INCREMENT,
 	advert_organisation_entity_id INTEGER NOT NULL,
-	advert_name TEXT NULL,
-	advert_description TEXT NULL,
-	advert_url TEXT NULL,
-	advert_start_date TIMESTAMP NULL,
-	advert_end_date TIMESTAMP NULL,
-	advert_max_total INTEGER NULL,
+	advert_name VARCHAR(255) NOT NULL,
+	advert_description TEXT NOT NULL,
+	advert_url VARCHAR(255) NOT NULL,
+	advert_start_date TIMESTAMP NOT NULL,
+	advert_end_date TIMESTAMP NOT NULL,
+	advert_max_total INTEGER NOT NULL,
+	
 	PRIMARY KEY(advert_id),
 );
@@ -825,7 +869,8 @@
 	advert_bill_id INTEGER NOT NULL AUTO_INCREMENT,
 	advert_bill_organisation_entity_id INTEGER NOT NULL,
-	advert_bill_total INTEGER NULL,
-	advert_bill_date TIMESTAMP NULL,
-	advert_bill_paid BOOL NULL,
+	advert_bill_total INTEGER NOT NULL,
+	advert_bill_date TIMESTAMP NOT NULL,
+	advert_bill_paid BOOL NOT NULL,
+	
 	PRIMARY KEY(advert_bill_id),
 );
@@ -833,10 +878,11 @@
 CREATE TABLE advert_bill_items (
 	advert_bill_item_id INTEGER NOT NULL AUTO_INCREMENT,
-	advert_instance_id INTEGER NOT NULL,
+	advert_bill_item_advert_instance_id INTEGER NOT NULL,
 	advert_bill_item_advert_bill_id INTEGER NOT NULL,
-	advert_bill_item_amount INTEGER NULL,
-	advert_bill_item_clicks INTEGER NULL,
-	advert_bill_item_views INTEGER NULL,
-	advert_bill_item_made_date INTEGER NULL,
+	advert_bill_item_amount INTEGER NOT NULL,
+	advert_bill_item_clicks INTEGER NOT NULL,
+	advert_bill_item_views INTEGER NOT NULL,
+	advert_bill_item_made_date INTEGER NOT NULL,
+	
 	PRIMARY KEY(advert_bill_item_id),
 );
@@ -846,11 +892,11 @@
 	advert_instance_space_type_id INTEGER NOT NULL,
 	advert_instance_advert_id INTEGER NOT NULL,
-	advert_instance_views INTEGER NULL,
-	advert_instance_clicks INTEGER NULL,
-	advert_instance_view_cost INTEGER NULL,
-	advert_instance_click_cost INTEGER NULL,
-	advert_instance_extension TEXT NULL,
-	advert_instance_deleted BOOL NULL,
-	advert_instance_timestamp TIMESTAMP NULL,
+	advert_instance_views INTEGER NOT NULL,
+	advert_instance_clicks INTEGER NOT NULL,
+	advert_instance_view_cost INTEGER NOT NULL,
+	advert_instance_click_cost INTEGER NOT NULL,
+	advert_instance_extension CHAR(4) NOT NULL,
+	advert_instance_deleted BOOL NOT NULL,
+
 	PRIMARY KEY(advert_instance_id),
 );
@@ -859,4 +905,5 @@
 	advert_related_article_advert_id INTEGER NOT NULL,
 	advert_related_article_article_id INTEGER NOT NULL,
+
 	PRIMARY KEY(advert_related_article_advert_id, advert_related_article_article_id),
 );
@@ -865,4 +912,5 @@
 	advert_related_organisation_advert_id INTEGER NOT NULL,
 	advert_related_organisation_organisation_entity_id INTEGER NOT NULL,
+
 	PRIMARY KEY(advert_related_organisation_advert_id, advert_related_organisation_organisation_entity_id),
 );
@@ -872,4 +920,5 @@
 	page_space_type_space_type_id INTEGER NOT NULL,
 	page_space_type_number INTEGER NOT NULL,
+
 	PRIMARY KEY(page_space_type_page_id, page_space_type_space_type_id, page_space_type_number),
 );
@@ -877,8 +926,9 @@
 CREATE TABLE space_types (
 	space_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	space_type_view_cost FLOAT NULL,
-	space_type_click_cost FLOAT NULL,
-	space_type_width INTEGER NULL,
-	space_type_height INTEGER NULL,
+	space_type_view_cost FLOAT NOT NULL,
+	space_type_click_cost FLOAT NOT NULL,
+	space_type_width INTEGER NOT NULL,
+	space_type_height INTEGER NOT NULL,
+
 	PRIMARY KEY(space_type_id)
 );
@@ -891,15 +941,17 @@
 CREATE TABLE buildings (
 	building_id INTEGER NOT NULL AUTO_INCREMENT,
-	buidling_name TEXT NULL,
-	building_code CHAR(5) NULL,
-	building_x INTEGER NULL,
-	buidling_y INTEGER NULL,
+	buidling_name VARCHAR(255) NOT NULL,
+	building_code CHAR(15) NOT NULL,
+	building_x INTEGER NOT NULL,
+	buidling_y INTEGER NOT NULL,
+	
 	PRIMARY KEY(building_id)
 );
 
 CREATE TABLE rooms (
-	room_id CHAR(10) NOT NULL AUTO_INCREMENT,
+	room_id CHAR(15) NOT NULL AUTO_INCREMENT,
 	room_room_building_id INTEGER NOT NULL,
 	room_type_id INTEGER NOT NULL,
+	
 	PRIMARY KEY(room_id),
 );
@@ -907,5 +959,6 @@
 CREATE TABLE room_types (
 	room_type_id INTEGER NOT NULL AUTO_INCREMENT,
-	room_type_name TEXT NULL,
+	room_type_name VARCHAR(255) NOT NULL,
+	
 	PRIMARY KEY(room_type_id)
 );
