Index: design/crosswords/crossword_uri_structure.dia
===================================================================
--- design/crosswords/crossword_uri_structure.dia (revision 85f056503972a2467ec4a9e86935252c419f1f05)
+++ design/crosswords/crossword_uri_structure.dia (revision 54a251897bc17b87337d3ac7bbcde2d8e13eaa73)
@@ -585,11 +585,11 @@
     <dia:object type="UML - Activity" version="0" id="O13">
       <dia:attribute name="obj_pos">
-        <dia:point val="-16.6225,6.055"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="-16.6225,6.055;-6.045,9.455"/>
+        <dia:point val="-16.6725,5.205"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="-16.6725,5.205;-6.095,10.205"/>
       </dia:attribute>
       <dia:attribute name="elem_corner">
-        <dia:point val="-16.6225,6.055"/>
+        <dia:point val="-16.6725,5.205"/>
       </dia:attribute>
       <dia:attribute name="elem_width">
@@ -597,5 +597,5 @@
       </dia:attribute>
       <dia:attribute name="elem_height">
-        <dia:real val="3.4000000000000004"/>
+        <dia:real val="5"/>
       </dia:attribute>
       <dia:attribute name="line_colour">
@@ -608,5 +608,7 @@
         <dia:composite type="text">
           <dia:attribute name="string">
-            <dia:string>#/office/crosswords/cats/$catid
+            <dia:string>#/office/crosswords/cats/add
+/office/crosswords/cats/$catid
+[CROSSWORD_CATEGORY_ADD]
 [CROSSWORD_CATEGORY_VIEW]
 view category information#</dia:string>
@@ -619,5 +621,5 @@
           </dia:attribute>
           <dia:attribute name="pos">
-            <dia:point val="-11.3337,7.0975"/>
+            <dia:point val="-11.3838,6.2475"/>
           </dia:attribute>
           <dia:attribute name="color">
@@ -635,11 +637,11 @@
       </dia:attribute>
       <dia:attribute name="obj_bb">
-        <dia:rectangle val="-20.6964,7.255;-16.5725,8.255"/>
+        <dia:rectangle val="-20.6964,7.205;-16.6225,8.205"/>
       </dia:attribute>
       <dia:attribute name="bez_points">
         <dia:point val="-20.645,7.64"/>
         <dia:point val="-18.8475,7.59"/>
-        <dia:point val="-18.0169,7.755"/>
-        <dia:point val="-16.6225,7.755"/>
+        <dia:point val="-18.0669,7.705"/>
+        <dia:point val="-16.6725,7.705"/>
       </dia:attribute>
       <dia:attribute name="corner_types">
@@ -710,12 +712,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O16">
       <dia:attribute name="obj_pos">
-        <dia:point val="-6.045,7.755"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="-6.09637,7.255;-2.2725,8.255"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="-6.045,7.755"/>
-        <dia:point val="-4.2475,7.705"/>
+        <dia:point val="-6.095,7.705"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="-6.14637,7.255;-2.2725,8.255"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="-6.095,7.705"/>
+        <dia:point val="-4.2975,7.655"/>
         <dia:point val="-3.7169,7.755"/>
         <dia:point val="-2.3225,7.755"/>
@@ -921,12 +923,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O21">
       <dia:attribute name="obj_pos">
-        <dia:point val="-6.045,7.755"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="-6.50054,3.16835;11.2948,8.0531"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="-6.045,7.755"/>
-        <dia:point val="-2.87572,2.4125"/>
+        <dia:point val="-6.095,7.705"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="-6.55054,3.16834;11.2948,8.0031"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="-6.095,7.705"/>
+        <dia:point val="-2.92572,2.3625"/>
         <dia:point val="10.2502,6.49645"/>
         <dia:point val="10.7935,3.30145"/>
@@ -1569,12 +1571,12 @@
     <dia:object type="Standard - BezierLine" version="0" id="O37">
       <dia:attribute name="obj_pos">
-        <dia:point val="-11.3338,6.055"/>
-      </dia:attribute>
-      <dia:attribute name="obj_bb">
-        <dia:rectangle val="-11.8362,-0.570222;-10.3119,6.14622"/>
-      </dia:attribute>
-      <dia:attribute name="bez_points">
-        <dia:point val="-11.3338,6.055"/>
-        <dia:point val="-11.1132,3.4"/>
+        <dia:point val="-11.3838,5.205"/>
+      </dia:attribute>
+      <dia:attribute name="obj_bb">
+        <dia:rectangle val="-11.8862,-0.570213;-10.312,5.29623"/>
+      </dia:attribute>
+      <dia:attribute name="bez_points">
+        <dia:point val="-11.3838,5.205"/>
+        <dia:point val="-11.1632,2.55"/>
         <dia:point val="-10.8132,2.3"/>
         <dia:point val="-10.812,-0.52"/>
