Index: system/application/controllers/roses.php
===================================================================
--- system/application/controllers/roses.php (revision 85213558b1ffb0e08df588f5f8770e52c99aaceb)
+++ system/application/controllers/roses.php (revision bab715cff4290252863e22b5da69488dcbc2d973)
@@ -22,5 +22,5 @@
 
 		// Obtain banner for homepage
-		$data['banner'] = $this->Home_Model->GetBannerImageForHomepage('sport');
+		$data['banner'] = $this->home_model->GetBannerImageForHomepage('sport');
 
 		$this->pages_model->SetPageCode('homepage_roses');
