Index: system/application/controllers/comments.php
===================================================================
--- system/application/controllers/comments.php (revision a4617064a5efd277843180b742af9d6b5a4764fb)
+++ system/application/controllers/comments.php (revision 5c4eac57305c24b1f90e53d3f33da0ffeccd0dd4)
@@ -405,7 +405,9 @@
 						'.implode(', ', $selects).'
 					FROM '.$data['table'];
-				foreach ($data['joins'] as $join) {
-					$sql .= '
-					INNER JOIN '.$join['table'].' ON '.$join['on'];
+				if (isset($data['joins'])) {
+					foreach ($data['joins'] as $join) {
+						$sql .= '
+						INNER JOIN '.$join['table'].' ON '.$join['on'];
+					}
 				}
 				$sql .= '
