Index: export/template/public_html/podcast
===================================================================
--- export/template/public_html/podcast (revision 48150e4b30a6dea537d8e6d0a386cece30b69a65)
+++  (revision )
@@ -1,1 +1,0 @@
-../../podcast_html
Index: export/template/public_html/static
===================================================================
--- export/template/public_html/static (revision 6472acaad2ef8fb03162d98305e9e519e0c19596)
+++ export/template/public_html/static (revision 6472acaad2ef8fb03162d98305e9e519e0c19596)
@@ -0,0 +1,1 @@
+link ../../static_html
