Index: system/application/models/roses_model.php
===================================================================
--- system/application/models/roses_model.php (revision 882b087a2c4fa7f8b0e5a143077fa946f8d04f04)
+++ system/application/models/roses_model.php (revision afe4f7a37ff710e602ebe04648a067b0676c5ea6)
@@ -25,5 +25,5 @@
 					event_lancaster_score,
 					event_york_score,
-					UNIX_TIMESTAMP(event_score_time) AS event_score_time
+					event_score_time AS event_score_time
 			FROM		roses_scores
 			ORDER BY	event_time ASC,
Index: system/application/views/office/liveblog/scores.php
===================================================================
--- system/application/views/office/liveblog/scores.php (revision 882b087a2c4fa7f8b0e5a143077fa946f8d04f04)
+++ system/application/views/office/liveblog/scores.php (revision afe4f7a37ff710e602ebe04648a067b0676c5ea6)
@@ -1,7 +1,17 @@
 <form action="/office/liveblog/scores" method="post">
-	<table>
+	<p><a href="#" onclick="document.getElementById('waiting').className='hide';document.getElementById('done').className='';return false;">SEE COMPLETED EVENTS</a></p>
+	<table id="waiting">
 		<tr>
+			<th>Sport</th>
+			<th>Event</th>
+			<th>Venue</th>
+			<th>Points</th>
+			<th>York</th>
+			<th>Lancs</th>
 		</tr>
-		<?php foreach ($allevents as $event) { ?>
+		<?php
+		foreach ($allevents as $event) {
+			if (empty($event['event_score_time'])) {
+		?>
 		<tr>
 			<td><?php echo($event['event_sport']); ?></td>
@@ -13,5 +23,34 @@
 			<td><input type="submit" name="updatescore" value="<?php echo($event['event_id']); ?>" /></td>
 		</tr>
-		<?php } ?>
+		<?php
+			}
+		} ?>
+	</table>
+
+	<table id="done" class="hide">
+		<tr>
+			<th>Sport</th>
+			<th>Event</th>
+			<th>Venue</th>
+			<th>Points</th>
+			<th>York</th>
+			<th>Lancs</th>
+		</tr>
+		<?php
+		foreach ($allevents as $event) {
+			if (!empty($event['event_score_time'])) {
+		?>
+		<tr>
+			<td><?php echo($event['event_sport']); ?></td>
+			<td><?php echo($event['event_name']); ?></td>
+			<td><?php echo($event['event_venue']); ?></td>
+			<td><?php echo($event['event_points']); ?></td>
+			<td><input type="text" name="yscore<?php echo($event['event_id']); ?>" value="<?php echo(isset($valid[$event['event_york_score']]) ? $valid[$event['event_york_score']] : $event['event_york_score']); ?>" size="3" /></td>
+			<td><input type="text" name="lscore<?php echo($event['event_id']); ?>" value="<?php echo(isset($valid[$event['event_lancaster_score']]) ? $valid[$event['event_lancaster_score']] : $event['event_lancaster_score']); ?>" size="3" /></td>
+			<td><input type="submit" name="updatescore" value="<?php echo($event['event_id']); ?>" /></td>
+		</tr>
+		<?php
+			}
+		} ?>
 	</table>
 </form>
