Index: system/application/views/frames/public_frame.php
===================================================================
--- system/application/views/frames/public_frame.php (revision ed47f54b421284966d39e2bef38df3a5ca6156a2)
+++ system/application/views/frames/public_frame.php (revision cf37f5dc38a93346640165520d0f22630f383e97)
@@ -121,5 +121,6 @@
 	array('sport', '/sport', array()),
 	array('arts', '/arts', array()),
-	array('lifestyle', '/lifestyle', array('last')),
+	array('lifestyle', '/lifestyle', array()),
+	array('roses', '/roses', array('last')),
 );
 // Has a tab been set as selected?
