Index: public_html/stylesheets/crosswords.css
===================================================================
--- public_html/stylesheets/crosswords.css (revision e40fb99f8646d9019648e7c2c173f3d79d958f6c)
+++ public_html/stylesheets/crosswords.css (revision d7c68673e80bba6201f47d497e8246ee0ac33200)
@@ -177,4 +177,8 @@
 	width: 33%;
 }
+/* e.g. quick clues radio when quick clues disabled */
+div.crosswordCluesHeader fieldset.undesired {
+	color: #888888;
+}
 div.crosswordCluesHeader fieldset label {
 	float: right;
Index: system/application/controllers/crosswords.php
===================================================================
--- system/application/controllers/crosswords.php (revision 491bfe1f7619eec30370e3b588656e16de9660dc)
+++ system/application/controllers/crosswords.php (revision d7c68673e80bba6201f47d497e8246ee0ac33200)
@@ -131,4 +131,5 @@
 		if (null === $operation || 'view' === $operation) {
 			$crosswordView = new CrosswordView($puzzle);
+			$crosswordView->setClueTypes($crossword['has_quick_clues'], $crossword['has_cryptic_clues']);
 			if (!$loggedIn) {
 				$crosswordView->setReadOnly(true);
Index: system/application/controllers/office/crosswords.php
===================================================================
--- system/application/controllers/office/crosswords.php (revision f78347eed0783cbbc00aa2b99aff08c70244510e)
+++ system/application/controllers/office/crosswords.php (revision d7c68673e80bba6201f47d497e8246ee0ac33200)
@@ -653,4 +653,7 @@
 				}
 
+				// Which clues are enabled may have just changed
+				$crosswordView->setClueTypes($crossword_info['has_quick_clues'], $crossword_info['has_cryptic_clues']);
+
 				$data['Configuration'] = &$config;
 				$data['Grid'] = &$crosswordView;
Index: system/application/helpers/crosswords_helper.php
===================================================================
--- system/application/helpers/crosswords_helper.php (revision e40fb99f8646d9019648e7c2c173f3d79d958f6c)
+++ system/application/helpers/crosswords_helper.php (revision d7c68673e80bba6201f47d497e8246ee0ac33200)
@@ -639,4 +639,8 @@
 	private $m_edit;
 	private $m_readonly;
+	private $m_allowToggleClueType;
+	private $m_cluesQuick;
+	private $m_cluesCryptic;
+	private $m_defaultCryptic;
 
 	function __construct(&$crossword, $edit = false)
@@ -644,4 +648,10 @@
 		$this->m_crossword = $crossword;
 		$this->m_edit = $edit;
+		$this->m_readonly = false;
+		// so that the hidden clues aren't lost when saved
+		$this->m_allowToggleClueType = $edit;
+		$this->m_cluesQuick = true;
+		$this->m_cluesCryptic = false;
+		$this->m_defaultCryptic = false;
 
 		// Set up stylesheets and javascript
@@ -664,4 +674,13 @@
 	{
 		$this->m_readonly = $readonly;
+	}
+
+	function setClueTypes($quick, $cryptic)
+	{
+		// or edit so that the hidden clues aren't lost when saved
+		$this->m_allowToggleClueType = $this->m_edit || ($quick && $cryptic);
+		$this->m_cluesQuick = $quick;
+		$this->m_cluesCryptic = $cryptic;
+		$this->m_defaultCryptic = $cryptic;
 	}
 
@@ -780,6 +799,10 @@
 				}
 				// Choice between quick and cryptic clues
-				if (true) {
-					?><fieldset><?php
+				if ($this->m_allowToggleClueType) {
+					?><fieldset<?php
+							if (!$this->m_cluesQuick) {
+								?> class="undesired"<?php
+							}
+							?>><?php
 						?><label	for="<?php echo("$name-clues-show-quick"); ?>">show quick clues</label><?php
 						?><input	id="<?php echo("$name-clues-show-quick"); ?>"<?php
@@ -789,8 +812,14 @@
 								?>	value="quick"<?php
 								?>	onchange="<?php echo(xml_escape("return crosswordClueTypeUpdated('$name');")); ?>"<?php
+							if (!$this->m_defaultCryptic) {
 								?>	checked="checked"<?php
+							}
 								?>	/><?php
 					?></fieldset><?php
-					?><fieldset><?php
+					?><fieldset<?php
+							if (!$this->m_cluesCryptic) {
+								?> class="undesired"<?php
+							}
+							?>><?php
 						?><label	for="<?php echo("$name-clues-show-cryptic"); ?>">show cryptic clues</label><?php
 						?><input	id="<?php echo("$name-clues-show-cryptic"); ?>"<?php
@@ -800,4 +829,7 @@
 								?>	value="cryptic"<?php
 								?>	onchange="<?php echo(xml_escape("return crosswordClueTypeUpdated('$name');")); ?>"<?php
+							if ($this->m_defaultCryptic) {
+								?>	checked="checked"<?php
+							}
 								?>	/><?php
 					?></fieldset><?php
@@ -812,4 +844,8 @@
 		$orClasses = array(CrosswordGrid::$HORIZONTAL => "horizontal",
 		                   CrosswordGrid::$VERTICAL   => "vertical");
+		$commonClasses = 'hideValues';
+		if ($this->m_allowToggleClueType) {
+			$commonClasses .= ' '.($this->m_defaultCryptic ? 'hideQuick' : 'hideCryptic');
+		}
 		$dx = array(CrosswordGrid::$HORIZONTAL	=> 1,
 					CrosswordGrid::$VERTICAL	=> 0);
@@ -817,6 +853,8 @@
 					CrosswordGrid::$VERTICAL	=> 1);
 		foreach ($clues as $orientation => &$oclues) {
-			?><div class="crosswordCluesBox"><?php
-			?><div id="<?php echo("$name-$orientation-clues"); ?>" class="<?php echo($orClasses[$orientation]); ?> hideValues hideCryptic"><?php
+			?><div	class="crosswordCluesBox"><?php
+			?><div	id="<?php echo("$name-$orientation-clues"); ?>"<?php
+				?>	class="<?php echo($orClasses[$orientation].' '.$commonClasses); ?>"<?php
+				?>><?php
 			?><h2><?php
 			echo(xml_escape($titles[$orientation]));
@@ -838,10 +876,14 @@
 					echo(' ');
 
-					?><span class="quickClue" id="<?php echo("$name-$orientation-cluetext0-$x-$y"); ?>"><?php
-					echo(xml_escape($clue->quickClue()));
-					?></span><?php
-					?><span class="crypticClue" id="<?php echo("$name-$orientation-cluetext1-$x-$y"); ?>"><?php
-					echo(xml_escape($clue->crypticClue()));
-					?></span><?php
+					if ($this->m_allowToggleClueType || !$this->m_defaultCryptic) {
+						?><span class="quickClue" id="<?php echo("$name-$orientation-cluetext0-$x-$y"); ?>"><?php
+						echo(xml_escape($clue->quickClue()));
+						?></span><?php
+					}
+					if ($this->m_allowToggleClueType || $this->m_defaultCryptic) {
+						?><span class="crypticClue" id="<?php echo("$name-$orientation-cluetext1-$x-$y"); ?>"><?php
+						echo(xml_escape($clue->crypticClue()));
+						?></span><?php
+					}
 
 					$lengths = $clue->wordLengths();
