Index: system/application/views/homepages/roses.php
===================================================================
--- system/application/views/homepages/roses.php (revision ab9d8b7bcfd976ba7b7dae73ee9130c7165896f7)
+++ system/application/views/homepages/roses.php (revision 3b47ba93be6804427c6bc742f6cc3f247b8c4bab)
@@ -19,6 +19,6 @@
 	foreach ($latest as $l) {
 		echo('<div>');
-		$cache = str_replace('//medium//', '/small/', $l['cache']);
-		$cache = str_replace('//large//', '/small/', $cache);
+		$cache = str_replace('/medium/', '/small/', $l['cache']);
+		$cache = str_replace('/large/', '/small/', $cache);
 		echo($cache);
 		echo('</div>');
@@ -74,5 +74,5 @@
 				<td>
 					<?php
-					if ($event['event_york_score'] > 0 && $event['event_lancaster_score'] > 0) {
+					if ($event['event_york_score'] > 0 || $event['event_lancaster_score'] > 0) {
 						echo(xml_escape($event['event_york_score'] . ' - ' . $event['event_lancaster_score']));
 					} ?>
