Index: public_html/javascript/crossword.js
===================================================================
--- public_html/javascript/crossword.js (revision 908f094a17624cda0bb96cc9b3634236ccd83c6e)
+++ public_html/javascript/crossword.js (revision 1a5bf6b3d8be0e1f7f33e9052f188c077c7a687c)
@@ -355,4 +355,5 @@
 		var keyCode = e.keyCode;
 		var charCode = e.charCode;
+		alert("keyCode: " + keyCode + ", charCode: " + e.charCode + ", which: " + e.which);
 		// Character keys
 		if (charCode != 0) {
