Index: system/application/views/frames/public_frame.php
===================================================================
--- system/application/views/frames/public_frame.php (revision cf37f5dc38a93346640165520d0f22630f383e97)
+++ system/application/views/frames/public_frame.php (revision bab8e2b098716262ea99656180e109f393ef7ccd)
@@ -117,5 +117,5 @@
 <?php
 $menu = array(
-	array('home', '/', array('first')),
+	array('home', '/home', array('first')),
 	array('news', '/uninews', array()),
 	array('sport', '/sport', array()),
