Index: system/application/views/crosswords/office/crossword_edit.php
===================================================================
--- system/application/views/crosswords/office/crossword_edit.php (revision e08771b67b2c1f1fc67bee3ba544d1971abb2226)
+++ system/application/views/crosswords/office/crossword_edit.php (revision 304631e64ddc1834d2c3eff103e9ca06923e6e24)
@@ -40,12 +40,4 @@
 <div class="BlueBox">
 
-	<h2>tips</h2>
-
-	<?php $Tips->Load(); ?>
-
-</div>
-
-<div class="BlueBox">
-
 	<h2>edit crossword</h2>
 
@@ -80,2 +72,10 @@
 
 </div>
+
+<div class="BlueBox">
+
+	<h2>tips</h2>
+
+	<?php $Tips->Load(); ?>
+
+</div>
