Index: system/application/views/office/liveblog/admin.php
===================================================================
--- system/application/views/office/liveblog/admin.php (revision 1b9b919ad8c49533c61400905c4aa561e682db5c)
+++ system/application/views/office/liveblog/admin.php (revision a8af65fc0bb0192c4381e9f55b7015b3f81b1ecf)
@@ -11,6 +11,6 @@
 	if ($article_id == 1682) echo('TEST');
 	elseif ($article_id == 2965) echo('FRIDAY');
-	elseif ($article_id == 1691) echo('SATURDAY');
-	elseif ($article_id == 1692) echo('SUNDAY');
+	elseif ($article_id == 2971) echo('SATURDAY');
+	elseif ($article_id == 2972) echo('SUNDAY');
 	elseif ($article_id == 2858) echo('Elections 2009');
 	else echo('UNKNOWN');
@@ -20,7 +20,7 @@
 		<a href="/office/liveblog/admin/1682">Test Article</a> |
 		<a href="/office/liveblog/admin/2965">FRIDAY</a> |
+		<a href="/office/liveblog/admin/2971">SATURDAY</a> |
+		<a href="/office/liveblog/admin/2972">SUNDAY</a>
 		<!--
-		<a href="/office/liveblog/admin/1691">SATURDAY</a> |
-		<a href="/office/liveblog/admin/1692">SUNDAY</a>
 		<a href="/office/liveblog/admin/2858">Elections 2009</a>
 		-->
