Index: system/application/helpers/crosswords_helper.php
===================================================================
--- system/application/helpers/crosswords_helper.php (revision 291a336a9f82da59fb8c9906167fb1306f1ac411)
+++ system/application/helpers/crosswords_helper.php (revision 9be2e497c8bc987c42963d2a517c6b8d7bc67067)
@@ -765,6 +765,6 @@
 
 		// List of clues
-		$titles = array(CrosswordGrid::$HORIZONTAL => "Across",
-		                CrosswordGrid::$VERTICAL   => "Down");
+		$titles = array(CrosswordGrid::$HORIZONTAL => "across",
+		                CrosswordGrid::$VERTICAL   => "down");
 		$orClasses = array(CrosswordGrid::$HORIZONTAL => "horizontal",
 		                   CrosswordGrid::$VERTICAL   => "vertical");
