Index: system/application/views/crosswords/category.php
===================================================================
--- system/application/views/crosswords/category.php (revision 6e2c86c9a08b0e018aa9918ae55993be5e7d4037)
+++ system/application/views/crosswords/category.php (revision 358942de07923f8df38e6461fcecf7d77da13b19)
@@ -25,5 +25,5 @@
 		?><div class="crossword_preview"><?php
 			?><a href="<?php echo(site_url('crosswords/'.$crossword['id'])); ?>"><?php
-				?><img src="<?php echo(site_url('images/crosswords/xw.png')); ?>" /><?php
+				?><img alt="" src="<?php echo(site_url('images/crosswords/xw.png')); ?>" /><?php
 			?></a><?php
 		?></div><?php
Index: system/application/views/crosswords/index.php
===================================================================
--- system/application/views/crosswords/index.php (revision 6e2c86c9a08b0e018aa9918ae55993be5e7d4037)
+++ system/application/views/crosswords/index.php (revision 358942de07923f8df38e6461fcecf7d77da13b19)
@@ -41,5 +41,5 @@
 				?><div class="crossword_preview"><?php
 					?><a href="<?php echo(site_url('crosswords/'.$crossword['id'])); ?>"><?php
-						?><img src="<?php echo(site_url('images/crosswords/xw.png')); ?>" /><?php
+						?><img alt="" src="<?php echo(site_url('images/crosswords/xw.png')); ?>" /><?php
 					?></a><?php
 				?></div><?php
