Index: system/application/controllers/crossword.php
===================================================================
--- system/application/controllers/crossword.php (revision 7ddfd6a4191cadf346599116a52bd283fb6842d7)
+++ system/application/controllers/crossword.php (revision 89a451b38135f6648271f9dd41aa1d449afd12c7)
@@ -21,5 +21,5 @@
 		$crossword->addSlot(7, 0, CrosswordGrid::$HORIZONTAL, new CrosswordClue("Abyss", "chasm"));
 		$crossword->addSlot(0, 2, CrosswordGrid::$HORIZONTAL, new CrosswordClue("Applaud with shouts", "cheer"));
-		$crossword->addSlot(6, 2, CrosswordGrid::$HORIZONTAL, new CrosswordClue("Destroy completely", "wipeout"));
+		$crossword->addSlot(6, 2, CrosswordGrid::$HORIZONTAL, new CrosswordClue("Destroy completely", "wipe out"));
 		$crossword->addSlot(0, 4, CrosswordGrid::$HORIZONTAL, new CrosswordClue("Small frozen blocks", "ice cubes"));
 		$crossword->addSlot(9, 4, CrosswordGrid::$HORIZONTAL, new CrosswordClue("Slight quarrel", "spat"));
