Index: system/application/controllers/crosswords.php
===================================================================
--- system/application/controllers/crosswords.php (revision 010b643fd0bb13d46ef4b2a3c414ca4bcebdbe84)
+++ system/application/controllers/crosswords.php (revision 60dade23c47e514e19f2c5f41fd6e556079a9f7a)
@@ -135,5 +135,5 @@
 			$crosswordView->setClueTypes($crossword['has_quick_clues'], $crossword['has_cryptic_clues']);
 			if (!$loggedIn) {
-				$crosswordView->setReadOnly(true);
+				$crosswordView->setReadOnly(!$crossword['expired']);
 			}
 			else {
Index: system/application/views/crosswords/crossword.php
===================================================================
--- system/application/views/crosswords/crossword.php (revision 010b643fd0bb13d46ef4b2a3c414ca4bcebdbe84)
+++ system/application/views/crosswords/crossword.php (revision 60dade23c47e514e19f2c5f41fd6e556079a9f7a)
@@ -72,8 +72,16 @@
 		?>
 		<div>
-			<p><b>You must be logged in to do crosswords online.</b>
-			Please <a href="<?php echo(xml_escape($login_url)); ?>">log in now</a> to take advantage of the following features:</p>
+			<?php if (!$Crossword['expired']) { ?>
+			<p>
+				<b>You must be logged in to do this crossword online</b>,
+				otherwise you will have to wait until all the winner positions have been filled.
+			<p>
+			<?php } ?>
+			<p>
+				The Yorker provides an intuitive and interactive online interface for doing crosswords.
+				Please <a href="<?php echo(xml_escape($login_url)); ?>">log in</a> to take advantage of the following features:
+			</p>
 			<ul>
-				<li>Intuitive and interactive online interface.</li>
+				<li>Do crosswords online before winner positions have been filled.</li>
 				<li>List your name on the crossword's winners list if you are one of the first to complete it.</li>
 				<li>Save unfinished crosswords for later completion.</li>
