Index: system/application/controllers/crosswords.php
===================================================================
--- system/application/controllers/crosswords.php (revision 40e513f7e9cd1913a82d5fff4a90736f5308f0ec)
+++ system/application/controllers/crosswords.php (revision b19b1ad4123afe63256d15423775471be505037e)
@@ -23,4 +23,5 @@
 		// "tips"
 		if ('tips' === $arg) {
+			array_shift($args);
 			return call_user_func_array(array(&$this, 'tips'), $args);
 		}
@@ -277,7 +278,34 @@
 	}
 
-	function tips()
-	{
-		if (!CheckPermissions('public')) return;
+	function tips($category = null)
+	{
+		if (!CheckPermissions('public')) return;
+
+		if (null === $category) {
+		}
+		else {
+			$category_info = null;
+			if (is_numeric($category)) {
+				$category_info = $this->crosswords_model->GetTipCategories((int)$category);
+				if (empty($category_info)) {
+					$category_info = null;
+				}
+				else {
+					$category_info = $category_info[0];
+				}
+			}
+
+			if (null === $category_info) {
+				show_404();
+			}
+
+			$data = array(
+				'Category' => $category_info,
+				'Tips' => new CrosswordTipsList($category_info['id'], null),
+				'PostAction' => $this->uri->uri_string(),
+			);
+
+			$this->main_frame->setContentSimple('crosswords/tip_cat_view', $data);
+		}
 		$this->main_frame->Load();
 	}
Index: system/application/views/crosswords/office/tip_cat_edit.php
===================================================================
--- system/application/views/crosswords/office/tip_cat_edit.php (revision e08771b67b2c1f1fc67bee3ba544d1971abb2226)
+++ system/application/views/crosswords/office/tip_cat_edit.php (revision b19b1ad4123afe63256d15423775471be505037e)
@@ -1,5 +1,5 @@
 <?php
 /**
- * @file views/crosswords/office/tip_edit.php
+ * @file views/crosswords/office/tip_cat_edit.php
  * @param $Permissions array[string => bool]
  * @param $Form InputInterfaces
Index: system/application/views/crosswords/office/tip_cat_view.php
===================================================================
--- system/application/views/crosswords/office/tip_cat_view.php (revision e08771b67b2c1f1fc67bee3ba544d1971abb2226)
+++ system/application/views/crosswords/office/tip_cat_view.php (revision b19b1ad4123afe63256d15423775471be505037e)
@@ -1,5 +1,5 @@
 <?php
 /**
- * @file views/crosswords/office/tip_view.php
+ * @file views/crosswords/office/tip_cat_view.php
  * @param $Permissions array[string => bool]
  * @param $Category array of tip category data
@@ -26,5 +26,5 @@
 
 <div class="BlueBox">
-	<h2>Tips</h2>
+	<h2><?php echo(xml_escape($Category['name'])); ?></h2>
 
 	<?php $Tips->Load(); ?>
Index: system/application/views/crosswords/tip_cat_view.php
===================================================================
--- system/application/views/crosswords/tip_cat_view.php (revision b19b1ad4123afe63256d15423775471be505037e)
+++ system/application/views/crosswords/tip_cat_view.php (revision b19b1ad4123afe63256d15423775471be505037e)
@@ -0,0 +1,17 @@
+<?php
+/**
+ * @file views/crosswords/tip_cat_view.php
+ * @param $Category array of tip category data
+ *	- id
+ *  - name
+ *  - description
+ * @param $Tips CrosswordTipsList
+ * @param $PostAction
+ */
+?>
+
+<div class="BlueBox">
+	<h2><?php echo(xml_escape($Category['name'])); ?></h2>
+
+	<?php $Tips->Load(); ?>
+</div>
