Index: system/application/controllers/crosswords.php
===================================================================
--- system/application/controllers/crosswords.php (revision ed27e5fee7948b4eee7af0985fa6978e51f74270)
+++ system/application/controllers/crosswords.php (revision 624f42e571d833e6365717592f4bd226a0e466ed)
@@ -43,5 +43,5 @@
 		if (!CheckPermissions('public')) return;
 
-		$this->main_frame->SetFeedTitle('The Yorker - Crosswords');
+		$this->main_frame->SetFeedTitle('The Yorker Crosswords');
 		if (FeedOutputMode()) {
 			$this->main_frame->Channel()->SetDescription('All crosswords.');
@@ -72,4 +72,6 @@
 			$this->main_frame->IncludeCss('stylesheets/crosswords_index.css');
 			$this->main_frame->SetContentSimple('crosswords/index', $data);
+			/// @todo Use pages interface
+			$this->main_frame->SetTitle('Crosswords');
 		}
 		$this->main_frame->Load();
@@ -86,5 +88,5 @@
 		if (!CheckPermissions('public')) return;
 
-		$this->main_frame->SetFeedTitle('The Yorker - Crosswords - '.$category['name']);
+		$this->main_frame->SetFeedTitle('The Yorker '.$category['name']);
 		if (FeedOutputMode()) {
 			$this->main_frame->Channel()->SetDescription('All crosswords in category "'.$category['name'].'".');
@@ -103,4 +105,6 @@
 			$this->main_frame->IncludeCss('stylesheets/crosswords_index.css');
 			$this->main_frame->SetContentSimple('crosswords/category', $data);
+			/// @todo Use pages interface
+			$this->main_frame->SetTitle($category['name']);
 		}
 		elseif ($arg2 == 'archive') {
@@ -116,4 +120,6 @@
 			$this->main_frame->IncludeCss('stylesheets/crosswords_index.css');
 			$this->main_frame->SetContentSimple('crosswords/category', $data);
+			/// @todo Use pages interface
+			$this->main_frame->SetTitle($category['name'].' Archive');
 		}
 		else {
@@ -187,4 +193,7 @@
 
 			$this->main_frame->SetContentSimple('crosswords/crossword', $data);
+			/// @todo Use pages interface
+			$pub = new Academic_time($crossword['publication']);
+			$this->main_frame->SetTitle('Crossword '.$pub->Format('D ').$pub->AcademicTermNameUnique().' week '.$pub->AcademicWeek());
 		}
 		elseif ($operation == 'ajax') {
@@ -312,5 +321,5 @@
 				'SelfUri' => $this->uri->uri_string(),
 			);
-			$this->main_frame->SetFeedTitle('The Yorker - Crossword Tips');
+			$this->main_frame->SetFeedTitle('The Yorker Crossword Tips');
 			if (FeedOutputMode()) {
 				$this->main_frame->Channel()->SetDescription('All crossword tips.');
@@ -319,4 +328,6 @@
 			else {
 				$this->main_frame->setContentSimple('crosswords/tips', $data);
+				/// @todo Use pages interface
+				$this->main_frame->SetTitle('Crossword Tips');
 			}
 		}
@@ -337,5 +348,5 @@
 			}
 
-			$this->main_frame->SetFeedTitle('The Yorker - Crossword Tips - '.$category_info['name']);
+			$this->main_frame->SetFeedTitle('The Yorker Crossword Tips - '.$category_info['name']);
 			if (FeedOutputMode()) {
 				$this->main_frame->Channel()->SetDescription('All crossword tips in category "'.$category_info['name'].'".');
@@ -350,4 +361,6 @@
 
 				$this->main_frame->setContentSimple('crosswords/tip_cat_view', $data);
+				/// @todo Use pages interface
+				$this->main_frame->SetTitle('Crossword Tips - '.$category_info['name']);
 			}
 		}
