Index: system/application/controllers/office/index.php
===================================================================
--- system/application/controllers/office/index.php (revision 0819008547355253c9a66f068ea39c91f6a747ce)
+++ system/application/controllers/office/index.php (revision d38c637f4ffd74ce5a5318aa1cbd96e3fdacb4a0)
@@ -1,21 +1,77 @@
 <?php
 
-/// Main office controller.
+/**
+ *	@brief	Office homepage 2.0
+ *	@author	Chris Travis (cdt502 - ctravis@gmail.com)
+ */
+
 class Index extends Controller
 {
-	/**
-	 * @brief Default constructor.
-	 */
+
 	function __construct()
 	{
 		parent::Controller();
 		$this->load->model('notifications_model');
-		/// Load requests office model
+
 		$this->load->model('requests_model');
-		/// Load photos office model
 		$this->load->model('photos_model');
 	}
 
 	function index()
+	{
+		if (!CheckPermissions('office')) return;
+		$this->pages_model->SetPageCode('office_index');
+		$data = array();
+
+		// Notifications
+		$data['notifications'] = array(
+			array(
+				'permission'	=>	'',
+				'query'			=>	'checkAnnouncements',
+				'title'			=>	'Announcement',
+				'type'			=>	'announcement',
+				'link'			=>	'/office/announcements'
+			),
+			array(
+				'permission'	=>	'BYLINES_PENDING',
+				'query'			=>	'checkPendingBylines',
+				'title'			=>	'Pending byline',
+				'type'			=>	'byline',
+				'link'			=>	'/office/bylines/pending'
+			),
+		);
+
+		foreach ($data['notifications'] as &$n) {
+			if (empty($n['permission']) || $this->permissions_model->hasUserPermission($n['permission'])) {
+				$n['count'] = $this->notifications_model->$n['query']();
+			}
+		}
+
+		// Activity
+		$data['activity'] = $this->notifications_model->getActivity();
+
+
+
+		//$article_requests = $this->requests_model->GetMyRequests($this->user_auth->entityId);
+		//$photos_requests = $this->photos_model->GetMyRequests($this->user_auth->entityId);
+
+		//$all_requests = array();
+		//while ((count($article_requests) > 0) || (count($photos_requests) > 0)) {
+		//	if ((count($article_requests) > 0) && ((count($photos_requests) == 0) || ($article_requests[0]['deadline'] < $photos_requests[0]['deadline']))) {
+		//		$all_requests[] = array_shift($article_requests);
+		//	} else {
+		//		$all_requests[] = array_shift($photos_requests);
+		//	}
+		//}
+
+		//$data['announcements'] = $this->notifications_model->getAnnouncements();
+		//$data['my_requests'] = $all_requests;
+
+		// Set up the content
+		$this->main_frame->SetContentSimple('office/index', $data);
+		$this->main_frame->Load();
+	}
+
+	function index_old()
 	{
 		if (!CheckPermissions('office')) return;
Index: system/application/views/office/index.php
===================================================================
--- system/application/views/office/index.php (revision 33c60727943a898af8f647cc27ab52ff2dd8895d)
+++ system/application/views/office/index.php (revision d38c637f4ffd74ce5a5318aa1cbd96e3fdacb4a0)
@@ -1,2 +1,52 @@
+<style type="text/css">
+div.activity p {
+	display: inline;
+}
+</style>
+
+<div class="BlueBox Box23b">
+	<h2>recent activity</h2>
+	
+	<div class="activity">
+<?php
+foreach ($activity as $a) {
+	echo('<div>');
+	switch ($a->type) {
+		case 'byline':
+		case 'byline-teams':
+			echo($a->user_name . ' ' . $a->wikitext);
+			break;
+		case 'announcement':
+			echo($a->user_name . ' posted an announcement titled "<a href="/office/announcements">' . $a->subject . '</a>".');
+			break;
+		default:
+			echo('ERROR: Unknown activity type (' . $a->type . ')');
+	}
+	echo('</div>');
+}
+?>
+	</div>
+
+	<pre>
+	<?php print_r($activity); ?>
+	</pre>
+</div>
+
+<div class="BlueBox Box13b FlexiBoxLast">
+	<h2>links</h2>
+	<p><a href="/office/bylines">Bylines</a></p>
+	<h2>notifications</h2>
+	<?php
+	foreach ($notifications as $notification) {
+		if (!empty($notification['count'])) {
+			echo('<div><a href="' . $notification['link'] . '">' . $notification['count'] . ' ' . $notification['title'] . ($notification['count'] > 1 ? 's' : '') . '</a></div>');
+		}
+	}
+	?>
+</div>
+
+
+
+<?php /*
 <script type="text/javascript">
 onLoadFunctions.push(loadPage);
@@ -161,2 +211,5 @@
 	</div>
 </div>
+
+*/
+?>
