Index: system/application/views/crosswords/crossword.php
===================================================================
--- system/application/views/crosswords/crossword.php (revision 52a81e29044b174cdf7b4d2e16a67e8e7065a98b)
+++ system/application/views/crosswords/crossword.php (revision fbddb156e0fa4716316d368a8bbe0b731010af00)
@@ -101,5 +101,5 @@
 
 	?><h2><?php
-		?>crossword<?php
+		?>Crossword<?php
 		if (count($Crossword['author_fullnames']) > 0) {
 			?> by <?php
