Index: public_html/javascript/admin/permissions.js
===================================================================
--- public_html/javascript/admin/permissions.js (revision 506ff3d126fecad77ce063b28f8ebeffa8963cda)
+++ public_html/javascript/admin/permissions.js (revision 67717dad6377e929c553c7b95e31daef566af69e)
@@ -235,5 +235,5 @@
 			}
 		);
-		ajax.doGet();
+		ajax.doPost();
 	}
 	else {
Index: public_html/javascript/admin/static_analysis.js
===================================================================
--- public_html/javascript/admin/static_analysis.js (revision 743eeb603f9bb4da5be0aaf958ee60dfdf2a9372)
+++ public_html/javascript/admin/static_analysis.js (revision 67717dad6377e929c553c7b95e31daef566af69e)
@@ -25,5 +25,5 @@
 	post['tests'] = tests_string;
 	var ajax = new AJAXInteraction('/admin/tools/test/static/ajax', post, ajax_test_callback);
-	ajax.doGet();
+	ajax.doPost();
 }
 
Index: public_html/javascript/calendar_edit.js
===================================================================
--- public_html/javascript/calendar_edit.js (revision d4b75f0dfbca58260968286b720a83e2994de089)
+++ public_html/javascript/calendar_edit.js (revision 67717dad6377e929c553c7b95e31daef566af69e)
@@ -259,5 +259,5 @@
 	post['eved_recur_simple[range_method]'] = range_method;
 	var ajax = new AJAXInteraction(url, post, UpdateRecurCalPreviewCallback); 
-	ajax.doGet();
+	ajax.doPost();
 }
 
Index: public_html/javascript/crosswords.js
===================================================================
--- public_html/javascript/crosswords.js (revision 7024f4a5b339393b0c5a07b11a16baf03906c585)
+++ public_html/javascript/crosswords.js (revision 67717dad6377e929c553c7b95e31daef566af69e)
@@ -878,5 +878,5 @@
 
 		var ajax = new AJAXInteraction(action, post, postCallback, postCallbackFail);
-		ajax.doGet();
+		ajax.doPost();
 	}
 
Index: public_html/javascript/irc.js
===================================================================
--- public_html/javascript/irc.js (revision b62c2ebe264306112e178b13b02345accd62c849)
+++ public_html/javascript/irc.js (revision 67717dad6377e929c553c7b95e31daef566af69e)
@@ -296,5 +296,5 @@
 	post['channel'] = irc_screen_list[id];
 	var ajax = new AJAXInteraction(irc_ajax_url, post, irc_ajax_callback);
-	ajax.doGet();
+	ajax.doPost();
 }
 
@@ -325,5 +325,5 @@
 	post['channel'] = channel;
 	var ajax = new AJAXInteraction(irc_ajax_url, post, irc_ajax_callback);
-	ajax.doGet();
+	ajax.doPost();
 }
 
@@ -417,5 +417,5 @@
 	post['cmd'] = 'ping';
 	var ajax = new AJAXInteraction(irc_ajax_url, post, irc_ping_ajax_callback);
-	ajax.doGet();
+	ajax.doPost();
 }
 
@@ -673,5 +673,5 @@
 		} );
 	irc_clear_error();
-	ajax.doGet();
+	ajax.doPost();
 }
 
@@ -696,5 +696,5 @@
 			irc_current_screen = null;
 		} );
-	ajax.doGet();
+	ajax.doPost();
 	irc_connected = false;
 }
Index: public_html/javascript/ppedit_inline.js
===================================================================
--- public_html/javascript/ppedit_inline.js (revision 2094aa3cf8a1729df1628a90a4d85ef1d34b2255)
+++ public_html/javascript/ppedit_inline.js (revision 67717dad6377e929c553c7b95e31daef566af69e)
@@ -43,5 +43,5 @@
 		}
 	);
-	ajax.doGet();
+	ajax.doPost();
 	return false;
 }
Index: public_html/javascript/simple_ajax.js
===================================================================
--- public_html/javascript/simple_ajax.js (revision f78347eed0783cbbc00aa2b99aff08c70244510e)
+++ public_html/javascript/simple_ajax.js (revision 67717dad6377e929c553c7b95e31daef566af69e)
@@ -42,7 +42,7 @@
 		}
 	}
-	
-	this.doGet = function() {
-		// make a HTTP GET request to the URL asynchronously
+
+	this.getPostString = function()
+	{
 		var post_string = "";
 		var first = 1;
@@ -55,4 +55,18 @@
 			post_string += key+"="+encodeURIComponent(post[key]);
 		}
+		return post_string;
+	}
+	
+	this.doGet = function() {
+		// make a HTTP GET request to the URL asynchronously
+		var post_string = this.getPostString();
+		req.open("GET", url+'?'+post_string, true);
+		// send request
+		req.send(null);
+	}
+	
+	this.doPost = function() {
+		// make a HTTP POST request to the URL asynchronously
+		var post_string = this.getPostString();
 		req.open("POST", url, true);
 		// proper header information
Index: system/application/models/subcontrollers/calendar_subcontroller.php
===================================================================
--- system/application/models/subcontrollers/calendar_subcontroller.php (revision f78347eed0783cbbc00aa2b99aff08c70244510e)
+++ system/application/models/subcontrollers/calendar_subcontroller.php (revision 67717dad6377e929c553c7b95e31daef566af69e)
@@ -1855,6 +1855,6 @@
 		$error = false;
 		$years = array();
-		if (isset($_POST['years'])) {
-			$year_ranges = split(',',$_POST['years']);
+		if (isset($_GET['years'])) {
+			$year_ranges = split(',',$_GET['years']);
 			foreach ($year_ranges as $year_range) {
 				if ($year_range === '') {
