Index: public_html/javascript/crosswords.js
===================================================================
--- public_html/javascript/crosswords.js (revision 0177f0911757ca06f6cdab12bdf5ed0d16448096)
+++ public_html/javascript/crosswords.js (revision e40fb99f8646d9019648e7c2c173f3d79d958f6c)
@@ -1196,5 +1196,40 @@
 		return true;
 	}
-
+	
+	this.inlineAnswersUpdated = function()
+	{
+		var checkbox = document.getElementById(this.m_name+"-clues-inline");
+		if (null != checkbox) {
+			var show_inline = checkbox.checked;
+			var items = [
+				document.getElementById(this.m_name+"-0-clues"),
+				document.getElementById(this.m_name+"-1-clues")
+			];
+			for (var i = 0; i < items.length; ++i) {
+				(show_inline?CssRemove:CssAdd)(items[i], "hideValues");
+			}
+		}
+		return false;
+	}
+
+	this.clueTypeUpdated = function()
+	{
+		var cryptic_radio = document.getElementById(this.m_name+"-clues-show-cryptic");
+		if (null != cryptic_radio) {
+			var cryptic = cryptic_radio.checked;
+			var items = [
+				document.getElementById(this.m_name+"-0-clues"),
+				document.getElementById(this.m_name+"-1-clues")
+			];
+			for (var i = 0; i < items.length; ++i) {
+				(cryptic ? CssRemove : CssAdd)(items[i], "hideCryptic");
+				(cryptic ? CssAdd : CssRemove)(items[i], "hideQuick");
+			}
+		}
+		return false;
+	}
+
+	this.inlineAnswersUpdated();
+	this.clueTypeUpdated();
 	this.updateCompleteness();
 }
@@ -1239,30 +1274,11 @@
 }
 
-function crosswordToggleInlineAnswers(name)
-{
-	var items = [
-		document.getElementById(name+"-0-clues"),
-		document.getElementById(name+"-1-clues")
-	];
-	var hideValues = CssCheck(items[0], "hideValues");
-	for (var i = 0; i < items.length; ++i) {
-		CssToggle(items[i], "hideValues");
-	}
-	var link = document.getElementById("toggleInlineAnswers");
-	setInnerText(link, (hideValues ? "Hide inline answers" : "Show inline answers"));
-}
-
-function crosswordToggleCrypticClues(name)
-{
-	var items = [
-		document.getElementById(name+"-0-clues"),
-		document.getElementById(name+"-1-clues")
-	];
-	var hideCryptic = CssCheck(items[0], "hideCryptic");
-	for (var i = 0; i < items.length; ++i) {
-		CssToggle(items[i], "hideCryptic");
-		CssToggle(items[i], "hideQuick");
-	}
-	var link = document.getElementById("toggleCrypticClues");
-	setInnerText(link, (hideCryptic ? "Show quick clues" : "Show cryptic clues"));
-}
+function crosswordInlineAnswersUpdated(name)
+{
+	crossword(name).inlineAnswersUpdated();
+}
+
+function crosswordClueTypeUpdated(name)
+{
+	crossword(name).clueTypeUpdated();
+}
Index: public_html/stylesheets/crosswords.css
===================================================================
--- public_html/stylesheets/crosswords.css (revision 491bfe1f7619eec30370e3b588656e16de9660dc)
+++ public_html/stylesheets/crosswords.css (revision e40fb99f8646d9019648e7c2c173f3d79d958f6c)
@@ -167,4 +167,23 @@
 }
 
+div.crosswordCluesHeader {
+	float: left;
+	width: 60%;
+	padding: 0px;
+	margin: 0px;
+}
+div.crosswordCluesHeader fieldset {
+	float: left;
+	width: 33%;
+}
+div.crosswordCluesHeader fieldset label {
+	float: right;
+	width: 75%;
+	text-align: left;
+}
+div.crosswordCluesHeader fieldset input {
+	float: right;
+}
+
 div.crosswordCluesBox {
 	float: left;
@@ -174,4 +193,5 @@
 }
 
+div.crosswordCluesHeader div.header,
 div.crosswordCluesBox div.horizontal,
 div.crosswordCluesBox div.vertical {
Index: system/application/helpers/crosswords_helper.php
===================================================================
--- system/application/helpers/crosswords_helper.php (revision 0177f0911757ca06f6cdab12bdf5ed0d16448096)
+++ system/application/helpers/crosswords_helper.php (revision e40fb99f8646d9019648e7c2c173f3d79d958f6c)
@@ -688,5 +688,5 @@
 			?><div id="<?php echo("$name-complete"); ?>" class="crosswordAjaxNotify hidden"><?php
 				?><fieldset><input	class="button" type="button" value="submit for marking" <?php
-								?>	onclick="<?php echo(xml_escape('crossword("xw").submit()')); ?>"<?php
+								?>	onclick="<?php echo(xml_escape("crossword('$name').submit()")); ?>"<?php
 								?>	/>crossword complete</fieldset><?php
 			?></div><?php
@@ -757,9 +757,52 @@
 		}
 		?></table><?php
-
-		if (!$this->m_readonly) {
-			?><a id="toggleInlineAnswers" onclick="<?php echo(xml_escape('crosswordToggleInlineAnswers("xw");')); ?>">Show inline answers</a><br /><?php
-		}
-		?><a id="toggleCrypticClues" onclick="<?php echo(xml_escape('crosswordToggleCrypticClues("xw");')); ?>">Show cryptic clues</a><br /><?php
+		?></div><?php
+
+		// Clues bar
+		?><div class="crosswordCluesHeader"><?php
+			?><div class="header"><?php
+				// Toggles inline display of grid cells for each clue
+				if (!$this->m_readonly) {
+					?><fieldset><?php
+						?><label	for="<?php echo("$name-clues-inline"); ?>"><?php
+						if ($this->m_edit) {
+							?>show solutions with clues<?php
+						}
+						else {
+							?>show your answers with the clues<?php
+						}
+						?></label><?php
+						?><input	id="<?php echo("$name-clues-inline"); ?>"<?php
+								?>	type="checkbox"<?php
+								?>	onchange="<?php echo(xml_escape("return crosswordInlineAnswersUpdated('$name');")); ?>"<?php
+								?>	/><?php
+					?></fieldset><?php
+				}
+				// Choice between quick and cryptic clues
+				if (true) {
+					?><fieldset><?php
+						?><label	for="<?php echo("$name-clues-show-quick"); ?>">show quick clues</label><?php
+						?><input	id="<?php echo("$name-clues-show-quick"); ?>"<?php
+								?>	name="<?php echo("$name[cluetype]"); ?>"<?php
+								?>	class="radio"<?php
+								?>	type="radio"<?php
+								?>	value="quick"<?php
+								?>	onchange="<?php echo(xml_escape("return crosswordClueTypeUpdated('$name');")); ?>"<?php
+								?>	checked="checked"<?php
+								?>	/><?php
+					?></fieldset><?php
+					?><fieldset><?php
+						?><label	for="<?php echo("$name-clues-show-cryptic"); ?>">show cryptic clues</label><?php
+						?><input	id="<?php echo("$name-clues-show-cryptic"); ?>"<?php
+								?>	name="<?php echo("$name[cluetype]"); ?>"<?php
+								?>	class="radio"<?php
+								?>	type="radio"<?php
+								?>	value="cryptic"<?php
+								?>	onchange="<?php echo(xml_escape("return crosswordClueTypeUpdated('$name');")); ?>"<?php
+								?>	/><?php
+					?></fieldset><?php
+				}
+				?><div style="clear:both"></div><?php
+			?></div><?php
 		?></div><?php
 
