Index: system/application/controllers/roses.php
===================================================================
--- system/application/controllers/roses.php (revision f10f231ff8b59a44e2fa0d93b04470b268bfee1f)
+++ system/application/controllers/roses.php (revision c200b1b61d1537a115f7e31a26da802fc737bd3e)
@@ -77,5 +77,5 @@
 				'extras'		=>	'license,owner,date_upload',
 				'per_page'		=>	'500',
-				'page'			=>	'1'
+				'page'			=>	'2'
 			);
 			$encoded_params = array();
