Index: export/template/system/application/config/yorker_override.php
===================================================================
--- export/template/system/application/config/yorker_override.php (revision e71456289bd48cec1d420c382d2b5a0bfff9fb24)
+++ export/template/system/application/config/yorker_override.php (revision 8a2b4dee1d7fa025bde919d11c63292b08a19cc0)
@@ -13,3 +13,6 @@
 $config['enable_analytics'] = true;
 
+$config['temp_local_path'] = '/srv/www/public_html/tmp/uploads/';
+$config['static_local_path'] = '/srv/static';
+
 ?>
