Index: system/application/controllers/office/liveblog.php
===================================================================
--- system/application/controllers/office/liveblog.php (revision e856b510970bdeafc371a2aad5f150fcc6a96cab)
+++ system/application/controllers/office/liveblog.php (revision 7e42227166b7c13c638a37815515d6ab588080e2)
@@ -709,6 +709,6 @@
 
 		// Range
-		$o .= '&chxr=1,0,140,25';
-		$o .= '&chds=0,140,0,140';
+		$o .= '&chxr=1,0,180,25';
+		$o .= '&chds=0,180,0,180';
 
 		$sql = 'SELECT * FROM roses_scores WHERE event_score_time IS NOT NULL ORDER BY event_score_time ASC';
