Index: design/theyorker_tables.sql
===================================================================
--- design/theyorker_tables.sql (revision 694495289206ab1a0d3101116ebd7a173e972e6d)
+++ design/theyorker_tables.sql (revision 2b775a15f8eea0938ca6ac94d9e0b84fb2bf46c9)
@@ -649,12 +649,12 @@
 CREATE TABLE progress_reports (
 	progress_report_id 				INTEGER 	NOT NULL 	AUTO_INCREMENT,
-	progress_report_charity_id 			INTEGER 	NOT NULL,
-	progress_report_campaign_id 			INTEGER 	NOT NULL,
+	progress_report_charity_id 			INTEGER 	NULL		COMMENT='Either this or progress_report_campaign_id cant be null.',
+	progress_report_campaign_id 			INTEGER 	NULL		COMMENT='Either this or progress_report_charity_id cant be null.',
+	progress_report_last_update_user_entity_id	INTEGER		NOT NULL,
 	progress_report_text 				TEXT 		NOT NULL,
-	progress_report_order 				INTEGER 	NOT NULL,
 	progress_report_deleted 			BOOL 		NOT NULL,
-	progress_report_timestamp 			TIMESTAMP	NOT NULL	DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
-	progress_report_good_bad 			ENUM('good', 'bad') NOT NULL,
-	
+	progress_report_last_updated 			TIMESTAMP	NOT NULL	DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+	progress_report_publish_date 			TIMESTAMP	NOT NULL,
+
 	PRIMARY KEY(progress_report_id)
 ) COMMENT='Contains updates on how parts of a campaign are going.';
