Index: system/application/config/routes.php
===================================================================
--- system/application/config/routes.php (revision 882b087a2c4fa7f8b0e5a143077fa946f8d04f04)
+++ system/application/config/routes.php (revision 56aed1a98154044c59890bf4a04c459f6f886fec)
@@ -34,5 +34,5 @@
 */
 
-$route['default_controller'] = "roses";
+$route['default_controller'] = "home";
 $route['scaffolding_trigger'] = "";
 
