Index: system/application/libraries/Wikiparser.php
===================================================================
--- system/application/libraries/Wikiparser.php (revision ed47f54b421284966d39e2bef38df3a5ca6156a2)
+++ system/application/libraries/Wikiparser.php (revision 80183dd21746619bf6a91fc4d3bd61897dcfefb1)
@@ -409,6 +409,17 @@
 			$mediaplayer_count++;
 			$mediaplayer_type = (!empty($matches[5])) ? str_replace('|', '', $matches[5]) : '';
-			$control_width = $mediaplayer_type == 'sound' || (strlen($href) > 4 && substr($href, -4) == '.mp3') ? 340 : 580;
-			$control_height = $mediaplayer_type == 'sound' || (strlen($href) > 4 && substr($href, -4) == '.mp3') ? 20 : 346;
+			if ($mediaplayer_type == 'ury') {
+				// URY Audio
+				$control_width = 340;
+				$control_height = 50;
+			} elseif ($mediaplayer_type == 'sound' || (strlen($href) > 4 && substr($href, -4) == '.mp3')) {
+				// Normal Audio
+				$control_width = 340;
+				$control_height = 20;
+			} else {
+				// Video
+				$control_width = 580;
+				$control_height = 346;
+			}
 			$output = '
 				<div id="mp' . $mediaplayer_count . '_container" style="text-align:center">
@@ -425,16 +436,33 @@
 				// File properties
 				so.addVariable("file","' . $href . '");
-' . (!empty($mediaplayer_type) ? 'so.addVariable("type","' . $mediaplayer_type . '");' : '') . '
-//				so.addVariable("image","##TODO##");
+				';
+			if (!empty($mediaplayer_type)) {
+				switch ($mediaplayer_type) {
+					case 'ury':
+						$output .= 'so.addVariable("image",/images/prototype/news/mediaplayer-ury.png"");';
+						$output .= 'so.addVariable("link","http://ury.york.ac.uk");';
+						$output .= 'so.addVariable("displayclick","link");';
+						$output .= 'so.addVariable("icons","false");';
+						$output .= 'so.addVariable("type","sound");';
+						$output .= 'so.addVariable("screencolor","000000");';
+						break;
+					default:
+						$output .= 'so.addVariable("type","' . $mediaplayer_type .'");';
+						$output .= 'so.addVariable("screencolor","FFFFFF");';
+				}
+			} else {
+				$output .= 'so.addVariable("logo","/images/prototype/news/video_overlay_icon.png");';
+				$output .= 'so.addVariable("screencolor","FFFFFF");';
+			}
+			$output .= '
 				// Colours
 				so.addVariable("backcolor","FF6A00");
 				so.addVariable("frontcolor","FFFFFF");
 				so.addVariable("lightcolor","FFFFFF");
-				so.addVariable("screencolor","FFFFFF");
 				// Layout
 				so.addVariable("height","' . $control_height . '");
 				so.addVariable("width","' . $control_width . '");
 				// Behaviour
-				so.addVariable("logo","/images/prototype/news/video_overlay_icon.png");
+				so.addVariable("volume","100");
 				// External
 				so.addVariable("id","' . $href . '");
