Index: system/application/controllers/office/liveblog.php
===================================================================
--- system/application/controllers/office/liveblog.php (revision 882b087a2c4fa7f8b0e5a143077fa946f8d04f04)
+++ system/application/controllers/office/liveblog.php (revision ebacb92e23a4397ceaf2e76d44b1a5251178d770)
@@ -303,6 +303,6 @@
 			// Set winning team indicators
 			$this->wikiparser->add_image_override(-1, '<img src="/images/version2/rose_lancashire.png" alt="Lancaster" />', 'Lancaster Win');
-			$this->wikiparser->add_image_override(-2, '<img src="/images/version2/news/rose_yorkshire.png" alt="York Win" />', 'York Win');
-			$this->wikiparser->add_image_override(-3, '<img src="/images/version2/news/rose_draw.png" alt="Draw" />', 'Draw');
+			$this->wikiparser->add_image_override(-2, '<img src="/images/version2/rose_yorkshire.png" alt="York Win" />', 'York Win');
+			$this->wikiparser->add_image_override(-3, '<img src="/images/version2/rose_draw.png" alt="Draw" />', 'Draw');
 			$photo_requests = $this->photos_model->GetPhotoRequestsForArticle($article_id);
 			foreach ($photo_requests as $photo) {
@@ -489,7 +489,7 @@
 									$this->roses_model->setResult($event->event_id, $score_lancs, $score_york, date('Y-m-d H:i:s'));
 									// Set winning team indicators
-									$this->wikiparser->add_image_override(-1, '<img src="/images/prototype/news/rose_lancashire.png" alt="Lancaster" />', 'Lancaster Win');
-									$this->wikiparser->add_image_override(-2, '<img src="/images/prototype/news/rose_yorkshire.png" alt="York Win" />', 'York Win');
-									$this->wikiparser->add_image_override(-3, '<img src="/images/prototype/news/rose_draw.png" alt="Draw" />', 'Draw');
+									$this->wikiparser->add_image_override(-1, '<img src="/images/version2/rose_lancashire.png" alt="Lancaster" />', 'Lancaster Win');
+									$this->wikiparser->add_image_override(-2, '<img src="/images/version2/rose_yorkshire.png" alt="York Win" />', 'York Win');
+									$this->wikiparser->add_image_override(-3, '<img src="/images/version2/rose_draw.png" alt="Draw" />', 'Draw');
 									// Add posted time to blog entry
 									if ($event->event_score_time === NULL) {
@@ -619,7 +619,7 @@
 		$this->load->library('wikiparser');
 		// Set winning team indicators
-		$this->wikiparser->add_image_override(-1, '<img src="/images/prototype/news/rose_lancashire.png" alt="Lancaster" />', 'Lancaster Win');
-		$this->wikiparser->add_image_override(-2, '<img src="/images/prototype/news/rose_yorkshire.png" alt="York Win" />', 'York Win');
-		$this->wikiparser->add_image_override(-3, '<img src="/images/prototype/news/rose_draw.png" alt="Draw" />', 'Draw');
+		$this->wikiparser->add_image_override(-1, '<img src="/images/version2/rose_lancashire.png" alt="Lancaster" />', 'Lancaster Win');
+		$this->wikiparser->add_image_override(-2, '<img src="/images/version2/rose_yorkshire.png" alt="York Win" />', 'York Win');
+		$this->wikiparser->add_image_override(-3, '<img src="/images/version2/rose_draw.png" alt="Draw" />', 'Draw');
 		$result = $this->roses_model->getAllResults();
 		$content = '';
